{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "18e42d79-8ef3-4ea2-9941-ac4d580ca668",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import time\n",
    "import requests\n",
    "import sys\n",
    "from datetime import datetime\n",
    "\n",
    "def post_json_request(url, obj):\n",
    "    return requests.post(url, json=obj).json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "444e5d0f-8eae-4e7c-a02c-450b6eb8decd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 1109.2847278118134 seconds ---\n"
     ]
    }
   ],
   "source": [
    "database=\"test10\"\n",
    "project=\"adc_cali\"\n",
    "start_time = time.time()\n",
    "pubmedResult = post_json_request(\n",
    "    \"http://localhost:5000/query\",\n",
    "    {\n",
    "        \"database\": database,\n",
    "        \"project\": project,\n",
    "        \"maxdocs\": 50,\n",
    "        \"patternid\": 1,\n",
    "        \"query\": \"breast[Title/Abstract] AND cancer[Title/Abstract]\",\n",
    "    },\n",
    ")[0][\"exit\"]\n",
    "print(\"--- %s seconds ---\" % (time.time() - start_time))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7d25f509-7147-4fae-af97-10e3330419e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "print(pubmedResult)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "260f61f9-b046-45c9-a970-52328d5f8f6b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['_id', 'title', 'abstract', 'authors', 'dbid', 'doi', 'lang', 'org', 'pat_id', 'url', 'year'])"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_documents = post_json_request(\n",
    "    \"http://localhost:5001/mongo-doc-list\",\n",
    "    {\n",
    "        \"db_name\": database,\n",
    "        \"coll_name\": f\"{project}_metadata_global\"\n",
    "    },\n",
    ")\n",
    "metadata_documents[0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "12e48f87-d548-42c6-8775-2ff5fd3e84ea",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['_id', 'author', 'doc_id', 'doi'])"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "authors_documents = post_json_request(\n",
    "    \"http://localhost:5001/mongo-doc-list\",\n",
    "    {\n",
    "        \"db_name\": database,\n",
    "        \"coll_name\": f\"{project}_author_vs_doc_id_global\"\n",
    "    },\n",
    ")\n",
    "authors_documents[0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "874002e2-4cff-4b6d-bea3-e3934bb77faf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36933187&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36933062&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36933050&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932957&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932886&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932743&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932670&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932388&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932252&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932246&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932232&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932156&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36932003&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36931740&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36931265&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36931171&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36931164&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36931128&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36931039&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36931031&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930999&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930943&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930854&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930833&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930678&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930677&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930555&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930460&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930419&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930347&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930346&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930345&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930344&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930147&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36930083&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929983&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929975&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929946&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929942&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929848&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929759&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929748&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929572&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929295&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929288&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929229&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36929087&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36928988&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36928951&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36928924&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36940464&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36940420&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36940301&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36940196&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36940038&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939982&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939976&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939902&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939876&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939778&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939607&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939483&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939453&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939445&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939381&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939370&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939293&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939143&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36939123&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36938944&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36938928&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36938826&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36938720&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36938678&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36938303&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36938225&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36938187&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937987&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937848&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937820&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937603&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937585&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937550&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937444&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937408&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937402&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937398&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937388&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36937381&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36936994&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36936536&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36936429&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36936420&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36936412&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36936362&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36936274&retmode=ref&cmd=prlinks',\n",
       " 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id=36936256&retmode=ref&cmd=prlinks']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[f\"http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&id={document.get('dbid')}&retmode=ref&cmd=prlinks\" for document in metadata_documents]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8777306e-ae64-4f36-89c4-794731b9a1f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "36933187, 36933062, 36933050, 36932957, 36932886, 36932743, 36932670, 36932388, 36932252, 36932246, 36932232, 36932156, 36932003, 36931740, 36931265, 36931171, 36931164, 36931128, 36931039, 36931031, 36930999, 36930943, 36930854, 36930833, 36930678, 36930677, 36930555, 36930460, 36930419, 36930347, 36930346, 36930345, 36930344, 36930147, 36930083, 36929983, 36929975, 36929946, 36929942, 36929848, 36929759, 36929748, 36929572, 36929295, 36929288, 36929229, 36929087, 36928988, 36928951, 36928924, 36940464, 36940420, 36940301, 36940196, 36940038, 36939982, 36939976, 36939902, 36939876, 36939778, 36939607, 36939483, 36939453, 36939445, 36939381, 36939370, 36939293, 36939143, 36939123, 36938944, 36938928, 36938826, 36938720, 36938678, 36938303, 36938225, 36938187, 36937987, 36937848, 36937820, 36937603, 36937585, 36937550, , 36937444, 36937408, 36937402, 36937398, 36937388, 36937381, 36936994, 36936536, 36936429, 36936420, 36936412, 36936362, 36936274, 36936256\n"
     ]
    }
   ],
   "source": [
    "print(\", \".join([document.get('dbid') for document in metadata_documents]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "02be493f-57f9-444f-a9fc-33d72e968890",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'url': None},\n",
       " {'url': 'https://link.springer.com/content/pdf/10.1007/s00018-023-04734-7.pdf'},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': 'http://www.plosone.org/article/fetchObject.action?uri=info:doi/10.1371/journal.pcbi.1010939&representation=PDF'},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': 'https://link.springer.com/content/pdf/10.1007/s10549-022-06815-w.pdf'},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': 'http://www.thelancet.com/pdfs/journals/lancet/PIIS0140-6736(22)02249-8.pdf'},\n",
       " {'url': 'http://www.thelancet.com/pdfs/journals/lancet/PIIS0140-6736(22)02241-3.pdf'},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020071&blobtype=pdf'},\n",
       " {'url': None},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020396&blobtype=pdf'},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': 'https://link.springer.com/content/pdf/10.1007/s00428-023-03527-4.pdf'},\n",
       " {'url': 'https://link.springer.com/content/pdf/10.1007/s00421-023-05177-5.pdf'},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10017099&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10017101&blobtype=pdf'},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': None},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020228&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10016023&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018230&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10022397&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018043&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10021088&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10019877&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10016607&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10016366&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10017187&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020607&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10014909&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018193&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10021108&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10019818&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018141&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020627&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10022533&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018014&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020208&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10014556&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020917&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020702&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10017271&blobtype=pdf'}]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "urlList=list(map(lambda document: {\"url\": document[\"url\"]}, metadata_documents))\n",
    "urlList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "65f41ae2-767c-45aa-a75c-9460ba211a19",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "avaliable pdfs: 35.714285714285715%\n"
     ]
    }
   ],
   "source": [
    "availablePDFs=len(list([url_dict[\"url\"]] for url_dict in urlList if url_dict[\"url\"]))/len(urlList)*100\n",
    "print(f\"avaliable pdfs: {availablePDFs}%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "37a3de6b-5e50-4274-b514-4e4debc92199",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'htext': 'MEDICINE INTERNATIONAL 3: 11, 2023\\n\\nThe combination of positive anti‑WDR1 antibodies with\\nnegative anti‑CFL1 antibodies in serum is a poor prognostic\\nfactor for patients with esophageal carcinoma\\nMASAAKI ITO1, SATOSHI YAJIMA2, TAKASHI SUZUKI2, YOKO OSHIMA2, TATSUKI NANAMI2,\\nMAKOTO SUMAZAKI2, FUMIAKI SHIRATORI2, HAO WANG3,4, LIUBING HU4, HIROTAKA TAKIZAWA5,\\nSHU‑YANG LI6, YASUO IWADATE6, TAKAKI HIWASA1,6 and HIDEAKI SHIMADA1,2\\n1\\n\\nDepartment of Clinical Oncology, Toho University Graduate School of Medicine; 2Department of Gastroenterological Surgery,\\nToho University School of Medicine, Tokyo 143‑8541, Japan; 3Stroke Center, The First Affiliated Hospital, Jinan University;\\n4\\nDepartment of Anesthesiology, Stroke Center, The First Affiliated Hospital and Health Science Center, Jinan University,\\nGuangzhou, Guangdong 510630, P.R. China; 5Port Square Kashiwado Clinic, Kashiwado Memorial Foundation,\\nChiba 260‑0025; 6Department of Neurological Surgery, Graduate School of Medicine, Chiba University, Chiba 260‑8670, Japan\\nReceived October 8, 2022; Accepted January 23, 2023\\nDOI: 10.3892/mi.2023.71\\n\\nAbstract. WD repeat‑containing protein 1 (WDR1) regulates\\nthe cofilin 1 (CFL1) activity, promotes cytoskeleton remod‑\\neling, and thus, facilitates cell migration and invasion. A\\nprevious study reported that autoantibodies against CFL1 and\\nβ‑actin were useful biomarkers for diagnosing and predicting\\nthe prognosis of patients with esophageal carcinoma.\\nTherefore, the present study aimed to evaluate the serum\\nlevels of anti‑WDR1 antibodies (s‑WDR1‑Abs) combined\\nwith serum levels of anti‑CFL1 antibodies (s‑CFL1‑Abs) in\\npatients with esophageal carcinoma. Serum samples obtained\\nfrom 192 patients with esophageal carcinoma and other solid\\ncancers. And s‑WDR1‑Ab and s‑CFL1‑Ab titers were analyzed\\nusing the amplified luminescent proximity homogeneous\\nassay‑linked immunosorbent assay. Compared with those\\nof healthy donors, the s‑WDR1‑Ab levels were significantly\\nhigher in the 192 patients with esophageal, whereas these\\nwere not significantly higher in the samples from patients with\\ngastric, colorectal, lung, or breast cancer. In 91 patients treated\\nwith surgery, sex, tumor depth, lymph node metastasis, stage\\nand C‑reactive protein levels were significantly associated\\nwith overall survival, as determined using the log‑rank test,\\nwhereas the squamous cell carcinoma antigen, p53 antibody\\nand s‑WDR1‑Ab levels tended to be associated with a worse\\nprognosis. Although no significant difference was observed\\nin the survival between the positive and negative groups of\\n\\nCorrespondence to: Dr Hideaki Shimada, Department of\\nGastroenterological Surgery, Toho University School of Medicine,\\n6‑11‑1 Omori‑Nishi, Ota‑ku, Tokyo 143‑8541, Japan\\nE‑mail: hideaki.shimada@med.toho‑u.ac.jp\\n\\nKey words: WD repeat‑containing protein 1, cofilin 1, esophageal\\ncarcinoma, antibody biomarker, overall survival\\n\\ns‑WDR1‑Abs or s‑CFL1‑Abs alone in the Kaplan‑Meier test,\\nthe patients in the s‑WDR1‑Ab‑positive and s‑CFL1‑Ab‑nega‑\\ntive groups exhibited a significantly poorer prognosis in the\\noverall survival analysis. On the whole, the present study\\ndemonstrates that the combination of positive anti‑WDR1 anti‑\\nbodies with negative anti‑CFL1 antibodies in serum may be a\\npoor prognostic factor for patients with esophageal carcinoma.\\nIntroduction\\nEsophageal carcinoma is a rapidly progressive disease and\\nis life‑threatening even at the early stages. The therapeutic\\nefficacy of esophageal carcinoma is less satisfactory than that\\nof several other types of cancer; therefore, the detection and\\ndiagnosis at the early stage of carcinoma are indispensable for\\nimproving the therapeutic outcomes of patients (1).\\nThe serum levels of several antigens and antibodies have\\nlong been measured using the serological analysis of recom‑\\nbinant tumor cDNA expression libraries (SEREX) method, a\\nuseful screening method for tumor markers (2). SEREX uses\\npatient serum to immunoscreen cDNA libraries prepared from\\ncancer specimens. By sequencing the isolated cDNA clones,\\nthe SEREX method is suitable as for the large‑scale screening\\nfor tumor antigens. Large‑scale SEREX screening has already\\nidentified numerous antibodies or antigens, such as trophoblast\\ncell surface antigen 2 (3), solute carrier family 2 member 1 (4),\\nstriatin 4 (5), LDL receptor related protein associated protein\\n1 (6), proprotein convertase subtilisin/kexin type 9 (7), cofilin\\n1 (CFL1) and β ‑actin (ACTB) (8). CFL1 can depolymerize\\nF‑actin in a pH‑dependent manner and is also involved in lung,\\npancreatic, and gastric cancer invasion and metastasis (9‑11).\\nWD repeat‑containing protein 1 (WDR1; also known as\\nactin‑interacting protein 1) regulates cofilin activity, promotes\\ncytoskeletal remodeling, and thus facilitates cell migration and\\ninvasion (12). The presence of autoantibodies against WDR1\\nhas been found in the sera of patients who have suffered a\\ntransient ischemic attack using SEREX screening (13). The\\n\\n\\x0c'}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def url2head(url):\n",
    "    try:\n",
    "        return post_json_request(\n",
    "            \"http://localhost:5002/url2htext\",\n",
    "            { \"url\": url }\n",
    "        )\n",
    "    except:\n",
    "        return {\"htext\": \"\"}\n",
    "url2head(\"http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC9983066&blobtype=pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "47e9632c-4336-4cc2-9a6b-4186b69bce13",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'url': 'https://link.springer.com/content/pdf/10.1007/s00018-023-04734-7.pdf'},\n",
       " {'url': 'http://www.plosone.org/article/fetchObject.action?uri=info:doi/10.1371/journal.pcbi.1010939&representation=PDF'},\n",
       " {'url': 'https://link.springer.com/content/pdf/10.1007/s10549-022-06815-w.pdf'},\n",
       " {'url': 'http://www.thelancet.com/pdfs/journals/lancet/PIIS0140-6736(22)02249-8.pdf'},\n",
       " {'url': 'http://www.thelancet.com/pdfs/journals/lancet/PIIS0140-6736(22)02241-3.pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020071&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020396&blobtype=pdf'},\n",
       " {'url': 'https://link.springer.com/content/pdf/10.1007/s00428-023-03527-4.pdf'},\n",
       " {'url': 'https://link.springer.com/content/pdf/10.1007/s00421-023-05177-5.pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10017099&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10017101&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020228&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10016023&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018230&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10022397&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018043&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10021088&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10019877&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10016607&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10016366&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10017187&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020607&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10014909&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018193&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10021108&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10019818&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018141&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020627&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10022533&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10018014&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020208&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10014556&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020917&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10020702&blobtype=pdf'},\n",
       " {'url': 'http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC10017271&blobtype=pdf'}]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "urls = list(filter(lambda url: url[\"url\"] != None, urlList))\n",
    "urls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bd528fd0-a3db-41ef-bc52-c905c576f323",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 1131.2687497138977 seconds ---\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['Cellular and Molecular Life Sciences\\n(2023) 80:100\\nhttps://doi.org/10.1007/s00018-023-04734-7\\n\\nCellular and Molecular Life Sciences\\n\\nORIGINAL ARTICLE\\n\\nLoss of Kmt2c in vivo leads to EMT, mitochondrial dysfunction\\nand improved response to lapatinib in breast cancer\\nNikiana Simigdala1 · Anna Chalari1 · Aimilia D. Sklirou2 · Evangelia Chavdoula1,3,4 · George Papafotiou1 ·\\nPelagia Melissa1 · Aimilia Kafalidou1 · Nikolaos Paschalidis1 · Ioannis S. Pateras5 · Emmanouil Athanasiadis1 ·\\nDimitris Konstantopoulos6 · Ioannis P. Trougakos2 · Apostolos Klinakis1\\nReceived: 4 May 2022 / Revised: 22 January 2023 / Accepted: 22 February 2023\\n© The Author(s) 2023\\n\\nAbstract\\nDeep sequencing of human tumours has uncovered a previously unappreciated role for epigenetic regulators in tumorigenesis.\\nH3K4 methyltransferase KMT2C/MLL3 is mutated in several solid malignancies, including more than 10% of breast tumours.\\nTo study the tumour suppressor role of KMT2C in breast cancer, we generated mouse models of Erbb2/Neu, Myc or PIK3CAdriven tumorigenesis, in which the Kmt2c locus is knocked out specifically in the luminal lineage of mouse mammary glands\\nusing the Cre recombinase. Kmt2c knock out mice develop tumours earlier, irrespective of the oncogene, assigning a bona\\nfide tumour suppressor role for KMT2C in mammary tumorigenesis. Loss of Kmt2c induces extensive epigenetic and transcriptional changes, which lead to increased ERK1/2 activity, extracellular matrix re-organization, epithelial-to-mesenchymal\\ntransition and mitochondrial dysfunction, the latter associated with increased reactive oxygen species production. Loss of\\nKmt2c renders the Erbb2/Neu-driven tumours more responsive to lapatinib. Publicly available clinical datasets revealed\\nan association of low Kmt2c gene expression and better long-term outcome. Collectively, our findings solidify the role of\\nKMT2C as a tumour suppressor in breast cancer and identify dependencies that could be therapeutically amenable.\\nKeywords KMT2C · Tumour suppressor · EMT · Breast cancer · Mitochondrial respiration · Lapatinib\\n* Nikiana Simigdala\\nnsimigdala@bioacademy.gr\\n\\nDimitris Konstantopoulos\\nkonstantopoulos@fleming.gr\\n\\n* Apostolos Klinakis\\naklinakis@bioacademy.gr\\n\\nIoannis P. Trougakos\\nitrougakos@biol.uoa.gr\\n\\nAnna Chalari\\nachalari@bioacademy.gr\\n\\n1\\n\\nAimilia D. Sklirou\\nasklirou@biol.uoa.gr\\n\\nPresent Address: Biomedical Research Foundation Academy\\nof Athens, Athens, Greece\\n\\n2\\n\\nEvangelia Chavdoula\\nechavdoula@gmail.com\\n\\nDepartment of Cell Biology and Biophysics, Faculty\\nof Biology, National and Kapodistrian University of Athens,\\nAthens, Greece\\n\\n3\\n\\nGeorge Papafotiou\\ngeo.papafotiou@gmail.com\\n\\nDepartment of Cancer Biology and Genetics, The Ohio State\\nUniversity, Columbus, OH, USA\\n\\n4\\n\\nThe Ohio State University Comprehensive Cancer\\nCenter-Arthur G. James Cancer Hospital and Richard\\nJ. Solove Research Institute, Columbus, OH, USA\\n\\n5\\n\\n2nd Department of Pathology, Medical School, “Attikon”\\nUniversity Hospital, National and Kapodistrian University\\nof Athens, Athens, Greece\\n\\n6\\n\\nInstitute for Fundamental Biomedical Research, BSRC\\n‘Alexander Fleming’, 16672 Vari, Greece\\n\\nPelagia Melissa\\npmelissa@bioacademy.gr\\nAimilia Kafalidou\\nakafalidou@bioacademy.gr\\nNikolaos Paschalidis\\nnpaschal@bioacademy.gr\\nIoannis S. Pateras\\nipateras@med.uoa.gr\\nEmmanouil Athanasiadis\\nmathan@bioacademy.gr\\n\\n13\\n\\nVol.:(0123456789)\\n\\n\\x0c',\n",
       " 'PLOS COMPUTATIONAL BIOLOGY\\n\\nRESEARCH ARTICLE\\n\\nPCB: A pseudotemporal causality-based\\nBayesian approach to identify EMT-associated\\nregulatory relationships of AS events and\\nRBPs during breast cancer progression\\nLiangjie Sun1, Yushan Qiu ID2*, Wai-Ki Ching1, Pu Zhao3, Quan Zou ID4\\n\\na1111111111\\na1111111111\\na1111111111\\na1111111111\\na1111111111\\n\\n1 Department of Mathematics, The University of Hong Kong, Hong Kong, China, 2 College of Mathematics\\nand Statistics, Shenzhen University, Shenzhen, China, 3 College of Life and Health Sciences, Northeastern\\nUniversity, Shenyang, China, 4 Institute of Fundamental and Frontier Sciences, University of Electronic\\nScience and Technology of China, Chengdu, China\\n* yushanqiu2526374@163.com\\n\\nAbstract\\nOPEN ACCESS\\nCitation: Sun L, Qiu Y, Ching W-K, Zhao P, Zou Q\\n(2023) PCB: A pseudotemporal causality-based\\nBayesian approach to identify EMT-associated\\nregulatory relationships of AS events and RBPs\\nduring breast cancer progression. PLoS Comput\\nBiol 19(3): e1010939. https://doi.org/10.1371/\\njournal.pcbi.1010939\\nEditor: Ananda Mondal, Florida International\\nUniversity, UNITED STATES\\nReceived: July 19, 2022\\nAccepted: February 9, 2023\\nPublished: March 17, 2023\\nCopyright: © 2023 Sun et al. This is an open\\naccess article distributed under the terms of the\\nCreative Commons Attribution License, which\\npermits unrestricted use, distribution, and\\nreproduction in any medium, provided the original\\nauthor and source are credited.\\nData Availability Statement: Processed TCGA\\nBRCA Level 3 RNA-SeqV2 gene expression data\\nwere downloaded from the Genomic Data\\nCommons (GDC) Legacy Archive (https://portal.\\ngdc.cancer.gov/legacy-archive) The data and code\\nfor PCB are available at: http://hkumath.hku.hk/\\n*wkc/PCB(data+code).zip.\\nFunding: YQ was supported by grants from the\\nNational Natural Science Foundation of China\\n[grant number 62002234] the Guangdong Basic\\n\\nDuring breast cancer metastasis, the developmental process epithelial-mesenchymal (EM)\\ntransition is abnormally activated. Transcriptional regulatory networks controlling EM transition are well-studied; however, alternative RNA splicing also plays a critical regulatory role\\nduring this process. Alternative splicing was proved to control the EM transition process,\\nand RNA-binding proteins were determined to regulate alternative splicing. A comprehensive understanding of alternative splicing and the RNA-binding proteins that regulate it during EM transition and their dynamic impact on breast cancer remains largely unknown. To\\naccurately study the dynamic regulatory relationships, time-series data of the EM transition\\nprocess are essential. However, only cross-sectional data of epithelial and mesenchymal\\nspecimens are available. Therefore, we developed a pseudotemporal causality-based\\nBayesian (PCB) approach to infer the dynamic regulatory relationships between alternative\\nsplicing events and RNA-binding proteins. Our study sheds light on facilitating the regulatory\\nnetwork-based approach to identify key RNA-binding proteins or target alternative splicing\\nevents for the diagnosis or treatment of cancers. The data and code for PCB are available\\nat: http://hkumath.hku.hk/*wkc/PCB(data+code).zip.\\n\\nAuthor summary\\nIn this paper, pseudotime causality-based Bayesian (PCB) model is proposed to detect the\\nregulatory relationships between alternative splicing events and RNA-binding proteins in\\nthe EM transition process of breast cancer. It is the first time to reveal the global role of\\nalternative splicing events and RNA-binding proteins during EM transition, in which\\nRNA-binding proteins can dynamically regulate alternative splicing. Furthermore, to\\naddress the challenges that lack of temporal information in sample-based transcriptomic\\ndata, we propose to decode the latent temporal information underlying cancer progression through ordering patient sample based on transcriptomic similarity, and design a\\n\\nPLOS Computational Biology | https://doi.org/10.1371/journal.pcbi.1010939 March 17, 2023\\n\\n1 / 24\\n\\n\\x0c',\n",
       " 'Breast Cancer Research and Treatment\\nhttps://doi.org/10.1007/s10549-022-06815-w\\n\\nEPIDEMIOLOGY\\n\\nPost-diagnostic statin use and breast cancer‑specific mortality:\\na population‑based cohort study\\nOliver William Scott1\\nJ. Mark Elwood1\\n\\n· Sandar TinTin1 · Sixten Harborg2 · Marion J. J. Kuper‑Hommel3 · Ross Lawrenson4 ·\\n\\nReceived: 5 September 2022 / Accepted: 9 November 2022\\n© The Author(s) 2023\\n\\nAbstract\\nPurpose Statins are the most widely prescribed cholesterol lowering medications and have been associated with both\\nimproved and unchanged breast cancer outcomes in previous studies. This study examines the association between the\\npost-diagnostic use of statins and breast cancer outcomes (death and recurrence) in a large, representative sample of New\\nZealand (NZ) women with breast cancer.\\nMethods Women diagnosed with a first primary breast cancer between 2007 and 2016 were identified from four populationbased regional NZ breast cancer registries and linked to national pharmaceutical data, hospital discharges, and death records.\\nCox proportional hazard models were used to estimate the hazard of breast cancer-specific death (BCD) associated with any\\npost-diagnostic statin use.\\nResults Of the 14,976 women included in analyses, 27% used a statin after diagnosis and the median follow up time was\\n4.51 years. Statin use (vs non-use) was associated with a statistically significant decreased risk of BCD (adjusted hazard\\nratio: 0.74; 0.63–0.86). The association was attenuated when considering a subgroup of ‘new’ statin users (HR: 0.91;\\n0.69–1.19), however other analyses revealed that the protective effect of statins was more pronounced in estrogen receptor\\npositive patients (HR: 0.77; 0.63–0.94), postmenopausal women (HR: 0.74; 0.63–0.88), and in women with advanced stage\\ndisease (HR: 0.65; 0.49–0.84).\\nConclusion In this study, statin use was associated with a statistically significant decreased risk of breast cancer death, with\\nsubgroup analyses revealing a more protective effect in ER+ patients, postmenopausal women, and in women with advanced\\nstage disease. Further research is warranted to determine if these associations are replicated in other clinical settings.\\nKeywords Breast cancer · Mortality · Statins · Pharmacoepidemiology · Cohort study\\n\\nIntroduction\\n\\n* Oliver William Scott\\no.scott@auckland.ac.nz\\n1\\n\\nDepartment of Epidemiology and Biostatistics, School\\nof Population Health, University of Auckland, Building 507,\\n85 Park Road, Grafton, Auckland 1023, New Zealand\\n\\n2\\n\\nDepartment of Oncology, Aarhus University Hospital/Aarhus\\nUniversity, Entrance C, Level 1, C118, Palle Juul‑Jensens\\nBoulevard 99, 8200 Aarhus N, Denmark\\n\\n3\\n\\nDepartment of Oncology, Waikato District Health Board,\\nHamilton, New Zealand\\n\\n4\\n\\nDepartment of NIDEA (National Institute of Demographic\\nand Economic Analysis), Waikato Medical Research Centre,\\nUniversity of Waikato, Hamilton, New Zealand\\n\\nBreast cancer is the most common cancer in women and\\nthe leading cause of female cancer mortality worldwide [1].\\nComorbidities are common in patients with breast cancer\\n[2], and there is a high and increasing prevalence of risk\\nfactors for both breast cancer and ischemic heart disease\\namong Western women [3–5]. As such, many patients with\\nbreast cancer use prescribed medications for cardiovascular\\nconditions. Examining the association between commonly\\nused cardiovascular medications and breast cancer outcomes\\nis therefore warranted. Statins (3-hydroxy-3-methylglutaryl\\ncoenzyme A reductase inhibitors) are the most widely prescribed cholesterol lowering medications [6] and are used for\\nboth the primary and secondary prevention of cardiovascular\\ndisease [7].\\n\\n13\\n\\nVol.:(0123456789)\\n\\n\\x0c',\n",
       " 'Meeting Abstracts\\n\\nSmoking cessation interventions for smokers diagnosed\\nwith cancer: a systematic review\\nKate Frazer, Nancy Bhardwaj, Patricia Fox, Vikram Niranjan, Diarmuid Stokes, Seamus Quinn, Cecily C Kelleher, Patricia Fitzpatrick\\n\\nAbstract\\n\\nBackground Smoking continuation or uptake after a diagnosis of cancer is associated with decreased treatment\\nefficacy, health deterioration, and worse survival outcomes; nonetheless, many patients with cancer continue smoking\\nafter their diagnosis. The extent and effectiveness of smoking cessation services in this cohort is poorly understood,\\nbut engagement with smoking cessation services is known to be scarce. The aim of this systematic review was to\\nidentify interventions for smokers diagnosed with cancer, as part of a larger study to develop a quit pathway.\\nMethods We searched PubMed, Embase, PsycInfo, and the Cumulative Index to Nursing and Allied Health Literature\\ndatabase for articles published between Jan 1, 2015, and Dec 15, 2020, in English, reporting an intervention for\\nsmokers aged 18 years or older with a diagnosis of lung, breast, cervical, or head and neck cancer (smoking-related\\ncancers), and with no restriction on study design. Outcomes included intervention descriptions and quit rates. Data\\nwere independently screened, extracted, and assessed for quality. This systematic review is registered with the\\nInternational Prospective Register of Systematic Reviews (number CRD42020214204).\\nFindings 4293 unique records were found, of which 23 studies (nine randomised controlled trials, ten observational\\nstudies, three quality improvement studies, and one qualitative study met the inclusion criteria. Studies included\\nbetween 11 and 2652 participants. 17 (74%) studies reported data from the USA, with the remainder 6 (26%) reporting\\ndata from Canada, Australia, Lebanon, and England. Ten (43%) studies reported data from head and neck cancer,\\nthree (13%) from lung cancer, one (4%) for breast cancer, and the remaining nine studies (39%) reported combined\\ndata for multiple cancers. Duration of smoking cessation interventions ranged between 1 month and 24 months. The\\nheterogeneity of the data prompted a narrative synthesis. Key smoking cessation interventions included long-term\\nuse of pharmacotherapy, high tech e-referral systems, national quitting programmes, quitting apps, enhanced\\ncounselling, and an opt-out referral system. Four randomised controlled trials and four observational studies reported\\nsustained quit attempts. Factors associated with quitting included timing and frequency of quitting conversations\\n(pre-operatively), extended use of varenicline (up to 24 weeks), and counselling. Patient challenges included pain,\\nfatigue, guilt, and lack of confidence.\\n\\nPublished Online\\nNovember 24, 2022\\nUniversity College Dublin,\\nDublin, Ireland (K Frazer PhD,\\nN Bhardwaj MD, P Fox PhD,\\nV Niranjan PhD, D Stokes MSc,\\nS Quinn, Prof C C Kelleher MD,\\nProf P Fitzpatrick MD);\\nSt Vincent’s University\\nHospital, Dublin, Ireland\\n(Prof C C Kelleher,\\nProf P Fitzpatrick)\\nCorrespondence to:\\nAssoc Prof Kate Frazer, University\\nCollege Dublin,\\nDublin D04 V1W8, Ireland\\nkathleen.frazer@ucd.ie\\n\\nInterpretation A multifaceted approach including personalised treatment plans is needed to promote smoking\\ncessation in patients with cancer.\\nFunding Irish Cancer Society (grant SCA19FIT).\\nCopyright © 2022 Published by Elsevier Ltd. All rights reserved.\\nContributors\\nKF, CCK, and PFi conceptualised the study. DS and KF developed the search strategy. DS searched for data. KF, NB, VN, PFo, and PFi screened,\\nextracted, assessed the quality, and analysed the data. KF drafted the manuscript. All authors critically reviewed and have seen and approved the final\\nversion of the abstract for publication.\\nDeclaration of interests\\nWe declare no competing interests.\\n\\nwww.thelancet.com\\n\\n39\\n\\n\\x0c',\n",
       " 'Meeting Abstracts\\n\\nUltra-processed food consumption, cancer risk, and cancer\\nmortality: a prospective cohort study of the UK Biobank\\nKiara Chang, Christopher Millett, Fernanda Rauber, Renata B Levy, Inge Huybrechts, Nathalie Kliemann, Marc J Gunter, Eszter P Vamos\\n\\nAbstract\\n\\nBackground Dietary patterns worldwide are increasingly displaced by many cheap, highly palatable, and ready-to-eat\\nultra-processed foods (UPFs). Higher UPF consumption has been linked to increased risk for obesity and\\ncardiometabolic diseases, but prospective evidence is limited on cancer outcomes. This study aimed to examine the\\nassociation between UPF consumption and risk for overall and site-specific cancer incidence and cancer mortality\\nusing the UK Biobank cohort.\\nMethods 197 426 participants of the UK Biobank from England, Scotland, and Wales with 24-h dietary recall completed\\nbetween 2009 and 2012 were included. Incident cancer cases were identified through data linkage to national cancer\\nand mortality registries. Food items consumed were categorised according to their degree of food processing using\\nthe NOVA classification system. Individual UPF consumption were derived as a percentage of daily food intake.\\nProspective association was assessed using multivariable Cox proportional hazards models adjusted for baseline\\nsociodemographic and lifestyle characteristics. For female-specific cancers, menopausal status, use of oral\\ncontraceptives, hormone replacement therapy, and parity were additionally adjusted.\\nFindings Mean age was 58·0 years (SD 8·0); 54·6% of participants were women. During a median follow-up time of\\n9·8 years (IQR 9·4–10·6), 15 921 (8∙1%) of 197 426 individuals developed cancer and 4009 (2∙0%) cancer deaths were\\nencountered. Consumption of UPFs was associated with a higher incidence of overall cancer (hazard ratio per 10%\\nincrement in UPF consumption was 1·02 [95% CI 1·01–1·04]; p=0∙005) and ovarian cancer in females\\n(1·19 [1·08–1·30]; p<0∙001). Positive associations were identified for mortality of overall, breast, and ovarian cancers.\\nInterpretation This large UK cohort study presents evidence of positive associations between UPF consumption and\\nrisks for incidence and mortality of overall and certain site-specific cancers. Limitations include non-representativeness\\nhaving based on a cohort of middle-aged volunteers, insufficient details for some food items although a conservative\\napproach was undertaken classifying foods to lower processing categories, and potential residual confounding due to\\nthe observational nature of the data. Further investigation into the mechanistic pathways is warranted to better\\nidentify targets for intervention.\\n\\nPublished Online\\nNovember 24, 2022\\nPublic Health Policy Evaluation\\nUnit, Imperial College London,\\nLondon, UK (K Chang PhD,\\nProf C Millett PhD,\\nE P Vamos PhD); Center for\\nEpidemiological Research in\\nNutrition and Health\\n(F Rauber PhD, R B Levy PhD)\\nand Department of Preventive\\nMedicine, School of Medicine\\n(F Rauber, R B Levy), University\\nof São Paulo, São Paulo, Brazil;\\nNutrition and Metabolism\\nBranch, International Agency\\nfor Research on Cancer, Lyon,\\nFrance (I Huybrechts PhD,\\nN Kliemann PhD,\\nM J Gunter PhD)\\nCorrespondence to:\\nDr Kiara Chang, Public Health\\nPolicy Evaluation Unit, Imperial\\nCollege London, London W6 8RP,\\nUK\\nchu-mei.chang@imperial.ac.uk\\n\\nFunding Cancer Research UK and World Cancer Research Fund.\\nCopyright © 2022 Published by Elsevier Ltd. All rights reserved.\\nContributors\\nKC, EPV, and CM conceptualised the study. KC did the statistical analyses and drafted the Abstract. All authors critically reviewed for intellectual\\ncontent and contributed to interpretations. All authors have seen and approved the final version of the Abstract for publication.\\nDeclaration of interests\\nWe declare no competing interests.\\nAcknowledgments\\nThis study was supported by Cancer Research UK (C33493/A29678) and funded by the World Cancer Research Fund (IIG_FULL_2020_033), as part\\nof the World Cancer Research Fund International grant programme. This research has been conducted using the UK Biobank Resource under\\nApplication Number 29239.\\n\\nwww.thelancet.com\\n\\n31\\n\\n\\x0c',\n",
       " 'Supportive Care in Cancer\\n(2023) 31:218\\nhttps://doi.org/10.1007/s00520-023-07663-0\\n\\nRESEARCH\\n\\nTraditional and new lifestyle interventions to prevent breast cancer\\nrecurrence (TANICA): a qualitative study\\nTanisha F. Aflague1 · Monica K. Esquivel2\\nJoseph Keawe‘aimoku Kaholokula3\\n\\n· Kristi Hammond1\\n\\n· Bernice Delos Reyes2\\n\\n·\\n\\nReceived: 15 October 2022 / Accepted: 27 February 2023\\nThis is a U.S. Government work and not under copyright protection in the US; foreign copyright protection may apply 2023\\n\\nAbstract\\nPurpose Breast cancer is the second cause of death from cancer in Guam and Hawai‘i and disproportionately impacts Native\\nHawaiian, CHamoru, and Filipino women. Although a few culturally informed interventions addressing breast cancer survivorship exist, none have been developed or tested for Native Hawaiian, CHamoru, and Filipino women. To address this,\\nthe TANICA study began with key informant interviews in 2021.\\nMethods Purposive sampling and grounded theory approaches were used to conduct semi-structured interviews with individuals experienced in providing healthcare or implementing community programs and/or research with ethnic groups of\\ninterest in Guam and Hawai‘i. A literature review and expert consultation identified intervention components, engagement\\nstrategies, and settings. Interview questions aimed to understand the relevance of evidence-based interventions and explored\\nsocio-cultural factors. Participants completed demographics and cultural affiliation surveys. Interviews were independently\\nanalyzed by trained researchers. Themes were mutually agreed upon by reviewers and key themes were identified based on\\nfrequencies.\\nResults Nineteen interviews were conducted in Hawai‘i (n=9) and Guam (n=10). Interviews confirmed the relevance of most\\nof the previously identified evidence-based intervention components for Native Hawaiian, CHamoru, and Filipino breast\\ncancer survivors. Ideas around culturally responsive intervention components and strategies emerged that were shared across\\nand unique to each ethnic group and site.\\nConclusion Evidence-based intervention components appear relevant, yet cultural and place-based strategies are needed\\nfor Native Hawaiian, CHamoru, and Filipino women in Guam and Hawai‘i. Future research should triangulate these findings with the lived experiences of Native Hawaiian, CHamoru, and Filipino breast cancer survivors to develop culturally\\ninformed interventions.\\nKeywords Breast cancer · Native Hawaiian · Pacific Islanders · Asians · Obesity\\n\\n1\\n\\nUniversity of Guam, College of Natural and Applied\\nSciences, Mangilao, Guam, USA\\n\\nMonica K. Esquivel\\nmonicake@hawaii.edu\\n\\n2\\n\\nUniversity of Hawai‘i, College of Tropical Agriculture\\nand Human Resources, Honolulu, Hawai‘i, USA\\n\\nKristi Hammond\\nkristimh@hawaii.edu\\n\\n3\\n\\nUniversity of Hawai‘i, John A Burns School of Medicine,\\nHonolulu, Hawai‘i, USA\\n\\n* Tanisha F. Aflague\\ntaflague@triton.uog.edu\\n\\nBernice Delos Reyes\\nbernice7@hawaii.edu\\nJoseph Keawe‘aimoku Kaholokula\\nkaholoku@hawaii.edu\\n\\n13\\n\\nVol.:(0123456789)\\n\\n\\x0c',\n",
       " '(2023) 14:46\\nZhang et al. Insights into Imaging\\nhttps://doi.org/10.1186/s13244-023-01387-9\\n\\nORIGINAL ARTICLE\\n\\nOpen Access\\n\\nCorrelation of androgen receptor\\nwith ultrasound, clinicopathological features\\nand clinical outcomes in breast cancer\\nXudong Zhang1†, Hao Cui1†, Nana Hu1, Peng Han1, Wei Fan1, Panting Wang1, Xiaoxuan Zuo1, Dantong Zhao1,\\nHe Huang2, Shuo Li2, Hanqing Kong1, Fuhui Peng1, Jiawei Tian1 and Lei Zhang1* \\u200a\\u200a\\n\\nAbstract\\nBackground This study aimed to explore whether there is an association between androgen receptor (AR) expression and ultrasound, clinicopathological features and prognosis of breast cancer.\\nMethods A total of 141 breast cancer patients were included in this retrospective study. AR expression was analyzed\\nby immunohistochemistry. The images of B-mode, color Doppler and strain elastography from 104 patients were collected continuously, and the corresponding ultrasound characteristics were obtained. The differences in ultrasound\\nand clinicopathological features in different AR status were analyzed. Progression-free survival (PFS) of patients was\\nobtained through up to 90 months of follow-up; then, the effect of AR on PFS was analyzed. Subsequently, a nomogram was constructed to predict the AR status. The predictive accuracy was calculated using C-index.\\nResults The positive expression of AR (AR +) was associated with lower histological grade (p = 0.034) and lower Ki-67\\nlevel (p = 0.029). Triple-negative breast cancer (TNBC) had the lowest probability of AR + (p < 0.001). The AR + group\\nmostly showed unsmooth margin (p < 0.001), posterior acoustic shadowing (p = 0.002) and higher elasticity score\\n(p = 0.022) on ultrasound. The echo pattern of most tumors with AR + was heterogeneous (p = 0.024) in Luminal A\\nsubtype. AR + could be a sign of a better prognosis in overall breast cancer (p < 0.001), as well as in human epidermal growth factor receptor 2 (HER2) overexpression and Luminal B subtypes (p = 0.001 and 0.025). The nomogram\\nshowed relatively reliable performance with a C-index of 0.799.\\nConclusion Our research demonstrated that AR expression was closely related to ultrasound, clinicopathological\\nfeatures and prognosis of breast cancer.\\nKey points\\n• In terms of clinicopathology, the probability of AR expression in different subtypes of breast cancer was various,\\nand the positive expression of AR was associated with lower histological grade and lower Ki-67 level.\\n\\n†\\n\\nXudong Zhang and Hao Cui contributed equally to this work and equal first\\nauthors.\\n*Correspondence:\\nLei Zhang\\nzhanglei6@hrbmu.edu.cn\\nFull list of author information is available at the end of the article\\n© The Author(s) 2023. Open Access This article is licensed under a Creative Commons Attribution 4.0 International License, which\\npermits use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the\\noriginal author(s) and the source, provide a link to the Creative Commons licence, and indicate if changes were made. The images or\\nother third party material in this article are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line\\nto the material. If material is not included in the article’s Creative Commons licence and your intended use is not permitted by statutory\\nregulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this\\nlicence, visit http://\\u200bcreat\\u200biveco\\u200bmmons.\\u200borg/\\u200blicen\\u200bses/\\u200bby/4.\\u200b0/.\\n\\n\\x0c',\n",
       " 'Virchows Archiv\\nhttps://doi.org/10.1007/s00428-023-03527-4\\n\\nORIGINAL ARTICLE\\n\\nHypoxia‑induced autophagy in triple negative breast cancer:\\nassociation with prognostic variables, patients’ survival and response\\nto neoadjuvant chemotherapy\\nDina M. El‑Guindy1 · Fatma MKh Ibrahim1 · Dina A. Ali2 · Hemat El‑Sayed El‑Horany3 · Nesreen M. Sabry4 ·\\nRasha A. Elkholy2 · Wael Mansour4 · Duaa S. Helal1\\nReceived: 14 December 2022 / Revised: 3 March 2023 / Accepted: 7 March 2023\\n© The Author(s) 2023\\n\\nAbstract\\nAutophagy is a cellular response to diverse stresses within tumor microenvironment (TME) such as hypoxia. It enhances\\ncell survival and triggers resistance to therapy. This study investigated the prognostic importance of HIF-1α and miR-210 in\\ntriple negative breast cancer (TNBC). Also, we studied the relation between beclin-1 and Bcl-2 and their prognostic relevance\\nin triple negative breast cancer. Furthermore, the involvement of hypoxia-related markers, beclin-1 and Bcl-2 in mediating\\nresistance to neoadjuvant chemotherapy (NACT) in TNBC was evaluated. Immunohistochemistry was performed to evaluate\\nHIF-1α, beclin-1 and Bcl-2 expression whereas, miR-210 mRNA was detected by quantitative reverse transcription PCR\\n(q-PCR) in 60 TNBC patients. High HIF-1α expression was related to larger tumors, grade III cases, positive lymphovascular invasion, advanced stage, high Ki-67 and poor overall survival (OS). High miR-210 and negative Bcl-2 expression were\\nrelated to nodal metastasis, advanced stage and poor OS. High beclin-1 was associated with grade III, nodal metastasis,\\nadvanced stage and poor OS. Also, high beclin-1 and negative Bcl-2 were significantly associated with high HIF-1α and\\nhigh miR-210. High HIF- 1α, miR-210 and beclin-1 as well as negative Bcl-2 were inversely related to pathologic complete\\nresponse following NACT. High beclin-1 and lack of Bcl-2 are significantly related to hypoxic TME in TNBC. High HIF-1α,\\nmiR-210, and beclin-1 expression together with lack of Bcl-2 are significantly associated with poor prognosis as well as poor\\nresponse to NACT. HIF-1α and miR-210 could accurately predict response to NACT in TNBC.\\nKeywords Hypoxia-inducible factor-1α (HIF-1α) · miR-210 · Beclin-1 · Bcl-2 · Triple-negative breast cancer (TNBC)\\n\\nIntroduction\\nTriple-negative breast cancer (TNBC) assigns to a subgroup\\nof breast cancer (BC) determined by the absence of estrogen\\nreceptor (ER), progesterone receptor (PR), and human epidermal growth factor receptor 2 (HER2). TNBC phenotype\\n* Dina M. El‑Guindy\\ndina.elgendy3@med.tanta.edu.eg\\n1\\n\\nPathology Department, Faculty of Medicine, Tanta\\nUniversity, Tanta, Egypt\\n\\n2\\n\\nClinical Pathology Department, Faculty of Medicine, Tanta\\nUniversity, Tanta, Egypt\\n\\n3\\n\\nMedical Biochemistry Department, Faculty of Medicine,\\nTanta University, Tanta, Egypt\\n\\n4\\n\\nClinical Oncology Department, Faculty of Medicine, Tanta\\nUniversity, Tanta, Egypt\\n\\nis characterized by high heterogeneity, aggressive clinical\\nbehaviour and lack of treatment modalities [1]. Chemotherapy is considered the approved option for TNBC treatment;\\nhowever, patients repeatedly acquire resistance. It is now\\nwidely accepted that the expression of tumour microenvironment (TME)-related factors could assist substantially in\\nchemoresistance [2].\\nTumour hypoxia, which is the lack of oxygen within a\\ntumour, is one of the most common characteristics of the\\nTME due to rapid cell growth and oxygen consumption.\\nHypoxia-inducible factor (HIF-1α), a nuclear transcription\\nfactor, is a marker of the hypoxic TME [3]. At present, it\\nis well established that hypoxia serves as an independent\\nunfavourable prognostic factor in several tumours and leads\\nto the ultimate failure of most anticancer therapeutics [4,\\n5]. MicroRNAs (miRNAs) are classified as small non-coding, single-stranded RNAs (ribonucleic acids) which exerts\\nessential roles in different fundamental processes through\\n\\n13\\n\\nVol.:(0123456789)\\n\\n\\x0c',\n",
       " 'European Journal of Applied Physiology\\nhttps://doi.org/10.1007/s00421-023-05177-5\\n\\nORIGINAL ARTICLE\\n\\nExacerbated central fatigue and reduced exercise capacity\\nin early‑stage breast cancer patients treated with chemotherapy\\nElyse Hucteau1,2,3 · Joris Mallard1,2,3 · Xavier Pivot3 · Roland Schott3 · Carole Pflumio3 · Philippe Trensz3 ·\\nFabrice Favret1,2 · Allan F. Pagano1,2 · Thomas J. Hureau1,2\\nReceived: 6 October 2022 / Accepted: 7 March 2023\\n© The Author(s) 2023\\n\\nAbstract\\nPurpose The present study aimed to characterize the etiology of exercise-induced neuromuscular fatigue and its consequences on the force-duration relationship to provide mechanistic insights into the reduced exercise capacity characterizing\\nearly-stage breast cancer patients.\\nMethods Fifteen early-stage breast cancer patients and fifteen healthy women performed 60 maximal voluntary isometric\\nquadriceps contractions (MVCs, 3 s of contraction, 2 s of relaxation). The critical force was determined as the mean force\\nof the last six contractions, while W’ was calculated as the force impulse generated above the critical force. Quadriceps\\nmuscle activation during exercise was estimated from vastus lateralis, vastus medialis and rectus femoris EMG. Central and\\nperipheral fatigue were quantified via changes in pre- to postexercise quadriceps voluntary activation (ΔVA) and quadriceps\\ntwitch force (ΔQTw) evoked by supramaximal electrical stimulation, respectively.\\nResults Early-stage breast cancer patients demonstrated lower MVC than controls preexercise (− 15%, P = 0.022), and this\\nreduction persisted throughout the 60-MVC exercise (− 21%, P = 0.002). The absolute critical force was lower in patients\\nthan in controls (144 ± 29N vs. 201 ± 47N, respectively, P < 0.001), while W’ was similar (P = 0.546), resulting in lower\\ntotal work done (− 23%, P = 0.001). This was associated with lower muscle activation in the vastus lateralis (P < 0.001),\\nvastus medialis (P = 0.003) and rectus femoris (P = 0.003) in patients. Immediately following exercise, ΔVA showed a\\ngreater reduction in patients compared to controls (− 21.6 ± 13.3% vs. − 12.6 ± 7.7%, P = 0.040), while ΔQTw was similar\\n(− 60.2 ± 13.2% vs. − 52.8 ± 19.4%, P = 0.196).\\nConclusion These findings support central fatigue as a primary cause of the reduction in exercise capacity characterizing\\nearly-stage breast cancer patients treated with chemotherapy.\\nClinical trials registration No. NCT04639609—November 20, 2020.\\nKeywords Neuromuscular fatigue · Central and peripheral fatigue · Neuromuscular function · Exercise tolerance · Critical\\nforce\\n\\nCommunicated by Nicolas Place.\\n* Thomas J. Hureau\\nt.hureau@unistra.fr\\n1\\n\\nBiomedicine Research Centre of Strasbourg (CRBS),\\nMitochondria, Oxidative Stress, and Muscular Protection\\nLaboratory (UR 3072), Strasbourg, France\\n\\n2\\n\\nFaculty of Sport Sciences, European Centre for Education,\\nResearch and Innovation in Exercise Physiology (CEERIPE),\\nUniversity of Strasbourg, 4 rue Blaise Pascal, CS 90032,\\n67081 Strasbourg Cedex, France\\n\\n3\\n\\nInstitute of Cancerology Strasbourg Europe (ICANS),\\nStrasbourg, France\\n\\nAbbreviations\\nANOVA\\t\\x07Analysis of variance\\nBF\\t\\x07Biceps femoris\\nCT\\t\\x07Contraction time\\nEMG\\t\\x07Electromyography\\nHRT\\t\\x07Half relaxation time\\nMEP\\t\\x07Motor evoked potentials\\nMmax\\t\\x07Maximal M-wave\\nRFD\\t\\x07Maximal rate of force development\\nMVC\\t\\x07Maximal voluntary contraction\\nQtw\\t\\x07Maximal single twitch peak force\\nRF\\t\\x07Rectus femoris\\nRMS\\t\\x07Root mean square\\n\\n13\\n\\nVol.:(0123456789)\\n\\n\\x0c',\n",
       " 'ERRATUM\\n\\nErratum to “Tumor Microenvironment-Specific\\nChemical Internalization for Enhanced Gene\\nTherapy of Metastatic Breast Cancer”\\nYun Zhou1†, Mian Yu1†, Changjun Tie2, Yang Deng1, Junqing Wang1,\\nYunfei Yi1, Fan Zhang1, Chenyi Huang1, Hairong Zheng2*, Lin Mei1,3*,\\nand Meiying Wu1*\\n1\\n\\nSchool of Pharmaceutical Sciences (Shenzhen), Sun Yat-sen University, Shenzhen 518107, China.\\nPaul C. Lauterbur Research Center for Biomedical Imaging, Institute of Biomedical and Health\\nEngineering, Shenzhen Institutes of Advanced Technology, Chinese Academy of Sciences, Shenzhen\\n518055, China. 3Tianjin Key Laboratory of Biomedical Materials, Key Laboratory of Biomaterials\\nand Nanotechnology for Cancer Immunotherapy, Institute of Biomedical Engineering, Chinese\\nAcademy of Medical Sciences and Peking Union Medical College, Tianjin 300192, China.\\n2\\n\\n*Address correspondence to: hr.zheng@siat.ac.cn (H.Z.); meilin7@mail.sysu.edu.cn (L.M.); wumy53@\\nmail.sysu.edu.cn (M.W.)\\n†These authors contributed equally to this work.\\n\\nIn the Research Article “Tumor Microenvironment-Specific\\nChemical Internalization for Enhanced Gene Therapy of\\nMetastatic Breast Cancer,” the authors made an error in Fig.\\n6A. In Fig. 6A, the image of “pre” in the “Free Cy5-siRNA”\\ngroup was misused. When incorporating the final images in\\nFig. 6A, another image of “24 h” in the “Free Cy5-siRNA”\\ngroup was inadvertently chosen as the “Pre” image, so they\\nappear to be remarkably similar. After the error was discovered, the authors repeated the experiments that concluded that\\nthis error did not affect the results, discussion, or conclusion\\n\\nZhou et al. 2023 | https://doi.org/10.34133/research.0076\\n\\nCitation: Zhou Y, Yu M, Tie C, Deng Y,\\nWang J, Yi Y, Zhang F, Huang C,\\nZheng H, Mei L, Wu M. Erratum\\nto “Tumor MicroenvironmentSpecific Chemical Internalization\\nfor Enhanced Gene Therapy of\\nMetastatic Breast Cancer”. Research\\n2023;6:Article 0076. https://doi.\\norg/10.34133/research.0076\\nSubmitted 24 January 2023\\nAccepted 24 Janu15-ary 2023\\nPublished 15 March 2023\\nCopyright © 2023 Yun Zhou et al.\\nExclusive Licensee Science and\\nTechnology Review Publishing House.\\nNo claim to original U.S. Government\\nWorks. Distributed under a Creative\\nCommons Attribution License\\n(CC BY 4.0).\\n\\nof this paper. Figure 6A has now been corrected in the PDF\\nand HTML (full text).\\n\\nReference\\nZhou Y, Yu M, Tie C, Deng Y, Wang J, Yi Y, Zhang F, Huang C,\\nZheng H, Mei L, et al. Tumor microenvironment-specific\\nchemical internalization for enhanced gene therapy of\\nmetastatic breast cancer. Research. 2021;2021:Article 9760398.\\n\\n1\\n\\n\\x0c',\n",
       " 'RESEARCH ARTICLE\\n\\nIntermittent F-actin Perturbations by Magnetic\\nFields Inhibit Breast Cancer Metastasis\\nXinmiao Ji1†, Xiaofei Tian2†, Shuang Feng1, Lei Zhang1, Junjun Wang1,\\nRuowen Guo1,3, Yiming Zhu1,3, Xin Yu1,3, Yongsen Zhang1, Haifeng Du1,\\nVitalii Zablotskii4, and Xin Zhang1,2,3,5*\\n1\\n\\nHigh Magnetic Field Laboratory of CAS (CHMFL), CAS Key Laboratory of High Magnetic Field and Ion Beam\\nPhysical Biology, HFIPS, Hefei, Anhui 230031, P.R China. 2Institutes of Physical Science and Information\\nTechnology, Anhui University, Hefei, Anhui, 230601, P. R. China. 3Science Island Branch of Graduate School,\\nUniversity of Science and Technology of China, Hefei, Anhui 230031, P.R China. 4Institute of Physics of the\\nCzech Academy of Sciences, Prague, Czech Republic. 5International Magnetobiology Frontier Research\\nCenter, Science Island, Hefei 230031, P.R. China.\\n*Address correspondence to: xinzhang@hmfl.ac.cn\\n†These authors contributed equally to this work.\\nF-actin (filamentous actin) has been shown to be sensitive to mechanical stimuli and play critical roles\\nin cell attachment, migration, and cancer metastasis, but there are very limited ways to perturb F-actin\\ndynamics with low cell toxicity. Magnetic field is a noninvasive and reversible physical tool that can easily\\npenetrate cells and human bodies. Here, we show that 0.1/0.4-T 4.2-Hz moderate-intensity low-frequency\\nrotating magnetic field-induced electric field could directly decrease F-actin formation in vitro and\\nin vivo, which results in decreased breast cancer cell migration, invasion, and attachment. Moreover, lowfrequency rotating magnetic fields generated significantly different effects on F-actin in breast cancer\\nvs. noncancerous cells, including F-actin number and their recovery after magnetic field retrieval. Using\\nan intermittent treatment modality, low-frequency rotating magnetic fields could significantly reduce\\nmouse breast cancer metastasis, prolong mouse survival by 31.5 to 46.0% (P < 0.0001), and improve their\\noverall physical condition. Therefore, our work demonstrates that low-frequency rotating magnetic fields\\nnot only can be used as a research tool to perturb F-actin but also can inhibit breast cancer metastasis\\nthrough F-actin modulation while having minimum effects on normal cells, which reveals their potential\\nto be developed as temporal-controlled, noninvasive, and high-penetration physical treatments for\\nmetastatic cancer.\\n\\nIntroduction\\nCytoskeleton is a fibrous network of protein filaments, including microtubules, actin, and intermediate filaments, which are\\nnot only responsible for cell morphology maintenance and\\norganelle position but also key components for multiple cellular\\nprocesses, including membrane trafficking, signal transduction,\\nand cell migration and division.\\nAs one of the major components of cytoskeleton, actin is\\nthe most abundant cellular protein in most types of eukaryote\\ncells. G-actin (globular actin) monomers in cells can polymerize\\ninto F-actin (filamentous actin), which is mainly found in the\\ncellular cortex, stress fibers, and pseudopodia. It has been\\nknown that in almost all steps of cancer metastatic spread, the\\nactin reorganization and reassembly are involved [1]. Since\\nmetastasis is the leading cause of cancer patient lethality, there are\\nnumerous efforts on developing treatment method to interfere\\nwith actin and its regulators in cancer research. For example,\\ninhibition of the actin-bundling protein fascin effectively inhibits\\nJi et al. 2023 | https://doi.org/10.34133/research.0080\\n\\nCitation: Ji X, Tian X, Feng S, Zhang L,\\nWang J, Guo R, Zhu Y, Yu X, Zhang Y,\\nDu H, Zablotskii V, Zhang X.\\nIntermittent F-actin Perturbations by\\nMagnetic Fields Inhibit Breast Cancer\\nMetastasis. Research 2023;6:Article\\n0080. https://doi.org/10.34133/\\nresearch.0080\\nSubmitted 24 October 2022\\nAccepted 31 January 2023\\nPublished 15 March 2023\\nCopyright © 2023 Xinmiao Ji et al.\\nExclusive Licensee Science and\\nTechnology Review Publishing House.\\nNo claim to original U.S. Government\\nWorks. Distributed under a Creative\\nCommons Attribution License\\n(CC BY 4.0).\\n\\ncancer metastasis in mice [2,3], indicating the effectiveness of\\ntargeting of actin regulation proteins for anticancer therapy.\\nIn the last few decades, physical methods have been rapidly\\ndeveloping strategies for alternative cancer treatment because\\nthey have multiple advantages, including low toxicity and high\\ntemporal control. For example, tumor-treating field utilizes\\nlow-intensity medium-frequency alternating electric fields to\\ninterfere with the mitotic spindles in dividing cancer cells and\\nhas been approved by Food and Drug Administration to be\\nused on glioblastoma [4–6]. Compared to electric field, magnetic field is less invasive and can provide better tissue penetration. It has been shown that even a weak magnetic field could\\naffect new tissue formation in vivo [7] and a combination of\\nstatic magnetic and electric field or a static magnetic field alone\\ncould both alleviate type 2 diabetes in mice [8,9]. Moreover,\\nmagnetic field has also been shown to be able to affect both\\npurified actin and cellular actin [10–21]. For example, the 4-Hz\\noscillating magnetic fields could interrupt F-actin network in\\nmesenchymal stem cells to inhibit their differentiation [21].\\n1\\n\\n\\x0c',\n",
       " 'International Journal of Public Health\\nORIGINAL ARTICLE\\npublished: 03 March 2023\\ndoi: 10.3389/ijph.2023.1605485\\n\\nCOVID-19 and Breast Cancer in Brazil\\nAline Ferreira Bandeira Melo Rocha 1*, Ruffo Freitas-Junior 2,\\nGlalber Luiz Rocha Ferreira 3, Danielle Cristina Netto Rodrigues 2 and\\nRosemar Macedo Sousa Rahal 2\\n1\\nPostgraduate Program in Health Sciences, Federal University of Goiás, Goiânia, Brazil, 2Advanced Center for Breast Diagnosis\\n(CORA), Federal University of Goiás, Goiânia, Brazil, 3Goiás State Education Department, Goiânia, Brazil\\n\\nObjectives: This study aimed to evaluate COVID-19 effects on breast cancer screening\\nand clinical stage at diagnosis in patients of 50–69 years of age receiving care within the\\npublic healthcare network (SUS) in 2013–2021 in Brazil and its macro-regions.\\nMethods: This ecological study used Poisson regression to analyze trends in screening\\nand staging. A secondary database was formed using SUS sources: outpatient data\\nsystem of the SUS network and Oncology—Brazil Panel.\\nResults: There was a reduction in screening, with an annual percent change of −5.9 (p <\\n0.022). The number of notiﬁed cases fell by 31.5% in 2020–2021 compared to\\n2018–2019. There was a 10.7% increase in the proportion of stage III/IV cases (p <\\n0.001) in 2020–2021 compared to 2013–2019, now surpassing the number of cases of\\nearly stage breast cancer.\\nConclusion: COVID-19 led to a reduction in breast cancer screening and an expressive\\nincrease in advanced tumors in users of the public healthcare network. Urgent\\ninterventions in public policies are required as the negative effects of the pandemic on\\nthe diagnosis/treatment of breast cancer are becoming apparent even earlier than\\nexpected.\\nKeywords: breast cancer, COVID-19, Brazil, breast cancer screening, clinical staging of breast cancer\\n\\nINTRODUCTION\\nEdited by:\\nAylene Bousquat,\\nUniversity of São Paulo, Brazil\\n*Correspondence:\\nAline Ferreira Bandeira Melo Rocha\\nalinebandeirageriatra@gmail.com\\nReceived: 13 October 2022\\nAccepted: 24 February 2023\\nPublished: 03 March 2023\\nCitation:\\nRocha AFBM, Freitas-Junior R,\\nFerreira GLR, Rodrigues DCN and\\nRahal RMS (2023) COVID-19 and\\nBreast Cancer in Brazil.\\nInt J Public Health 68:1605485.\\ndoi: 10.3389/ijph.2023.1605485\\n\\nThe COVID-19 pandemic impacted healthcare services worldwide due to the need to prioritize\\nresources for urgencies and reduce dissemination of the virus at healthcare services (1).\\nConsequently, there were negative repercussions on elective procedures, including cancer\\nscreening (2–4), which was paused in Brazil and in the vast majority of countries (2, 5, 6).\\nAccording to data published by the World Health Organization, over half the countries\\ninvestigated interrupted treatment services for hypertension either partially or totally, while 49%\\ninterrupted the treatment of diabetes and its complications; 42% interrupted cancer treatment; and\\n31% suspended cardiovascular emergencies. Furthermore, 50% of these countries reported a\\nreduction in public screening programs (7). National and international medical societies\\npublished recommendations on cancer care, taking individual risks and beneﬁts to patients into\\nconsideration (4, 8–11).\\nAs the pandemic progressed and vaccination programs were implemented, the restrictive\\nmeasures hitherto imposed were eased somewhat, allowing healthcare managers and\\nprofessionals to evaluate whether to begin offering elective procedures, particularly the\\nscreening, diagnosis and early treatment of malignant neoplasms (3, 12). Deciding on these\\n\\nInt J Public Health | Owned by SSPH+ | Published by Frontiers\\n\\n1\\n\\nMarch 2023 | Volume 68 | Article 1605485\\n\\n\\x0c',\n",
       " 'Open Access Original\\nArticle\\n\\nDOI: 10.7759/cureus.34925\\n\\nClinicopathological Patterns of Malignant Solid\\nTumors in Adult Patients: A Hospital-Based Study\\nFrom Bangladesh\\n\\nReview began 01/31/2023\\nReview ended 02/08/2023\\n\\nAyesha Siddika 1 , Sawrab Chowdhury 2 , Md Rockyb Hasan 3 , Md Moniruzzaman 4 , S.K. Jakaria Been Sayeed\\n5 , Tahsin Tabassum 6 , Mahidul Chowduary 7 , Tanzin Tabassum 8 , Azizul Islam 9 , Md. Mujibur Rahman 10\\n\\nPublished 02/13/2023\\n© Copyright 2023\\nSiddika et al. This is an open access article\\ndistributed under the terms of the Creative\\nCommons Attribution License CC-BY 4.0.,\\nwhich permits unrestricted use, distribution,\\nand reproduction in any medium, provided\\nthe original author and source are credited.\\n\\n1. Cardiology, National Heart Foundation Hospital & Research Institute, Dhaka, BGD 2. Computer Science and\\nElectrical Engineering, Chandpur Polytechnic Institute, Chandpur, BGD 3. Internal Medicine, Texas Tech University\\nHealth Sciences Center, Amarillo, USA 4. Clinical Neurology, National Institute of Neurosciences & Hospital, Dhaka,\\nBGD 5. Medicine and Rheumatology, National Institute of Neurosciences & Hospital, Dhaka, BGD 6. School of\\nCommunity Health and Policy, Morgan State University, Baltimore, USA 7. Internal Medicine, Interfaith Medical\\nCenter, Brooklyn, USA 8. General Surgery, Lister Hospital, Stevenage, GBR 9. Internal Medicine, Bangladesh Armed\\nForces, Directorate General of Medical Service (DGMS), Dhaka, BGD 10. Medicine, Bangabandhu Sheikh Mujib Medical\\nUniversity, Dhaka, BGD\\nCorresponding author: Md Rockyb Hasan, drmdrockybhasan@gmail.com\\n\\nAbstract\\nBackground: Solid malignant tumors are abnormal masses of tissue that usually do not contain any cysts or\\nliquid areas. The causation of these tumors is multifactorial, and the disease process differs at different\\nsites.\\nAim: This study aims to determine the clinicopathological patterns of malignant solid tumors in adult\\npatients admitted into the department of internal medicine of a tertiary care hospital in Bangladesh.\\nMethod: This cross-sectional study was conducted between January 2018 and June 2018 at the Department\\nof Medicine, Dhaka Medical College Hospital, Bangladesh. We recorded the complete socio-demographic\\ncharacteristics, clinical patterns, and pathological characteristics of malignant solid tumors in adult\\npatients.\\nResults: A total of 100 patients with confirmed malignant solid tumors were included in our study. The mean\\nage of the patients was 47.5 years (SD: ±4.20); most of them (27%) were between 58 and 67 years of age.\\nMale patients constitute 59% of the total study sample. Most of the patients were of the middle socioeconomic class (59%) and most of them (61%) were nonsmokers. Among the patient diagnoses, 19% had lung\\ncancer, 29.27% had breast cancer, and in 14%, lymphoma (tumor arising from the lymphatic system) was the\\ncommon solid tumor. Lung cancer was found to be the most common form of cancer in males. Additionally,\\nthe majority of those diagnosed with lung cancer smoked. Breast cancer was found to be the most common\\ntype of cancer in females.\\nConclusion: This study reflects that solid malignant tumors can affect any socio-economic class of people.\\nBased on findings in our study as well as others, prevention efforts should focus on the reduction in tobacco\\nuse and the incorporation of other lifestyle changes, such as diet modification and exercise. Additionally, the\\nincorporation of economic factors and how they affect cancer presentation in different contexts is crucial.\\n\\nCategories: Internal Medicine, Oncology\\nKeywords: solid tumor, risk factors, pathological pattern, clinical features, socio-economic groups, malignant solid\\ntumor\\n\\nIntroduction\\nCancers originate from cells almost anywhere in the human body and can grow uncontrollably. If the spread\\nof cancers is not controlled, it can often result in death. Although the etiology of different types of cancers\\nremains unknown, multiple factors are responsible for the abnormality in cellular development. A solid\\nmalignant tumor is an abnormal mass of tissue that usually does not contain cysts or liquid areas. These\\nsolid tumors vary according to their site of origin, for instance, sarcomas (tumors arising from connective\\ntissue), carcinomas (tumors arising from epithelial tissue), and lymphomas (tumors arising from the\\nlymphatic system). However, the term solid tumor is used to distinguish between a localized mass of tissue\\nand leukemia [1]. The biology of cancer is a complex interplay of many underlying processes, taking place at\\ndifferent scales both in space and time.\\nCancer remains a leading cause of death across the globe and approximately 10 million deaths have been\\nreported due to cancer or related complications in 2020 [2,3]. The most common cancers in both the\\ndeveloped and developing world are breast, lung, colon, rectum, and prostate cancers [2,3]. The prevalence\\n\\nHow to cite this article\\nSiddika A, Chowdhury S, Hasan M, et al. (February 13, 2023) Clinicopathological Patterns of Malignant Solid Tumors in Adult Patients: A HospitalBased Study From Bangladesh. Cureus 15(2): e34925. DOI 10.7759/cureus.34925\\n\\n\\x0c',\n",
       " 'Open Access Case\\nReport\\n\\nDOI: 10.7759/cureus.34950\\n\\nMultidisciplinary Antenatal Management of a\\nLate Pregnancy Complicated With Advanced\\nStage Breast Burkitt Lymphoma - Case Report and\\na Review of the Literature\\n\\nReview began 01/15/2023\\nReview ended 01/27/2023\\nPublished 02/13/2023\\n© Copyright 2023\\nHua et al. This is an open access article\\ndistributed under the terms of the Creative\\nCommons Attribution License CC-BY 4.0.,\\nwhich permits unrestricted use, distribution,\\nand reproduction in any medium, provided\\nthe original author and source are credited.\\n\\nZhong Hua 1 , Iqra Ijaz 2, 3 , Muhammad N. Shahzad 4, 5 , Duan Yi 6 , Gao Y. Hu 7 , Fu X. Dong 1\\n1. Department of Obstetrics, Luzhou Medical College Affiliated Hospital, Luzhou, CHN 2. Sichuan Provincial Center for\\nGynecological and Breast Diseases, Southwest Medical University, Luzhou, CHN 3. Department of Obstetrics and\\nGynecology, Holy Family Hospital, Rawalpindi, PAK 4. Department of Hematology, Luzhou Medical College Affiliated\\nHospital, Luzhou, CHN 5. Internal Medicine, Holy Family Hospital, Rawalpindi, PAK 6. Department of Pathology,\\nLuzhou Medical College Affiliated Hospital, Luzhou, CHN 7. Department of Radiology, Luzhou Medical College\\nAffiliated Hospital, Luzhou, CHN\\nCorresponding author: Fu X. Dong, sirinihao@163.com\\n\\nAbstract\\nThe multidisciplinary team (MDT) plays a pivotal role in establishing the diagnosis and tailoring treatment\\nfor challenging, complicated, rare obstetrical cases. At 28 weeks of gestation, a lady presented with an\\nunresolved unilateral proptosis and sustained severe mastitis. MDT managed the patient at a tertiary care\\nhospital for primary breast Burkitt lymphoma (PBBL). It is a rare and highly malignant condition requiring\\nan aggressive therapeutic approach. Antenatal chemotherapy (ANC) with an aggressive regimen of R-hyperCVAD/MA was started. A healthy baby was vaginally delivered after completing the second therapy cycle at\\n32+ weeks, weighing 1.6kg with a good Apgar score. Postnatally, the central nervous system (CNS)\\nprophylaxis was added; after completing eight chemo cycles, our patient remained stabilized for nine\\nmonths. Unfortunately, due to the refractory and aggressive nature of malignancy, it relapsed, giving an\\noverall survival (OS) of two years. MDT care should be considered at the earliest possible period to expedite\\nthe entire process. Positive results can be achieved with timely aggressive treatment and early management\\nof such cases.\\n\\nCategories: Obstetrics/Gynecology, Oncology, Hematology\\nKeywords: hematologic cancer, multidisciplinary analysis, antenatal chemotherapy, pregnancy, burkitt lymphoma\\n\\nIntroduction\\nLymphoma in pregnancy is rare, with an incidence of one in 6000 births [1]. Primary breast Burkitt\\nlymphoma (PBBL) is a sporadic, highly aggressive growing tumor in pregnancy with an incidence rate of\\n0.04%-0.5%. The typical age of onset is 55 years, while onset before the age of 35 is extremely rare. If treated\\npromptly, stages I and II give a favorable prognosis of up to 90%, but in a more advanced stage, when the\\ncentral nervous system is affected, survival drops to as low as 0-30% [2]. In the literature, there are reports of\\n21 cases of PBBL linked to pregnancy; 16 of these were diagnosed in pregnancy [3-15], while five were during\\nthe breastfeeding period [4,16,17], with an average age of 29.9 years (range 15-42). Eleven of the 16 cases\\nreported during pregnancy occurred in the third trimester, compared to four and one each in the second and\\nfirst trimesters. The recorded overall survival (OS) was one month in nine, six months in four, and 10\\nmonths for one patient, while six patients showed no signs of disease at 12+ months following prompt\\naggressive treatment.\\nBeing pregnant while having lymphoma can be challenging, in part because the disease is thought to be\\nincurable and in part because the patient worries that antenatal chemotherapy (ANC) will harm her unborn\\nchild. We present the first case of PBBL from China that was found in the third trimester. Our patient\\nknowingly consented to receive ANC with regimen R-hyper-CVAD/MA (rituximab, cyclophosphamide,\\nvincristine, doxorubicin, dexamethasone, methotrexate, and cytarabine). Systemic and/or intrathecal\\nchemotherapy with methotrexate and cytarabine is used for central nervous system (CNS) prophylaxis [18].\\nApart from the increased risk of preterm delivery, this regimen is considered safer in the second and third\\ntrimesters [19]. For newly diagnosed BL, dose-intensive multi-agent chemotherapy with CNS prophylaxis is\\npreferable, while the R-CHOP regimen is inadequate. Clinical trials are the preferred treatment for relapsing\\nor resistant conditions [20,21].\\nThus, the current recommendation for the treatment of Burkitt lymphoma (BL) follows the period of\\npregnancy in terms of the trimester, the disease nature, whether or not aggressive, and notably, the patient\\npreference [22].\\n\\nHow to cite this article\\nHua Z, Ijaz I, Shahzad M N, et al. (February 13, 2023) Multidisciplinary Antenatal Management of a Late Pregnancy Complicated With Advanced\\nStage Breast Burkitt Lymphoma - Case Report and a Review of the Literature. Cureus 15(2): e34950. DOI 10.7759/cureus.34950\\n\\n\\x0c',\n",
       " 'HHS Public Access\\nAuthor manuscript\\nAuthor Manuscript\\n\\nClin Thyroidol. Author manuscript; available in PMC 2023 March 17.\\nPublished in final edited form as:\\nClin Thyroidol. 2022 July ; 34(7): 286–289. doi:10.1089/ct.2022;34.286-289.\\n\\nThyroid Hormones Enhance the Growth of Estrogen Receptor–\\nPositive Breast Cancers\\nJoanna Klubo-Gwiezdzinska\\nMetabolic Disease Branch, National Institutes of Diabetes, Digestive and Kidney Diseases,\\nNational Institutes of Health, Bethesda, MD, U.S.A.\\n\\nAuthor Manuscript\\n\\nSUMMARY\\nBackground—Thyroid disorders have been associated with breast cancer. In fact, breast\\ncancer is the most common secondary malignancy in female patients with thyroid cancer (1).\\nMoreover, hyperthyroidism is associated with an 11% increased risk of breast cancer in women\\n(2). Importantly, up to 30% of patients with breast cancer are treated with thyroid hormone\\nreplacement therapy (THRT) for overt or subclinical hypothyroidism (3). These observations,\\ncoupled with the preclinical data showing growth stimulatory effects of thyroid hormones (THs) in\\nvarious cancer models (4), formed the rationale for the study by Wahdan-Alaswad et al. aimed at\\ninvestigating the role of THRT on the outcome of patients with nonmetastatic breast cancer (3).\\n\\nAuthor Manuscript\\n\\nMethods—The authors conducted an observational study analyzing the association between\\nTHRT, disease-free survival (DFS), and disease-specific survival (DSS) in two cohorts of patients\\nwith nonmetastatic breast cancer. The first cohort consistent of 820 patients followed for a median\\nof 10 years and treated for breast cancer between 1962 and1993, with THRT implemented in\\n69 patients. The second cohort included 160 patients treated more recently (between 2006 and\\n2009) and followed for a median of 8.8 years, with 50 patients exposed to THRT. The data\\non the age, tumor size, presence or absence of steroid (estrogen and/or progesterone) receptors\\n(SR+/SR−), and treatment regimen were incorporated in the multivariate model analyzing the\\nassociation between DFS/DSS and THRT at 5 and 10 years. To better understand the results of the\\nobservational cohort study, the authors performed functional in vitro and in vivo experiments to\\ninvestigate the molecular mechanisms underlying TH effects on breast cancer cells and to test the\\ninteractions between estrogen receptors (ERs) and TH receptors (THRs).\\n\\nAuthor Manuscript\\n\\nResults—In patients with SR+ breast cancer, THRT was associated with a significantly increased\\nrisk of recurrence (DFS RR, 2.9; P<0.001) and death (DSS RR, 3.4; P<0.001), independent of age,\\ntumor size and grade, while THRT in patients with SR− breast cancer was not associated with\\nworse outcomes. Moreover, patients with SR+ breast cancer undergoing therapy with aromatase\\ninhibitor combined with THRT were characterized by a shorter DFS (P<0.042) and a higher\\n10-year recurrence rate of 14%, as compared with 2% in patient treated with the aromatase\\ninhibitor alone.\\nThe functional in vitro and in vivo studies revealed growth stimulatory effects of monotherapy\\nwith TH or estrogens that were further potentiated with combination therapy in ER+ breast\\n\\nDisclosures: The author has no relevant conflicts of interest to declare.\\n\\n\\x0c',\n",
       " 'TYPE\\n\\nReview\\n02 March 2023\\n10.3389/fphar.2023.1142087\\n\\nPUBLISHED\\nDOI\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nVuanghao Lim,\\nUniversiti Sains Malaysia (USM), Malaysia\\nREVIEWED BY\\n\\nBalram Chowbay,\\nNational Cancer Centre Singapore,\\nSingapore\\nSiti Aisyah Abd Ghafar,\\nIslamic Science University of Malaysia,\\nMalaysia\\nNurul Shahﬁza Noor,\\nUniversiti Sains Malaysia (USM), Malaysia\\n\\nThe role of irreversible pan-HER\\ntyrosine kinase inhibitors in the\\ntreatment of HER2-Positive\\nmetastatic breast cancer\\nZihong Wu \\ue840 , Jiamei Wang, Fengming You, Xueke Li* and\\nChong Xiao*\\nHospital of Chengdu University of Traditional Chinese Medicine, Chengdu, China\\n\\n*CORRESPONDENCE\\n\\nXueke Li,\\n2017202040046@whu.edu.cn\\nChong Xiao,\\nxiaochong@stu.cdutcm.edu.cn\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nPharmacology of Anti-Cancer Drugs,\\na section of the journal\\nFrontiers in Pharmacology\\nRECEIVED\\n\\n11 January 2023\\n14 February 2023\\nPUBLISHED 02 March 2023\\nACCEPTED\\n\\nCITATION\\n\\nWu Z, Wang J, You F, Li X and Xiao C\\n(2023), The role of irreversible pan-HER\\ntyrosine kinase inhibitors in the treatment\\nof HER2-Positive metastatic\\nbreast cancer.\\nFront. Pharmacol. 14:1142087.\\ndoi: 10.3389/fphar.2023.1142087\\nCOPYRIGHT\\n\\n© 2023 Wu, Wang, You, Li and Xiao. This is\\nan open-access article distributed under\\nthe terms of the Creative Commons\\nAttribution License (CC BY). The use,\\ndistribution or reproduction in other\\nforums is permitted, provided the original\\nauthor(s) and the copyright owner(s) are\\ncredited and that the original publication\\nin this journal is cited, in accordance with\\naccepted academic practice. No use,\\ndistribution or reproduction is permitted\\nwhich does not comply with these terms.\\n\\nHuman epidermal growth factor receptor 2 (HER2)-positive metastatic breast\\ncancer (MBC) is the leading cause of cancer death in women. For patients with\\nHER2-positive MBC, after the failure of multiple lines of treatment, there is no\\noptimal line of therapy. A series of clinical trials conﬁrmed that treatment with\\nirreversible pan-HER tyrosine kinase inhibitors (TKIs) in combination with\\nchemotherapy signiﬁcantly improves patients’ survival outcomes. This review\\nfocuses on the pathogenesis of HER2-positive breast cancer, current standard\\ntreatments, mechanisms of approved irreversible TKIs, and key clinical trials. The\\navailable ﬁndings suggest that irreversible pan-HER TKIs, such as pyrotinib and\\nneratinib, in combination with chemotherapy, represent a beneﬁcial salvage\\ntherapy for patients with HER2-positive MBC with manageable toxicity.\\nHowever, further studies are needed to assess the efﬁcacy and safety of this\\ncombination therapy.\\nKEYWORDS\\n\\nHER2-positive metastatic breast cancer, irreversible pan-HER tyrosine kinase inhibitors,\\nfailure of multiple lines of treatment, clinical trials, salvage therapy\\n\\nAbbreviations: ADCs, antibody-drug conjugates; AEs, adverse events; BC, breast cancer; BBB, bloodbrain barrier; BTB, blood tumor barrier; CNS, central nervous system; EGFR, epidermal growth factor\\nreceptor; ERK, extracellular signal-regulated kinase; EMT, epithelial-mesenchymal transition; FDA, Food\\nand Drug Administration; HER2, human epidermal growth factor receptor 2; HR, hazard ratio; JAK, Janus\\nkinase; MAPK, mitogen-activated protein kinase; MBC, metastatic breast cancer; mOS, median overall\\nsurvival; mPFS, median progression-free survival; NSCLC, non-small-cell lung cancerl; NR, no report;\\nORR, objective response rate; PI3K, phosphoinositide 3-kinase; RWS, real-world study; RCT, randomized\\ncontrolled trial; SAEs, serious adverse events; STAT, signal transducer and activator of transcription; TKIs,\\ntyrosine kinase inhibitors; T-DM1, trastuzumab emtansine; T-Dxd, trastuzumab deruxtecan; US,\\nUnited States; WBRT, whole-brain radiotherapy.\\n\\nFrontiers in Pharmacology\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Case Report\\n\\nKEYNOTE—522 and male spindle cell\\ncarcinoma of the breast: A case report\\n\\nRare Tumors\\nVolume 15: 1–4\\n© The Author(s) 2023\\nArticle reuse guidelines:\\nsagepub.com/journals-permissions\\nDOI: 10.1177/20363613231163730\\njournals.sagepub.com/home/rtu\\n\\nHarper E Niver \\ue840, Edward Foxhall and Anup Lahiry\\n\\nAbstract\\nMetaplastic Breast Cancer (MBC) is a rare group of tumors often presenting as triple-negative. MBC accounts for less than\\n1% of all breast cancers with the spindle cell variant comprising less than 0.5%. While rare, spindle cell carcinoma is the\\ncommonest subtype in the western world. It has a more aggressive biological behavior with increased risk of recurrence\\nand death due to disease compared to triple negative breast cancers. There is no treatment guideline for management of\\nMBC due to the rarity of the disease. Instead, treatment is theorized based off success with other types of aggressive breast\\nand metaplastic cancers of different tissue. We present the ﬁrst known case report of male spindle cell carcinoma of the\\nbreast treated with KEYNOTE-522 regimen. Therapy included a ﬁrst phase with pembrolizumab (dose of 200 mg) every\\n3 weeks plus paclitaxel and carboplatin and second phase, with four cycles of pembrolizumab with doxorubicincyclophosphamide.\\n\\nKeywords\\nMale breast cancer, Spindle Cell breast cancer, KEYNOTE 522, Metaplastic Breast Cancer\\n\\nIntroduction\\n\\nCase Presentation\\n\\nMetaplastic Breast Cancer (MBC) is a rare group of tumors\\nwith an array of morphologies that is composed of mesenchymal cells or non-glandular epithelium accounting for\\nless than 1% of all breast cancers.1,2 The vast majority are\\ntriple negative but have shown to contain various\\nimmunoproﬁles.3,4 MBC spreads hematogenously with low\\nincidence of lymph node involvement and typically presents\\nwith a higher tumor stage with increased risk of recurrence\\nand death due to disease compared to triple negative breast\\ncancer and invasive ductal carcinoma.1-3,5 According to the\\n2019 World Health Organization (WHO), six subtypes of\\nMBCs are recognized: low-grade adenosquamous carcinoma, ﬁbromatosis-like metaplastic carcinoma, squamous\\ncell carcinoma, spindle cell carcinoma, metaplastic carcinoma with heterologous mesenchymal differentiation, and\\nmixed metaplastic carcinoma.6 The spindle cell variant\\ncomprises less than 0.5% of all breast cancers but is the most\\ncommon MBC subtype in the western world (34%) and is\\nbiologically more aggressive.7,8\\n\\nA 72-year-old male with history of hypertension and hyperlipidemia presented to his primary care physician with\\nright breast tenderness and swelling after accidentally hitting it against a wall. The patient experienced rapid breast\\nswelling that progressed over ﬁve weeks after injury and\\nnew purulent nipple discharge, which prompted him to seek\\nmedical attention (Figure 1).\\nMammogram (Figure 2) showed a large central mass\\nwith a group of suspicious microcalciﬁcations within the\\nmass. There was evidence of breast parenchymal edema and\\nskin thickening without nipple retraction. Ultrasound\\nshowed a large mass with mixed echogenicity measuring\\n\\nNortheast Georgia Medical Center and Health System, Gainesville, GA,\\nUnited States\\nCorresponding author:\\nHarper E Niver, Northeast Georgia Medical Center and Health System,\\nGainesville, GA, United States\\n\\nCreative Commons Non Commercial CC BY-NC: This article is distributed under the terms of the Creative Commons\\nAttribution-NonCommercial 4.0 License (https://creativecommons.org/licenses/by-nc/4.0/) which permits non-commercial use,\\nreproduction and distribution of the work without further permission provided the original work is attributed as speciﬁed on the\\nSAGE and Open Access pages (https://us.sagepub.com/en-us/nam/open-access-at-sage).\\n\\n\\x0c',\n",
       " 'Correlation Between Intensity of Vimentin Immuno-expression in Young Women with Triple Negative Breast Cancer\\n\\nORIGINAL PAPER\\ndoi: 10.5455/medarh.2022.76.454-457\\nMED ARCH. 2022 DEC; 76(6): 454-457\\nRECEIVED:\\n\\nNOV 12, 2022\\n\\nACCEPTED:\\n\\nDEC 15, 2022\\n\\nPhilosophy Doctor in Medicine Program,\\nFaculty of Medicine, Universitas Sumatera\\nUtara, Medan, Indonesia\\n1\\n\\n2\\nDepartment of Anatomy, Faculty of\\nMedicine, Universitas Sumatera Utara,\\nMedan, Indonesia\\n3\\nDepartment of Neurosurgery, Faculty of\\nMedicine, Universitas Sumatera Utara,\\nMedan, Indonesia\\n\\nDepartment of Surgery, Faculty of\\nMedicine, Universitas Sumatera Utara,\\nMedan, Indonesia\\n4\\n\\nDepartment of Anatomical Pathology,\\nFaculty of Medicine, Universitas Sumatera\\nUtara, Medan, Indonesia\\n5\\n\\nDepartment of Dermatology and\\nVenereology, Faculty of Medicine,\\nUniversitas Sumatera Utara, Medan,\\nIndonesia\\n6\\n\\n7\\nDepartment of Ear, Nose and Throat,\\nHead & Neck, Faculty of Medicine,\\nUniversitas Sumatera Utara, Medan,\\nIndonesia\\n\\nCorresponding author: dr. Fitriani\\nLumongga, M.Ked(PA), SpPA, Department\\nof Anatomy, Faculty of Medicine,\\nUniversitas Sumatera Utara. Jl.dr.T.Mansur\\nNo.5 Kampus USU Medan, Indonesia.\\nPhone: +628126454891. Email: fitriani@\\nusu.ac.id , ORCHID ID: https://orcid.\\norg/0000-0002-1913-1399.\\n\\n© 2022 Fitriani Lumongga, Ridha\\nDharmajaya, Kamal Basri Siregar, Delyuzar,\\nDiah Rini Handjari, Nelva Karmila Jusuf,\\nDelfitri Munir, Asrul\\nThis is an Open Access article distributed under\\nthe terms of the Creative Commons Attribution\\nNon-Commercial License (http://creativecommons.org/licenses/by-nc/4.0/) which\\npermits unrestricted non-commercial use,\\ndistribution, and reproduction in any medium,\\nprovided the original work is properly cited.\\n\\n454\\n\\nCorrelation Between Intensity\\nof Vimentin Immuno-expression\\nin Young Women with Triple\\nNegative Breast Cancer and Its\\nCliniocopathological Parameters\\nFitriani Lumongga1,2, Ridha Dharmajaya3, Kamal Basri Siregar4,\\nDelyuzar5, Diah Rini Handjari5, Nelva Karmila Jusuf6, Delfitri Munir7,\\nAsrul4\\nABSTRACT\\nBackground: Young breast cancer patients (≤40 years) have different unifying oncogenic\\nsignaling pathways when compared to older people. Vimentin is a filament intermediate.\\nVimentin expression has an important role in epithelial-to-mesenchymal transition (EMT)\\nand cancer progression. Objective: The aim of this study was to assess the correlation\\nbetween vimentin intensity and clinicopathological parameters in patients with triple negative breast cancer aged 40 years. Methods: Samples were taken from 45 paraffin blocks\\nof patients with young woman with triple negative invasive breast cancer NOS type that\\nmet the inclusion criteria, immunohistochemical examination was performed with vimentin. Vimentin intensity was assessed using ImageJ analysis. Results: Samples were taken\\nfrom paraffin blocks of patients with triple negative invasive breast cancer NOS type that\\nmet the inclusion criteria, immunohistochemical examination was performed with vimentin. Vimentin intensity was assessed using ImageJ analysis, Conclusion: This study shows\\nthat vimentin intensity is strongly associated with tumor progression, tumor mass size and\\ntumor invasion so that it can be used as a prognostic factor in young triple negative breast\\ncancer patients.\\nKeywords: triple negative breast cancer, young age, vimentin intensity.\\n\\n1. BACKGROUND\\n\\nCarcinoma of the breast is the most common type of cancer and ranks\\nfirst in the type of malignancy that often causes death in women worldwide\\n(1). According to some experts, breast cancer sufferers at a young age are\\npatients who are under 40 years old at the time of diagnosis. The incidence\\nof breast cancer at a young age varies in several countries, usually around 7%\\nof all women with breast cancer (2). Breast cancer in young patients is often\\nassociated with a poor prognosis and failure to respond to therapy, because at\\nthe time of diagnosis, hormonal receptors are often negative or Her2 negative\\n(Triple negative), high histological grade (grade 3) and cancer cell involvement in lymphnodes (3).\\nEpithelial Mesenchymal Transition (EMT) can occur in physiological and\\npathological conditions and is a response to signals received by cells from\\ntheir micro-environment (4). EMT is one of the molecular programs that\\noccur in the process of malignant transformation in cancer cells, in which\\nepithelial cells lose their epithelial characteristics and are replaced with mesenchymal phenotypes (5, 6).\\nVimentin is a type III filament intermediate protein which is mainly expressed on mesenchymal cell proteins such as fibroblasts, endothelial cells\\nand lymphocytes. Several studies suggest that the expression of vimentin in\\nbreast cancer is an important prognostic indicator (6).\\n\\n2. OBJECTIVE\\n\\nThe aim of this study was to assess the correlation between vimentin intensity and clinicopathological parameters in young women pasients with triple\\nnegative breast cancer.\\nORIGINAL PAPER | Med Arch. 2022 DEC; 76(6): 454-457\\n\\n\\x0c',\n",
       " 'Chemical\\nScience\\nEDGE ARTICLE\\n\\nCite this: Chem. Sci., 2023, 14, 2887\\nAll publication charges for this article\\nhave been paid for by the Royal Society\\nof Chemistry\\n\\nAll-in-One digital microﬂuidics pipeline for\\nproteomic sample preparation and analysis†\\nJiaxi Peng,abc Calvin Chan, a Shuailong Zhang, afg Alexandros A. Sklavounos,\\nMaxwell E. Olson,a Erica Y. Scott,abc Yechen Hu,abc Vigneshwar Rajesh, a\\nBingyu B. Li,bc M. Dean Chamberlain,abch Shen Zhang,di Hui Peng ae\\nand Aaron R. Wheeler *abc\\n\\nab\\n\\nHighly sensitive and reproducible analysis of samples containing low amounts of protein is restricted by\\nsample loss and the introduction of contaminants during processing. Here, we report an All-in-One\\ndigital microﬂuidic (DMF) pipeline for proteomic sample reduction, alkylation, digestion, isotopic labeling\\nand analysis. The system features end-to-end automation, with integrated thermal control for digestion,\\noptimized droplet additives for sample manipulation and analysis, and an automated interface to liquid\\nchromatography with tandem mass spectrometry (HPLC-MS/MS). Dimethyl labeling was integrated into\\nthe pipeline to allow for relative quantiﬁcation of the trace samples at the nanogram level, and the new\\npipeline was applied to evaluating cancer cell lines and cancer tissue samples. Several known proteins\\n(including HSP90AB1, HSPB1, LDHA, ENO1, PGK1, KRT18, and AKR1C2) and pathways were observed\\nbetween model breast cancer cell lines related to hormone response, cell metabolism, and cell\\nmorphology. Furthermore, diﬀerentially quantiﬁed proteins (such as PGS2, UGDH, ASPN, LUM, COEA1,\\nReceived 1st February 2023\\nAccepted 7th February 2023\\n\\nand PRELP) were found in comparisons of healthy and cancer breast tissues, suggesting potential utility\\nof the All-in-One pipeline for the emerging application of proteomic cancer sub-typing. In sum, the All-\\n\\nDOI: 10.1039/d3sc00560g\\nrsc.li/chemical-science\\n\\n1\\n\\nin-One pipeline represents a powerful new tool for automated proteome processing and analysis, with\\nthe potential to be useful for evaluating mass-limited samples for a wide range of applications.\\n\\nIntroduction\\n\\nMass spectrometry (MS)-based proteome pro\\ue103ling has emerged\\nas a powerful, unbiased technology for identi\\ue103cation and\\na\\n\\nDepartment of Chemistry, University of Toronto, 80 St. George Street, Toronto, ON,\\nM5S 3H6, Canada. E-mail: aaron.wheeler@utoronto.ca; Fax: +1-416-946-3865; Tel:\\n+1-416-946-3866\\n\\nb\\n\\nDonnelly Centre for Cellular and Biomolecular Research, University of Toronto, 160\\nCollege Street, Toronto, ON, M5S 3E1, Canada\\n\\nc\\nInstitute of Biomedical Engineering, University of Toronto, 164 College Street,\\nToronto, ON, M5S 3G9, Canada\\nd\\n\\nLunenfeld-Tanenbaum Research Institute, Mount Sinai Hospital, 600 University\\nAvenue, Toronto, ON, M5G 1X5, Canada\\n\\ne\\n\\nSchool of Environment, University of Toronto, 33 Willcocks Street, Toronto, ON, M5S\\n3E8, Canada\\n\\nf\\n\\nSchool of Mechatronical Engineering, Beijing Institute of Technology, Beijing, 100081,\\nChina\\n\\ng\\nBeijing Advanced Innovation Center for Intelligent Robots and Systems, Beijing\\nInstitute of Technology, Beijing, 100081, China\\nh\\n\\nSaskatchewan Cancer Agency, University of Saskatchewan, 107 Wiggins Road,\\nSaskatoon, SK S7N 5E5, Canada\\n\\ni\\n\\nClinical Research Center for Reproduction and Genetics in Hunan Province,\\nReproductive and Genetic Hospital of CITIC-XIANGYA, Changsha, Hunan, 410000,\\nChina\\n† Electronic supplementary information\\nhttps://doi.org/10.1039/d3sc00560g\\n\\n(ESI)\\n\\navailable.\\n\\nSee\\n\\nDOI:\\n\\n© 2023 The Author(s). Published by the Royal Society of Chemistry\\n\\nquantitation of the set of expressed proteins found in complex\\nbiological samples.1,2 The most common approach is “bottom\\nup” pro\\ue103ling, in which proteins are digested into constitutive\\npeptides prior to analysis by HPLC-MS/MS and database\\nsearching to identify the proteins that correlate with the\\nconstitutive peptides. State-of-the-art analysis techniques can\\nidentify such peptides when present in zeptomole quantities.3,4\\nThere are many exciting current trends in bottom-up proteome pro\\ue103ling for applications in biomedicine. One important\\ntrend focuses on pushing the boundary for how many proteins\\ncan be identi\\ue103ed from tiny samples, such as the lysate generated from a single mammalian cell. As recently as 3–5 years ago,\\nthe record for such studies was a few hundred proteins per\\ncell.5–7 More recently, methods that integrate ion mobility\\nspectroscopy (IMS) in-line with traditional MS approaches have\\npushed this record to thousands of proteins per cell.8–11 Another\\ntrend in bottom-up proteome pro\\ue103ling for applications in\\nbiomedicine is the development of strategies in which samples\\nare diﬀerentially labeled with mass-tags, to allow for semiquantitative comparisons between them.12–14 These techniques, which are increasingly being applied to clinical cancer\\ntyping,15,16 typically prioritize reproducible detection over the\\nabsolute number of proteins identi\\ue103ed. The current report falls\\nsquarely in this category of application.\\n\\nChem. Sci., 2023, 14, 2887–2900 | 2887\\n\\n\\x0c',\n",
       " 'International Journal of Nanomedicine\\n\\nDovepress\\nopen access to scientific and medical research\\n\\nOpen Access Full Text Article\\n\\nORIGINAL RESEARCH\\n\\nA Novel Nanoemulsion Formula for an Improved\\nDelivery of a Thalidomide Analogue to\\nTriple-Negative Breast Cancer; Synthesis,\\nFormulation, Characterization and Molecular\\nStudies\\nNoran M Tawfik 1,2 , Mohammed S Teiama 3,4 , Sameh Samir Iskandar 5 , Ahmed Osman 1,6 ,\\nSherif F Hammad 7,8\\n1\\n\\nBiotechnology Program, Basic and Applied Sciences Institute, Egypt-Japan University of Science and Technology, Alexandria, Egypt; 2Department of\\nZoology, Faculty of Science, Suez Canal University, Ismailia, Egypt; 3Department of Pharmaceutics and Industrial Pharmacy, Faculty of Pharmacy,\\nHelwan University, Cairo, Egypt; 4Department of Pharmaceutics, Faculty of Pharmacy, Galala University, Suez, Egypt; 5Fellow and Head of Surgical\\nOncology Department, Ismailia Teaching Oncology Hospital (GOTHI), Ismailia, Egypt; 6Department of Biochemistry, Faculty of Science, Ain Shams\\nUniversity, Cairo, Egypt; 7PharmD Programs, Egypt-Japan University of Science and Technology, Alexandria, Egypt; 8Department of Pharmaceutical\\nChemistry, Faculty of Pharmacy, Helwan University, Cairo, Egypt\\nCorrespondence: Noran M Tawfik; Ahmed Osman, Biotechnology Program, Basic and Applied Sciences Institute, Egypt-Japan University of Science\\nand Technology, New Borg El-Arab, Alexandria, 21934, Egypt, Email noran.tawfik@ejust.edu.eg; ahmed.osman@ejust.edu.eg\\n\\nBackground: Thalidomide (THD) and its analogues were recently reported as a promising treatment for different types of solid\\ntumors due to their antiangiogenic effect.\\nMethods: In this work, we synthesized a novel THD analogue (TA), and its chemistry was confirmed with different techniques such\\nas IR, mass spectroscopy, elemental analysis as well as 1H and 13C NMR. To increase solubility and anticancer efficacy, a new oil in\\nwater (O/W) nanoemulsion (NE) was used in the formulation of the analogue. The novel formula’s surface charge, size, stability,\\nFTIR, FE-TEM, in vitro drug release and physical characteristics were investigated. Furthermore, molecular docking studies were\\nconducted to predict the possible binding modes and molecular interactions behind the inhibitory activities of the THD and TA.\\nResults: TA showed a significant cytotoxic activity with IC50 ranging from 0.326 to 43.26 µmol/mL when evaluated against\\ncancerous cells such as MCF-7, HepG2, Caco-2, LNCaP and RKO cell lines. The loaded analogue showed more potential cytotoxicity\\nagainst MDA-MB-231 and MCF-7-ADR cell lines with IC50 values of 0.0293 and 0.0208 nmol/mL, respectively. Moreover, flow\\ncytometry of cell cycle analysis and apoptosis were performed showing a suppression in the expression levels of TGF-β, MCL-1,\\nVEGF, TNF-α, STAT3 and IL-6 in the MDA-MB-231 cell line.\\nConclusion: The novel NE formula dramatically reduced the anticancer dosage of TA from micromolar efficiency to nanomolar\\nefficiency. This indicates that the synthesized analogue exhibited high potency in the NE formulation and proved its efficacy against\\ntriple-negative breast cancer cell line.\\nKeywords: thalidomide analogue, antiangiogenic, nanoemulsion, cytotoxicity, triple-negative breast cancer, docking study\\n\\nIntroduction\\nAdvanced breast cancer is hard to treat since it is the most complex heterogenous recurrent disease in females.\\nSuppressing progression and prolongation of survival time is the main goal of clinical intervention.1 Its rigidity is due\\nto rapid metastasis, which is responsible for over 90% of mortality cases in patients.2 It is generally divided into different\\nsubtypes depending on the molecular expression of hormonal receptors; estrogen receptor (ER), progesterone receptor\\n(PR) and human epidermal growth factor receptor-2 (HER-2). The most aggressive type of breast cancer is the tripleInternational Journal of Nanomedicine 2023:18 1219–1243\\nReceived: 5 August 2022\\nAccepted: 20 December 2022\\nPublished: 11 March 2023\\n\\n1219\\n\\n© 2023 Tawfik et al. This work is published and licensed by Dove Medical Press Limited. The full terms of this license are available at https://www.dovepress.com/terms.\\nphp and incorporate the Creative Commons Attribution – Non Commercial (unported, v3.0) License (http://creativecommons.org/licenses/by-nc/3.0/). By accessing the\\nwork you hereby accept the Terms. Non-commercial uses of the work are permitted without any further permission from Dove Medical Press Limited, provided the work is properly attributed. For\\npermission for commercial use of this work, please see paragraphs 4.2 and 5 of our Terms (https://www.dovepress.com/terms.php).\\n\\n\\x0c',\n",
       " 'Exploration of Targeted Anti-tumor Therapy\\nOpen Access Original Article\\n\\nEnhancement of reactive oxygen species production in triple\\nnegative breast cancer cells treated with electric pulses\\nand resveratrol\\nPragatheiswar Giri1\\n\\n, Ignacio G. Camarillo2,3\\n\\n, Raji Sundararajan1*\\n\\nSchool of Engineering Technology, Purdue University, West Lafayette, IN 47907, USA\\nDepartment of Biological Sciences, Purdue University, West Lafayette, IN 47907, USA\\n3\\nPurdue University Center for Cancer Research, West Lafayette, IN 47907, USA\\n1\\n2\\n\\n*Correspondence: Raji Sundararajan, School of Engineering Technology, Purdue University, West Lafayette, IN 47907,\\nUSA. raji@purdue.edu\\nAcademic Editor: Katrin Sak, NGO Praeventio, Estonia\\nReceived: October 27, 2022 Accepted: December 28, 2022 Published: February 28, 2023\\n\\nCite this article: Giri P, Camarillo IG, Sundararajan R. Enhancement of reactive oxygen species production in triple\\nnegative breast cancer cells treated with electric pulses and resveratrol. Explor Target Antitumor Ther. 2023;4:42–56.\\nhttps://doi.org/10.37349/etat.2023.00122\\n\\nAbstract\\nAim: Triple negative breast cancer (TNBC) is difficult to treat since it lacks all the three most commonly\\ntargeted hormone receptors. Patients afflicted with TNBC are treated with platinum core chemotherapeutics,\\nsuch as cisplatin. Despite the initial effective anticancer effects of cisplatin, TNBC attenuates its effect and\\ndevelops resistance eventually, which results in tumor reoccurrence. Hence, there is a critical demand for\\neffective, alternative, and natural ways to treat TNBC. Towards this, a promising technique for inhibiting\\nTNBC cell proliferation involves promoting the production of reactive oxygen species (ROS), which triggers\\npro-apoptotic caspases 9 and 3. Resveratrol (RESV), an active bio compound found in naturally available\\nfruits, such as grapes, is utilized in this research for that. In addition, electrochemotherapy (ECT), which\\ninvolves the application of electrical pulses (EP), was utilized to enhance the uptake of RESV.\\nMethods: MDA-MB-231, human TNBC cells were treated with/out RESV, and eight 600–1,000 V/cm, 100 μs\\npulses at 1 Hz. The cells were characterized by using various assays, including viability assay, and ROS assay.\\n\\nResults: A TNBC cell viability of as low as 20% was obtained at 24 h (it was 13% at 60 h), demonstrating the\\npotential of this novel treatment. ROS production was the highest in the combination of EP at 1,000 V/cm\\nalong with RESV at 100 µmol/L.\\nConclusions: Results indicate that RESV has the potential as an anti-TNBC agent and that EP + RESV can\\nsignificantly enhance the cell death to reduce MDA-MB-231 cell viability by increasing ROS production and\\ntriggering apoptosis.\\n\\nKeywords\\n\\nResveratrol, triple negative breast cancer, reactive oxygen species, MDA-MB-231 cells, electrical pulses\\n© The Author(s) 2023. This is an Open Access article licensed under a Creative Commons Attribution 4.0 International\\nLicense (https://creativecommons.org/licenses/by/4.0/), which permits unrestricted use, sharing, adaptation, distribution\\nand reproduction in any medium or format, for any purpose, even commercially, as long as you give appropriate credit to the\\noriginal author(s) and the source, provide a link to the Creative Commons license, and indicate if changes were made.\\n\\nExplor Target Antitumor Ther. 2023;4:42–56 | https://doi.org/10.37349/etat.2023.00122\\n\\nPage 42\\n\\n\\x0c',\n",
       " 'TYPE\\n\\nEditorial\\n03 March 2023\\nDOI 10.3389/fonc.2023.1151742\\nPUBLISHED\\n\\nOPEN ACCESS\\nEDITED AND REVIEWED BY\\n\\nKara Britt,\\nPeter MacCallum Cancer Centre, Australia\\n*CORRESPONDENCE\\n\\nSherif Abdelaziz Ibrahim\\nisherif@cu.edu.eg;\\nisherif@sci.cu.edu.eg\\nMartin Götte\\nmgotte@uni-muenster.de\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nBreast Cancer,\\na section of the journal\\nFrontiers in Oncology\\n\\nEditorial: Cancer stem cells as\\nattractive targets for breast\\ncancer therapy\\nSherif Abdelaziz Ibrahim 1*, George W. Yip 2 and Martin Götte 3*\\n1\\nDepartment of Zoology, Faculty of Science, Cairo University, Giza, Egypt, 2 Department of Anatomy,\\nYong Loo Lin School of Medicine, National University of Singapore, Singapore, Singapore,\\n3\\nDepartment of Gynecology and Obstetrics, Münster University Hospital, Münster, Germany\\n\\nKEYWORDS\\n\\ncancer stem cells, cancer initiating cells, breast cancer, metastasis, epithelialmesenchymal transition, tumor microenvironment, immunotherapy\\n\\nRECEIVED\\n\\n26 January 2023\\n27 February 2023\\nPUBLISHED 03 March 2023\\n\\nEditorial on the Research Topic\\n\\nCITATION\\n\\nCancer stem cells as attractive targets for breast cancer therapy\\n\\nACCEPTED\\n\\nIbrahim SA, Yip GW and Götte M (2023)\\nEditorial: Cancer stem cells as attractive\\ntargets for breast cancer therapy.\\nFront. Oncol. 13:1151742.\\ndoi: 10.3389/fonc.2023.1151742\\nCOPYRIGHT\\n\\n© 2023 Ibrahim, Yip and Götte. This is an\\nopen-access article distributed under the\\nterms of the Creative Commons Attribution\\nLicense (CC BY). The use, distribution or\\nreproduction in other forums is permitted,\\nprovided the original author(s) and the\\ncopyright owner(s) are credited and that\\nthe original publication in this journal is\\ncited, in accordance with accepted\\nacademic practice. No use, distribution or\\nreproduction is permitted which does not\\ncomply with these terms.\\n\\nFrontiers in Oncology\\n\\nBreast cancer is the most prevalent cancer that affects females worldwide. It accounts\\nfor approximately 31% of cancers in women (1). Breast cancer can be classiﬁed into various\\nintrinsic subtypes based on the expression of estrogen receptor (ER), progesterone receptor\\n(PR), human epidermal growth factor receptor 2 (HER2), and the proliferative index Ki-67:\\nluminal A (ER-positive, PR-positive, HER2-negative, Ki-67 with less than 20% positivity);\\nluminal B-HER2- negative (ER-positive, HER2- negative, and either PR-low or Ki-67 with\\nmore than 20% positivity); luminal B HER2-positive (ER-positive or PR-positive, HER2positive or ampliﬁed, and regardless of Ki-67 status); HER2-enriched (ER-negative, PRnegative, HER-2-positive or ampliﬁed, and regardless of Ki-67 status); and triple-negative\\n(ER-negative, PR-negative, and HER2-negative) breast cancer (2–4). Despite recent\\nadvances in breast carcinoma treatment, including chemotherapy, radiotherapy, and\\nhormone (endocrine) therapy, as well as innovations in targeted therapeutics and\\nimmunotherapy (5), development of treatment resistance and tumor recurrence are still\\nmajor challenges. In contrast to differentiated cells, cancer stem cells (CSCs) or tumor\\ninitiating cells (TICs), which constitute a subpopulation of cells within the tumor bulk,\\nhave been linked to tumor relapse and resistance to conventional therapy (6–9). Therefore,\\ntargeting CSCs has emerged as a promising therapeutic strategy against cancer, including\\nbreast cancer. CSCs or TICs possess unique features, including self-renewal, expression of\\nmultidrug resistance proteins, high proliferative capacity, efﬁcient DNA repair capability,\\nresistance to apoptosis, and overexpression of therapeutic efﬂux proteins (10–15). Breast\\nCSCs are characterized by expression of cell surface makers CD44+/CD24- and/or\\nincreased detoxifying aldehyde dehydrogenase (ALDH) activity (6, 16). Breast CSCs are\\ninﬂuenced not only by cancer-cell autonomous factors but also by diverse cues derived\\nfrom the tumor microenvironment (TME) (17).\\nThis Research Topic collection comprises four papers (two original research, one\\nperspective, and one review article) coauthored by 28 researchers. We aimed to provide\\noriginal research and opinion pieces on breast CSCs, focusing on new molecular markers\\nand therapeutic targets, crosstalk between tumor microenvironmental cells and breast\\nCSCs, and signaling pathways involved in breast CSC regulation.\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Original Research\\n01 March 2023\\nDOI 10.3389/fonc.2023.1102254\\nTYPE\\n\\nPUBLISHED\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nL. J. Muhammad,\\nFederal University Kashere, Nigeria\\nREVIEWED BY\\n\\nAbbas Khan,\\nShanghai Jiao Tong University, China\\nKhurshid Ahmad,\\nYeungnam University, Republic of Korea\\n*CORRESPONDENCE\\n\\nLooket Dihge\\nlooket.dihge@med.lu.se\\nLisa Rydén\\nlisa.ryden@med.lu.se\\n†\\n\\nThese authors share last authorship\\n\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nBreast Cancer,\\na section of the journal\\nFrontiers in Oncology\\n18 November 2022\\n13 February 2023\\nPUBLISHED 01 March 2023\\n\\nThe implementation of NILS: A\\nweb-based artiﬁcial neural\\nnetwork decision support tool\\nfor noninvasive lymph node\\nstaging in breast cancer\\nLooket Dihge 1,2*, Pär-Ola Bendahl 3, Ida Skarping 3,4,\\nMalin Hjärtström 3, Mattias Ohlsson 5† and Lisa Rydén 1,6*†\\nDepartment of Clinical Sciences Lund, Division of Surgery, Lund University, Lund, Sweden,\\nDepartment of Plastic and Reconstructive Surgery, Skåne University Hospital, Malmö, Sweden,\\n3\\nDepartment of Clinical Sciences Lund, Division of Oncology, Lund University, Lund, Sweden,\\n4\\nDepartment of Clinical Physiology and Nuclear Medicine, Skåne University Hospital, Lund, Sweden,\\n5\\nDepartment of Astronomy and Theoretical Physics, Division of Computational Biology and Biological\\nPhysics, Lund University, Lund, Sweden, 6 Department of Surgery, Skåne University Hospital,\\nMalmö, Sweden\\n1\\n\\n2\\n\\nRECEIVED\\n\\nACCEPTED\\n\\nCITATION\\n\\nDihge L, Bendahl P-O, Skarping I,\\nHjärtström M, Ohlsson M and Rydén L\\n(2023) The implementation of NILS: A\\nweb-based artiﬁcial neural network\\ndecision support tool for noninvasive\\nlymph node staging in breast cancer.\\nFront. Oncol. 13:1102254.\\ndoi: 10.3389/fonc.2023.1102254\\nCOPYRIGHT\\n\\n© 2023 Dihge, Bendahl, Skarping,\\nHjärtström, Ohlsson and Rydén. This is an\\nopen-access article distributed under the\\nterms of the Creative Commons Attribution\\nLicense (CC BY). The use, distribution or\\nreproduction in other forums is permitted,\\nprovided the original author(s) and the\\ncopyright owner(s) are credited and that\\nthe original publication in this journal is\\ncited, in accordance with accepted\\nacademic practice. No use, distribution or\\nreproduction is permitted which does not\\ncomply with these terms.\\n\\nObjective: To implement artiﬁcial neural network (ANN) algorithms for\\nnoninvasive lymph node staging (NILS) to a decision support tool and facilitate\\nthe option to omit surgical axillary staging in breast cancer patients with low-risk\\nof nodal metastasis.\\nMethods: The NILS tool is a further development of an ANN prototype for the\\nprediction of nodal status. Training and internal validation of the original\\nalgorithm included 15 clinical and tumor-related variables from a consecutive\\ncohort of 800 breast cancer cases. The updated NILS tool included 10 topranked input variables from the original prototype. A workﬂow with four ANN\\npathways was additionally developed to allow different combinations of missing\\npreoperative input values. Predictive performances were assessed by area under\\nthe receiver operating characteristics curves (AUC) and sensitivity/speciﬁcity\\nvalues at deﬁned cut-points. Clinical utility was presented by estimating\\npossible sentinel lymph node biopsy (SLNB) reduction rates. The principles of\\nuser-centered design were applied to develop an interactive web-interface to\\npredict the patient’s probability of healthy lymph nodes. A technical validation of\\nthe interface was performed using data from 100 test patients selected to cover\\nall combinations of missing histopathological input values.\\nResults: ANN algorithms for the prediction of nodal status have been\\nimplemented into the web-based NILS tool for personalized, noninvasive\\nnodal staging in breast cancer. The estimated probability of healthy lymph\\nnodes using the interface showed a complete concordance with estimations\\nfrom the reference algorithm except in two cases that had been wrongly\\nincluded (ineligible for the technical validation). NILS predictive performance to\\ndistinguish node-negative from node-positive disease, also with missing values,\\ndisplayed AUC ranged from 0.718 (95% CI, 0.687-0.748) to 0.735 (95% CI, 0.7040.764), with good calibration. Sensitivity 90% and speciﬁcity 34% were\\n\\nFrontiers in Oncology\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Original Research\\n02 March 2023\\nDOI 10.3389/fonc.2023.1030124\\nTYPE\\n\\nPUBLISHED\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nBenedetta Pellegrino,\\nUniversity of Parma, Italy\\nREVIEWED BY\\n\\nSara Ravaioli,\\nMaria Cecilia Hospital, Italy\\nNguyen Minh Duc,\\nPham Ngoc Thach University of Medicine,\\nVietnam\\n*CORRESPONDENCE\\n\\nZhonghua Han\\nzhhan@fjmu.edu.cn\\nChunsen Xu\\nxuchunsen@fjmu.edu.cn\\n†\\nThese authors have contributed\\nequally to this work and share\\nﬁrst authorship\\n\\nClinical characteristics and\\noverall survival prognostic\\nnomogram for metaplastic\\nbreast cancer\\nCaihong Zheng 1,2,3†, Chengbin Fu 2,3,4†, Yahui Wen 1,2,3,\\nJiameng Liu 5, Shunguo Lin 2,3,4, Hui Han 2,3,4, Zhonghua Han 2,3,4*\\nand Chunsen Xu 2,3,4*\\n1\\nThe Graduate School of Fujian Medical University, Fuzhou, Fujian, China, 2Department of Breast Surgery,\\nFujian Medical University Union Hospital, Fuzhou, Fujian, China, 3Department of General Surgery, Fujian\\nMedical University Union Hospital, Fuzhou, Fujian, China, 4Breast Cancer Institute, Fujian Medical University,\\nFuzhou, Fujian, China, 5Department of Breast Surgery, Women and Children’s Hospital, School of Medicine,\\nXiamen University, Xiamen, Fujian, China\\n\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nBreast Cancer,\\na section of the journal\\nFrontiers in Oncology\\n28 August 2022\\n09 February 2023\\nPUBLISHED 02 March 2023\\nRECEIVED\\n\\nBackground: Metaplastic breast cancer (MBC) is a rare breast tumor and the\\nprognostic factors for survival in patients still remain controversial. This study\\naims to develop and validate a nomogram to predict the overall survival (OS) of\\npatients with MBC.\\n\\nACCEPTED\\n\\nCITATION\\n\\nZheng C, Fu C, Wen Y, Liu J, Lin S,\\nHan H, Han Z and Xu C (2023) Clinical\\ncharacteristics and overall survival\\nprognostic nomogram for metaplastic\\nbreast cancer.\\nFront. Oncol. 13:1030124.\\ndoi: 10.3389/fonc.2023.1030124\\nCOPYRIGHT\\n\\n© 2023 Zheng, Fu, Wen, Liu, Lin, Han, Han\\nand Xu. This is an open-access article\\ndistributed under the terms of the Creative\\nCommons Attribution License (CC BY). The\\nuse, distribution or reproduction in other\\nforums is permitted, provided the original\\nauthor(s) and the copyright owner(s) are\\ncredited and that the original publication in\\nthis journal is cited, in accordance with\\naccepted academic practice. No use,\\ndistribution or reproduction is permitted\\nwhich does not comply with these terms.\\n\\nFrontiers in Oncology\\n\\nMethods: We searched the Surveillance, Epidemiology, and End Results (SEER)\\ndatabase for data about patients including metaplastic breast cancer and\\ninﬁltrating ductal carcinoma (IDC) from 2010 to 2018. The survival outcomes\\nof patients between MBC and IDC were analyzed and compared with the KaplanMeier (KM) method. MBC patients were randomly allocated to the training set\\nand validation I set by a ratio of eight to two. Meanwhile, the performance of this\\nmodel was validated again by the validation II set, which consisted of MBC\\npatients from the Union Hospital of Fujian Medical University between 2010 and\\n2018. The independent prognostic factors were selected by univariate and\\nmultivariate Cox regression analyses. The nomogram was constructed to\\npredict individual survival outcomes for MBC patients. The discriminative\\npower, calibration, and clinical effectiveness of the nomogram were evaluated\\nby the concordance index (C-index), the receiver operating characteristic (ROC)\\ncurve, and the decision curve analysis (DCA).\\nResults: MBC had a signiﬁcantly higher T stage (T2 and above accounting for\\n75.1% vs 39.9%), fewer inﬁltrated lymph nodes (N0 accounted for 76.2% vs 67.7%),\\na lower proportion of ER (22.2% vs 81.2%), PR (13.6% vs 71.4%), and HER-2(6.7% vs\\n17.7%) positive, radiotherapy(51.6% vs 58.0%) but more chemotherapy(67.5% vs\\n44.7%), and a higher rate of mastectomy(53.2% vs 36.8%), which was discovered\\nwhen comparing the clinical baseline data between MBC and IDC. Age at\\ndiagnosis, T, N, and M stage, as well as surgery and radiation treatment, were\\nall signiﬁcant independent prognostic factors for overall survival (OS). In the\\nvalidation I cohort, the nomogram’s C-index (0.769 95% CI 0.710 -0.828) was\\nindicated to be considerably higher than the standard AJCC model’s (0.700 95%\\nCI 0.644 -0.756). Nomogram’s great predictive capability capacity further was\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Original Research\\n23 February 2023\\nDOI 10.3389/fonc.2023.819357\\nTYPE\\n\\nPUBLISHED\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nYun Dai,\\nFirst Afﬁliated Hospital of Jilin University,\\nJilin University, China\\nREVIEWED BY\\n\\nAlessandro De Vita,\\nScientiﬁc Institute of Romagna for the\\nStudy and Treatment of Tumors (IRCCS),\\nItaly\\nZhiqiang Liu,\\nTianjin Medical University, China\\n\\nAssociation of circulating tumor\\ncells and IMP3 expression with\\nmetastasis of osteosarcoma\\nShuangwu Dai 1†, Xinxin Shao 2†, Qingzhu Wei 1, Shaohua Du 1,\\nChanghe Hou 1, Haomiao Li 1* and Dadi Jin 1*\\n1\\nDepartment of Musculoskeletal Oncology, Center for Orthopaedic Surgery, The Third Afﬁliated\\nHospital of Southern Medical University, Guangzhou, China, 2 Department of Anesthesiology, The First\\nAfﬁliated Hospital of Sun Yat-Sen University, Guangzhou, China\\n\\n*CORRESPONDENCE\\n\\nDadi Jin\\nnyorthop@163.com\\nHaomiao Li\\nlihaomiao1977@hotmail.com\\n†\\nThese authors have contributed\\nequally to this work\\n\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nCancer Molecular Targets\\nand Therapeutics,\\na section of the journal\\nFrontiers in Oncology\\n21 November 2021\\nACCEPTED 16 January 2023\\nPUBLISHED 23 February 2023\\nRECEIVED\\n\\nCITATION\\n\\nDai S, Shao X, Wei Q, Du S, Hou C, Li H\\nand Jin D (2023) Association of circulating\\ntumor cells and IMP3 expression with\\nmetastasis of osteosarcoma.\\nFront. Oncol. 13:819357.\\ndoi: 10.3389/fonc.2023.819357\\nCOPYRIGHT\\n\\n© 2023 Dai, Shao, Wei, Du, Hou, Li and Jin.\\nThis is an open-access article distributed\\nunder the terms of the Creative Commons\\nAttribution License (CC BY). The use,\\ndistribution or reproduction in other\\nforums is permitted, provided the original\\nauthor(s) and the copyright owner(s) are\\ncredited and that the original publication in\\nthis journal is cited, in accordance with\\naccepted academic practice. No use,\\ndistribution or reproduction is permitted\\nwhich does not comply with these terms.\\n\\nBackground: Circulating tumor cells (CTCs) have been identiﬁed as a prognostic\\nbiomarker of tumors such as breast cancer and nasopharyngeal carcinoma,\\nbecause they are obtained through a simple and noninvasive blood draw or\\nliquid biopsy, but its clinical signiﬁcance in osteosarcoma is still unclear. In this\\nstudy, we analyzed the relationship between CTCs and clinicopathological\\nfeatures and discussed whether CTCs could be used as a biomarker for\\nmetastasis in osteosarcoma.\\nMethods: We enrolled 50 osteosarcoma patients with Enneking Stage IIB and\\nStage III and detected CTCs in 5 ml of peripheral blood samples collected from\\n®\\npatients using the Canpatrol CTC detection platform. Subsequently, multiplex\\nRNA in situ hybridization (RNA-ISH) based on various molecular markers was\\nperformed to identify and classify CTCs. The relationships between clinical\\npathological features and CTC counts, subtypes (epithelial type, E type; hybrid\\nepithelial/mesenchymal type, H type; mesenchymal type, M type), and insulin-like\\ngrowth factor mRNA-binding protein 3 (IMP3) expression in CTCs were analyzed.\\nResults: CTCs were detected in 86% (43/50) of the osteosarcoma patients. The\\nCTC counts, especially the total CTCs and H-type CTCs, signifcantly differed\\nbetween Enneking Stage IIB and Stage III patients (P < 0.05). No signiﬁcant\\ndifferences were observed between the CTC count or type and other\\nclinicopathological features (P > 0.05). There were signiﬁcant differences in the\\nexpression of IMP3 in different types of CTCs, and the IMP3 positive rates in E/H/M\\ntype CTCs were 38.4, 65.6, and 62.0%, respectively (P < 0.001). Receiver operating\\ncharacteristic (ROC) curve analysis showed that IMP3-positive CTC count had the\\nbest performance for diagnostic metastasis, with the largest area under the curve\\nof 0.873 and cutoff value of four cells/5ml blood (sensitivity = 87.5%; speciﬁcity =\\n82.4%). Serial CTC monitoring in one patient suggested that total CTCs and H-type\\nCTCs were associated with disease progression.\\nConclusion: This study demonstrates that the CTCs, especially the IMP3-positive\\nCTCs and H/M-type CTCs, are related to the metastasis of osteosarcoma.\\nKEYWORDS\\n\\ncirculating tumor cells, EMT, osteosarcoma, IMP3, metastasis\\n\\nFrontiers in Oncology\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Original Research\\n02 March 2023\\nDOI 10.3389/fonc.2023.1097513\\nTYPE\\n\\nPUBLISHED\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nAnastasios Papanastasiou,\\nUniversity of West Attica, Greece\\nREVIEWED BY\\n\\nIoannis S. Pateras,\\nNational and Kapodistrian University of\\nAthens, Greece\\nBeatriz Lozano,\\nUniversity of Cambridge, United Kingdom\\n\\nTumor senescence leads\\nto poor survival and\\ntherapeutic resistance in\\nhuman breast cancer\\nJingtong Zhai 1†, Jiashu Han 1,2†, Cong Li 1, Dan Lv 1, Fei Ma 1*\\nand Binghe Xu 1*\\n\\n*CORRESPONDENCE\\n\\nBinghe Xu\\nxubinghebm@163.com\\nFei Ma\\ndrmafei@126.com\\n\\n1\\nDepartment of Medical Oncology, National Cancer Center/National Clinical Research Center for\\nCancer/Cancer Hospital, Chinese Academy of Medical Sciences and Peking Union Medical College,\\nBeijing, China, 2 4 + 4 Medical Doctor Program, Chinese Academy of Medical Sciences & Peking\\nUnion Medical College, Beijing, China\\n\\n†\\nThese authors have contributed equally to\\nthis work\\n\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nBreast Cancer,\\na section of the journal\\nFrontiers in Oncology\\n14 November 2022\\n20 February 2023\\nPUBLISHED 02 March 2023\\nRECEIVED\\n\\nBackground: Breast cancer (BRCA) is the most common malignant tumor that\\nseriously threatens the health of women worldwide. Senescence has been\\nsuggested as a pivotal player in the onset and progression of tumors as well as\\nthe process of treatment resistance. However, the role of senescence in BRCA\\nremains unelucidated.\\n\\nACCEPTED\\n\\nCITATION\\n\\nZhai J, Han J, Li C, Lv D, Ma F and Xu B\\n(2023) Tumor senescence leads to poor\\nsurvival and therapeutic resistance in\\nhuman breast cancer.\\nFront. Oncol. 13:1097513.\\ndoi: 10.3389/fonc.2023.1097513\\nCOPYRIGHT\\n\\n© 2023 Zhai, Han, Li, Lv, Ma and Xu. This is\\nan open-access article distributed under the\\nterms of the Creative Commons Attribution\\nLicense (CC BY). The use, distribution or\\nreproduction in other forums is permitted,\\nprovided the original author(s) and the\\ncopyright owner(s) are credited and that\\nthe original publication in this journal is\\ncited, in accordance with accepted\\nacademic practice. No use, distribution or\\nreproduction is permitted which does not\\ncomply with these terms.\\n\\nFrontiers in Oncology\\n\\nMethods: The clinical and transcriptomic data of 2994 patients with BRCA were\\nobtained from The Cancer Genome Atlas and the METABRIC databases.\\nConsensus clustering revealed senescence-associated subtypes of BRCA\\npatients. Functional enrichment analysis explored biological effect of\\nsenescence. We then applied weighted gene co-expression network analysis\\n(WGCNA) and LASSO regression to construct a senescence scoring model,\\nSindex. Survival analysis validated the effectiveness of Sindex to predict the\\noverall survival (OS) of patients with BRCA. A nomogram was constructed by\\nmultivariate Cox regression. We used Oncopredict algorithm and real-world data\\nfrom clinical trials to explore the value of Sindex in predicting response to cancer\\ntherapy.\\nResults: We identiﬁed two distinct senescence-associated subtypes, noted low\\nsenescence CC1 and high senescence CC2. Survival analysis revealed worse OS\\nassociated with high senescence, which was also validated with patient samples\\nfrom the National Cancer Center in China. Further analysis revealed extensively\\ncell division and suppression of extracellular matrix process, along with lower\\nstromal and immune scores in the high senescence CC2. We then constructed a\\n37 signature gene scoring model, Sindex, with robust predictive capability in\\npatients with BRCA, especially for long time OS beyond 10 years. We\\ndemonstrated that the Sene-high subtype was resistant to CDK inhibitors but\\nsensitive to proteosome inhibitors, and there was no signiﬁcant difference in\\npaclitaxel chemotherapy and immunotherapy between patients with different\\nsenescence statuses.\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Original Research\\n02 March 2023\\nDOI 10.3389/fonc.2023.938189\\nTYPE\\n\\nPUBLISHED\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nAnna Diana,\\nOspedale del Mare, Italy\\nREVIEWED BY\\n\\nNguyen Minh Duc,\\nPham Ngoc Thach University of Medicine,\\nVietnam\\nEva Ruvalcaba-Limon,\\nFUCAM, Mexico\\n\\nThe utility of diffusion-weighted\\nimaging for differentiation of\\nphyllodes tumor from\\nﬁbroadenoma and breast cancer\\nJinzhi Fang 1†, Yuzhong Zhang 1†, Ruifeng Li 1†, Lanlan Liang 2,3†,\\nJuan Yu 4†, Ziqi Hu 1†, Lingling Zhou 1 and Renwei Liu 1*\\n\\n*CORRESPONDENCE\\n\\nRenwei Liu\\nlhhospital@sina.com\\n†\\nThese authors have contributed\\nequally to this work and share\\nﬁrst authorship\\n\\n1\\nDepartment of Radiology, Afﬁliated Longhua People’s Hospital, Southern Medical University\\n(Longhua People’s Hospital), Shenzhen, China, 2 Clinical Medical College of Dali University, Dali, China,\\n3\\nDepartment of Radiology, Afﬁliated Hospital of Yunnan University, Kunming, China, 4 Department of\\nRadiology, The First Afﬁliated Hospital of Shenzhen University, Health Science Center, Shenzhen\\nSecond People’s Hospital, Shenzhen, China\\n\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nBreast Cancer,\\na section of the journal\\nFrontiers in Oncology\\n07 May 2022\\n14 February 2023\\nPUBLISHED 02 March 2023\\nRECEIVED\\n\\nACCEPTED\\n\\nCITATION\\n\\nFang J, Zhang Y, Li R, Liang L, Yu J, Hu Z,\\nZhou L and Liu R (2023) The utility of\\ndiffusion-weighted imaging for\\ndifferentiation of phyllodes tumor from\\nﬁbroadenoma and breast cancer.\\nFront. Oncol. 13:938189.\\ndoi: 10.3389/fonc.2023.938189\\nCOPYRIGHT\\n\\n© 2023 Fang, Zhang, Li, Liang, Yu, Hu, Zhou\\nand Liu. This is an open-access article\\ndistributed under the terms of the Creative\\nCommons Attribution License (CC BY). The\\nuse, distribution or reproduction in other\\nforums is permitted, provided the original\\nauthor(s) and the copyright owner(s) are\\ncredited and that the original publication in\\nthis journal is cited, in accordance with\\naccepted academic practice. No use,\\ndistribution or reproduction is permitted\\nwhich does not comply with these terms.\\n\\nObjective: To evaluate the utility of apparent diffusion coefﬁcient (ADC) values\\nfor differentiating breast tumors.\\nMethods: The medical records of 17 patients with phyllodes tumor [PT; circular\\nregions of interest (ROI-cs) n = 171], 74 patients with ﬁbroadenomas (FAs; ROIcs, n = 94), and 57 patients with breast cancers (BCs; ROI-cs, n = 104) conﬁrmed\\nby surgical pathology were retrospectively reviewed.\\nResults: There were signiﬁcant differences between PTs, FAs, and BCs in\\nADCmean, ADCmax, and ADCmin values. The cutoff ADCmean for\\ndifferentiating PTs from FAs was 1.435 × 10−3 mm2/s, PTs from BCs was 1.100\\n× 10−3 mm2/s, and FAs from BCs was 0.925 × 10−3 mm2/s. There were signiﬁcant\\ndifferences between benign PTs, borderline PTs, and malignant PTs in ADCmean,\\nADCmax, and ADCmin values. The cutoff ADCmean for differentiating benign PTs\\nfrom borderline PTs was 1.215 × 10−3 mm2/s, and borderline PTs from malignant\\nPTs was 1.665 × 10−3 mm2/s.\\nConclusion: DWI provides quantitative information that can help distinguish\\nbreast tumors.\\nKEYWORDS\\n\\ndiffusion-weighted image (DWI), apparent diffusion coefﬁcient (ADC), value, magnetic\\nresonance imaging (MRI), breast tumors, phyllodes tumors\\n\\nIntroduction\\nPhyllodes tumor (PT), ﬁrst introduced by Muller in 1838, is a ﬁbroepithelial neoplasm\\nthat is histologically similar to a ﬁbroadenoma (FA). PTs are rare, accounting for 2% to\\n4.4% of all diagnosed FAs in one institution (1). Breast magnetic resonance imaging (MRI)\\nhas an overall sensitivity of 90% and a speciﬁcity of 72% for detecting breast lesions (2, 3).\\n\\nFrontiers in Oncology\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Editorial\\n03 March 2023\\n10.3389/fmolb.2023.1167337\\n\\nTYPE\\n\\nPUBLISHED\\nDOI\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nWilliam C. Cho,\\nQEH, Hong Kong SAR, China\\nREVIEWED BY\\n\\nTeresa Rubio-Tomás,\\nUniversity of Crete, Greece\\n*CORRESPONDENCE\\n\\nLan Zhang,\\nzhanglanx_9@126.com\\nJin Zhang,\\nzhangjin1989@szu.edu.cn\\nSPECIALTY SECTION\\n\\nThis article was submitted to Molecular\\nDiagnostics and Therapeutics,\\na section of the journal\\nFrontiers in Molecular Biosciences\\n16 February 2023\\n24 February 2023\\nPUBLISHED 03 March 2023\\n\\nEditorial: Gynecological and\\ngastrointestinal cancers: Recent\\nadvances in molecular diagnosis\\nand targeted therapy\\nLan Zhang 1*, Guan Wang 2, Lingjuan Zhu 3, Wei Wei 4 and\\nJin Zhang 5*\\n1\\nSichuan Engineering Research Center for Biomimetic Synthesis of Natural Drugs, School of Life Science\\nand Engineering, Southwest Jiaotong University, Chengdu, China, 2State Key Laboratory of Biotherapy\\nand Cancer Center, Nursing Key Laboratory of Sichuan Province, Innovation Center of Nursing Research,\\nWest China Hospital, and Collaborative Innovation Center of Biotherapy, Sichuan University, Chengdu,\\nChina, 3Key Laboratory of Structure-Based Drug Design and Discovery of Ministry of Education, School of\\nTraditional Chinese Materia Medica, Shenyang Pharmaceutical University, Shenyang, China, 4Blood Cell\\nDevelopment and Function Program, Fox Chase Cancer Center, Philadelphia, PA, United States, 5School\\nof Pharmaceutical Sciences, Health Science Center, Shenzhen University, Shenzhen, China\\n\\nRECEIVED\\n\\nACCEPTED\\n\\nCITATION\\n\\nZhang L, Wang G, Zhu L, Wei W and\\nZhang J (2023), Editorial: Gynecological\\nand gastrointestinal cancers: Recent\\nadvances in molecular diagnosis and\\ntargeted therapy.\\nFront. Mol. Biosci. 10:1167337.\\ndoi: 10.3389/fmolb.2023.1167337\\nCOPYRIGHT\\n\\n© 2023 Zhang, Wang, Zhu, Wei and\\nZhang. This is an open-access article\\ndistributed under the terms of the\\nCreative Commons Attribution License\\n(CC BY). The use, distribution or\\nreproduction in other forums is\\npermitted, provided the original author(s)\\nand the copyright owner(s) are credited\\nand that the original publication in this\\njournal is cited, in accordance with\\naccepted academic practice. No use,\\ndistribution or reproduction is permitted\\nwhich does not comply with these terms.\\n\\nFrontiers in Molecular Biosciences\\n\\nKEYWORDS\\n\\nbreast cancer, pancreatic adenocarcinoma, colorectal cancer, prognosis, molecular\\nmechanisms, targeted therapy\\n\\nEditorial on the Research Topic\\nGynecological and gastrointestinal cancers: Recent advances in molecular\\ndiagnosis and targeted therapy\\n\\nCancer is often rendered an uncurable disease, which poses a great threat to human\\nhealth worldwide. Among all kinds of cancers, gynecological cancers (including breast,\\novarian, cervical, and endometrial cancer) and gastrointestinal cancers (including gastric,\\ncolorectal, liver, and pancreatic cancer) are the most common cancer types worldwide (Ciou\\net al., 2022; Murphy et al., 2018). Although chemotherapy, radiotherapy, and surgical\\nexcision are commonly used to treat these gynecological and gastrointestinal cancers, cancer\\ncells frequently develop resistance to conventional cancer therapy due to a variety of\\nmechanisms (Davern et al., 2020; Marchetti et al., 2021). Accordingly, new therapeutic\\nstrategies such as the identiﬁcation of novel potential molecular targets or biomarkers, as well\\nas the development of new molecular diagnostic approaches and targeted therapy (e.g.,\\nproteins, transcription factors, non-coding RNA, targeted small-molecule drugs, etc.) are\\ngreatly needed to overcome inherent and acquired resistance, which would provide a\\npromising therapeutic approach for the treatment of various gynecological and\\ngastrointestinal cancers.\\nBreast cancer is the most common global malignancy and one of the most serious\\ngynecological cancers affecting women, which also represents currently a top biomedical\\nresearch priority (Akram et al., 2017). And its incidence and mortality rates are expected to\\nincrease continuously the next years (Barzaman et al., 2020). Accumulating evidence\\nsuggests that circular RNAs (circRNAs) are highly correlated with tumor progression\\nand pathogenesis in breast cancer (Sang et al., 2019). Whereas, their regulatory roles\\nand corresponding mechanisms in breast cancer are still unknown to some extent.\\nTherefore, Lin et al. explored the circRNA-mediated competive endogenous RNA\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Implementing an Inclusive,\\nMultidisciplinary Supportive\\nCare Model to Provide\\nIntegrated Care to Breast\\nand Gynaecological Cancer\\nSurvivors: A Case Study at an\\nAsian Ambulatory Cancer Centre\\nYU KE\\n\\nMOTHI BABU RAMALINGAM\\n\\nYUNG YING TAN\\n\\nDANIEL SONG CHIEK QUAH\\n\\nPATRICIA SOEK HUI NEO\\n\\nLITA CHEW\\n\\nGRACE MEIJUAN YANG\\n\\nPHEBE EN HUI SI\\n\\nKILEY WEI-JEN LOH\\n\\nBENG CHOO TAY\\n\\nSHIRLYNN HO\\n\\nALEXANDRE CHAN\\n\\nINTEGRATED CARE\\nCASE\\n\\nYEE PIN TAN\\n*Author affiliations can be found in the back matter of this article\\n\\nABSTRACT\\nIntroduction: Supportive care models considering inclusivity and community services\\nto improve integrated care for cancer survivors are limited. In this case study, we\\ndescribed the implementation of a multidisciplinary care model employing routine\\ndistress screening and embedded integrated care pathways to integrate care across\\ndisciplines and care sectors, while remaining inclusive of the multi-ethnic and\\nmultilingual population in Singapore. We reported implementation outcomes after 18\\nmonths of implementation.\\nDescription: We reviewed the model’s process indicators from September 2019 to\\nFebruary 2021 at the largest public ambulatory cancer centre. Outcomes assessed\\nincluded penetration, fidelity to screening protocol, and feasibility in three aspects\\n– inclusiveness of different ethnic and language groups, responsiveness to survivors\\nreporting high distress, and types of community service referrals.\\nDiscussion/conclusion: We elucidated opportunities to promote access to community\\nservices and inclusivity. Integration of community services from tertiary settings\\nshould be systematic through mutually beneficial educational and outreach\\ninitiatives, complemented by their inclusion in integrated care pathways to encourage\\nsystematic referrals and care coordination. A hybrid approach to service delivery is\\ncrucial in ensuring inclusivity while providing flexibility towards external changes such\\nas the COVID-19 pandemic. Future work should explore using telehealth to bolster\\ninclusiveness and advance community care integration.\\n\\nCORRESPONDING AUTHOR:\\nDr Alexandre Chan\\nUniversity of California, 101\\nTheory, Suite 100, Mail Code:\\n802 W Peltason Dr, Irvine,\\nCA 92697-4625, USA\\na.chan@uci.edu\\n\\nKEYWORDS:\\ndistress screening; cancer;\\nintegrated care; supportive\\ncare; inclusive; COVID-19\\nTO CITE THIS ARTICLE:\\nKe Y, Tan YY, Neo PSH, Yang\\nGM, Loh KW-J, Ho S, Tan YP,\\nRamalingam MB, Quah DSC,\\nChew L, Si PEH, Tay BC, Chan\\nA. Implementing an Inclusive,\\nMultidisciplinary Supportive\\nCare Model to Provide\\nIntegrated Care to Breast\\nand Gynaecological Cancer\\nSurvivors: A Case Study at\\nan Asian Ambulatory Cancer\\nCentre. International Journal of\\nIntegrated Care, 2023; 23(1):\\n14, 1–16. DOI: https://doi.\\norg/10.5334/ijic.6480\\n\\n\\x0c',\n",
       " 'Original Research\\n02 March 2023\\n10.3389/fgene.2023.1133443\\n\\nTYPE\\n\\nPUBLISHED\\nDOI\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nMiljana Tanic,\\nInstitute of Oncology and Radiology of\\nSerbia, Serbia\\nREVIEWED BY\\n\\nBarbara Ruszkowska-Ciastek,\\nNicolaus Copernicus University in Toruń,\\nPoland\\nMaria Teresa Vietri,\\nSeconda Università Degli Studi Di Napoli,\\nItaly\\n*CORRESPONDENCE\\n\\nAlexandra M. Binder,\\nambinder@hawaii.edu\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nEpigenomics and Epigenetics,\\na section of the journal\\nFrontiers in Genetics\\n\\nDNA methylation patterns\\nassociated with breast cancer\\nprognosis that are speciﬁc to\\ntumor subtype and menopausal\\nstatus\\nDo Hyun Kim 1, Alexandra M. Binder 2,3*, Hua Zhou 1,4 and\\nSu Yon Jung 5,6\\n1\\nDepartment of Biostatistics, Fielding School of Public Health, University of California, Los Angeles, Los\\nAngeles, CA, United States, 2Cancer Epidemiology Program, University of Hawaii Cancer Center,\\nHonolulu, HI, United States, 3Department of Epidemiology, Fielding School of Public Health, University of\\nCalifornia, Los Angeles, Los Angeles, CA, United States, 4Department of Computational Medicine,\\nUniversity of California, Los Angeles, Los Angeles, CA, United States, 5Translational Sciences Section,\\nSchool of Nursing, University of California, Los Angeles, Los Angeles, CA, United States, 6Jonsson\\nComprehensive Cancer Center, University of California, Los Angeles, Los Angeles, CA, United States\\n\\n29 December 2022\\n20 February 2023\\nPUBLISHED 02 March 2023\\nRECEIVED\\n\\nACCEPTED\\n\\nCITATION\\n\\nKim DH, Binder AM, Zhou H and Jung SY\\n(2023), DNA methylation patterns\\nassociated with breast cancer prognosis\\nthat are speciﬁc to tumor subtype and\\nmenopausal status.\\nFront. Genet. 14:1133443.\\ndoi: 10.3389/fgene.2023.1133443\\nCOPYRIGHT\\n\\n© 2023 Kim, Binder, Zhou and Jung. This\\nis an open-access article distributed\\nunder the terms of the Creative\\nCommons Attribution License (CC BY).\\nThe use, distribution or reproduction in\\nother forums is permitted, provided the\\noriginal author(s) and the copyright\\nowner(s) are credited and that the original\\npublication in this journal is cited, in\\naccordance with accepted academic\\npractice. No use, distribution or\\nreproduction is permitted which does not\\ncomply with these terms.\\n\\nFrontiers in Genetics\\n\\nTumor subtype and menopausal status are strong predictors of breast cancer (BC)\\nprognosis. We aimed to ﬁnd and validate subtype- or menopausal-status-speciﬁc\\nchanges in tumor DNA methylation (DNAm) associated with all-cause mortality or\\nBC progression. Associations between site-speciﬁc tumor DNAm and BC\\nprognosis were estimated among The Cancer Genome Atlas participants (n =\\n692) with Illumina Inﬁnium HumanMethylation450 BeadChip array data. All-cause\\nmortality and BC progression were modeled using Cox proportional hazards\\nmodels stratiﬁed by tumor subtypes, adjusting for age, race, stage, menopausal\\nstatus, tumor purity, and cell type proportion. Effect measure modiﬁcation by\\nsubtype and menopausal status were evaluated by incorporating a product term\\nwith DNAm. Site-speciﬁc inference was used to identify subtype- or menopausalstatus-speciﬁc differentially methylated regions (DMRs) and functional pathways.\\nThe validation of the results was carried out on an independent dataset\\n(GSE72308; n = 180). We identiﬁed a total of ﬁfteen unique CpG probes that\\nwere signiﬁcantly associated (P ≤ 1 × 10−7 ) with survival outcomes in subtype- or\\nmenopausal-status-speciﬁc manner. Seven probes were associated with overall\\nsurvival (OS) or progression-free interval (PFI) for women with luminal A subtype,\\nand four probes were associated with PFI for women with luminal B subtype. Five\\nprobes were associated with PFI for post-menopausal women. A majority of\\nsigniﬁcant probes showed a lower risk of OS or BC progression with higher DNAm.\\nWe identiﬁed subtype- or menopausal-status-speciﬁc DMRs and functional\\npathways of which top associated pathways differed across subtypes or\\nmenopausal status. None of signiﬁcant probes from site-speciﬁc analyses met\\ngenome-wide signiﬁcant level in validation analyses while directions and\\nmagnitudes of coefﬁcients showed consistent pattern. We have identiﬁed\\nsubtype- or menopausal-status-speciﬁc DNAm biomarkers, DMRs and\\nfunctional pathways associated with all-cause mortality or BC progression,\\nalbeit with limited validation. Future studies with larger independent cohort of\\nnon-post-menopausal women with non-luminal A subtypes are warranted for\\nidentifying subtype- and menopausal-status-speciﬁc DNAm biomarkers for BC\\nprognosis.\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Original Research\\n03 March 2023\\n10.3389/fgene.2023.1041042\\n\\nTYPE\\n\\nPUBLISHED\\nDOI\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nLili Chen,\\nHuazhong University of Science and\\nTechnology, China\\nREVIEWED BY\\n\\nLongquan Shao,\\nSouthern Medical University, China\\nYaru Guo,\\nPeking University Hospital of\\nStomatology, China\\n*CORRESPONDENCE\\n\\nYue Guo,\\nguoyue@csu.edu.cn\\nYun-Zhi Feng,\\nfengyunzhi001@csu.edu.cn\\n†\\n\\nThese authors have contributed equally\\nto this work and share ﬁrst authorship\\n\\nSPECIALTY SECTION\\n\\nThis article was submitted to Cancer\\nGenetics and Oncogenomics,\\na section of the journal\\nFrontiers in Genetics\\n10 September 2022\\n20 February 2023\\nPUBLISHED 03 March 2023\\n\\nPrognostic biomarker GSTK1 in\\nhead and neck squamous cell\\ncarcinoma and its correlation with\\nimmune inﬁltration and DNA\\nmethylation\\nYao Feng 1†, Ying-Hui Zhou 1,2†, Jie Zhao 1, Xiao-Lin Su 1,\\nNing-Xin Chen 1, Ya-Qiong Zhao 1, Qin Ye 1, Jing Hu 1,\\nZe-Yue Ou-Yang 1, Meng-Mei Zhong 1, Yi-Fan Yang 3,\\nPeng-Ju Han 4, Yue Guo 1* and Yun-Zhi Feng 1*\\n1\\nDepartment of Stomatology, The Second Xiangya Hospital, Central South University, Changsha, Hunan,\\nChina, 2National Clinical Research Center for Metabolic Diseases, Hunan Provincial Key Laboratory of\\nMetabolic Bone Diseases, and Department of Metabolism and Endocrinology, The Second Xiangya\\nHospital of Central South University, Changsha, Hunan, China, 3Xiangya School of Stomatology, Central\\nSouth University, Changsha, China, 4College of Life Sciences, Sichuan University, Chengdu, Sichuan,\\nChina\\n\\nRECEIVED\\n\\nACCEPTED\\n\\nCITATION\\n\\nFeng Y, Zhou Y-H, Zhao J, Su X-L,\\nChen N-X, Zhao Y-Q, Ye Q, Hu J,\\nOu-Yang Z-Y, Zhong M-M, Yang Y-F,\\nHan P-J, Guo Y and Feng Y-Z (2023),\\nPrognostic biomarker GSTK1 in head and\\nneck squamous cell carcinoma and its\\ncorrelation with immune inﬁltration and\\nDNA methylation.\\nFront. Genet. 14:1041042.\\ndoi: 10.3389/fgene.2023.1041042\\nCOPYRIGHT\\n\\n© 2023 Feng, Zhou, Zhao, Su, Chen,\\nZhao, Ye, Hu, Ou-Yang, Zhong, Yang,\\nHan, Guo and Feng. This is an openaccess article distributed under the terms\\nof the Creative Commons Attribution\\nLicense (CC BY). The use, distribution or\\nreproduction in other forums is\\npermitted, provided the original author(s)\\nand the copyright owner(s) are credited\\nand that the original publication in this\\njournal is cited, in accordance with\\naccepted academic practice. No use,\\ndistribution or reproduction is permitted\\nwhich does not comply with these terms.\\n\\nFrontiers in Genetics\\n\\nBackground: Glutathione S-transferase kappa 1 (GSTK1) is critical in sarcoma and\\nbreast cancer (BRCA) development. However, the clinical signiﬁcance of GSTK1 in\\nhead and neck squamous cell carcinoma (HNSC) remains unclear. This study is the\\nﬁrst investigation into the role of GSTK1 in HNSC.\\nMethods: All original data were downloaded from the Cancer Genome Atlas\\n(TCGA) dataset and veriﬁed by R Base Package 4.2.0. The expression of GSTK1 in\\nvarious cancers was explored with TIMER and TCGA databases. Prognostic value\\nof GSTK1 was analyzed via survival module of Kaplan-Meier plotter and Human\\nProtein Atlas database and Cox regression analysis. The association between\\nGSTK1 and clinical features was evaluated by Wilcoxon signed-rank test and\\nlogistic regression analysis. The relationship between GSTK1 and immune\\ninﬁltration and methylation level was further explored. The expression of\\nGSTK1 and its correlation with immune cell inﬁltration was veriﬁed by\\nImmunohistochemical staining (IHC).\\nResults: GSTK1 was lower in HNSC, BRCA, Lung squamous cell carcinoma, and\\nThyroid carcinoma than in para-carcinoma. Low GSTK1 expression was associated\\nwith worse overall survival in Bladder urothelial carcinoma, Kidney renal papillary\\ncell carcinoma, BRCA, and HNSC. However, only in BRCA and HNSC,\\nGSTK1 expression in tumors was lower than that in normal tissues. Cox\\nregression analyses conﬁrmed that GSKT1 was an independent prognostic\\nfactor of overall survival in HNSC patients. The decrease in GSTK1 expression\\nin HNSC was signiﬁcantly correlated with high T stage and smoker history. IHC\\nshowed that the expression level of GSTK1 in HNSC was lower than that in paracarcinoma. In addition, GSEA showed that three pathways related to immune\\ninﬁltration were positively correlated, while two pathways related to DNA\\nmethylation were negatively correlated with expression of GSTK1. Further\\nanalysis showed that GSTK1 was moderately positively correlated with the\\ninﬁltration level of T cells and Cytotoxic cells, which was further conﬁrmed by\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'Original Research\\n01 March 2023\\n10.3389/fgene.2023.1002157\\n\\nTYPE\\n\\nPUBLISHED\\nDOI\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nHaiming Dai,\\nMayo Clinic, United States\\nREVIEWED BY\\n\\nShengchun Liu,\\nFirst Afﬁliated Hospital of Chongqing\\nMedical University, China\\nNan Wang,\\nFirst Afﬁliated Hospital of Zhengzhou\\nUniversity, China\\n*CORRESPONDENCE\\n\\nHua Huang,\\nntdxliuyifei@sina.com\\nJian-Guo Zhang,\\njgz_edu@163.com\\n\\nConstruction of a fatty acid\\nmetabolism-related gene\\nsignature for predicting prognosis\\nand immune response in breast\\ncancer\\nLi Qian 1†, Yi-Fei Liu 1†, Shu-Min Lu 2, Juan-Juan Yang 1,\\nHua-Jie Miao 1, Xin He 1, Hua Huang 1* and Jian-Guo Zhang 1*\\n1\\nDepartment of Pathology, Afﬁliated Hospital of Nantong University, Nantong, China, 2Department of\\nOncology, Shanghai Jiaotong University School of Medicine Xinhua Hospital, Shanghai, China\\n\\n†\\n\\nThese authors have contributed equally\\nto this work and share ﬁrst authorship\\n\\nSPECIALTY SECTION\\n\\nThis article was submitted to Cancer\\nGenetics and Oncogenomics,\\na section of the journal\\nFrontiers in Genetics\\n24 July 2022\\n20 February 2023\\nPUBLISHED 01 March 2023\\nRECEIVED\\n\\nACCEPTED\\n\\nCITATION\\n\\nQian L, Liu Y-F, Lu S-M, Yang J-J,\\nMiao H-J, He X, Huang H and Zhang J-G\\n(2023), Construction of a fatty acid\\nmetabolism-related gene signature for\\npredicting prognosis and immune\\nresponse in breast cancer.\\nFront. Genet. 14:1002157.\\ndoi: 10.3389/fgene.2023.1002157\\nCOPYRIGHT\\n\\n© 2023 Qian, Liu, Lu, Yang, Miao, He,\\nHuang and Zhang. This is an open-access\\narticle distributed under the terms of the\\nCreative Commons Attribution License\\n(CC BY). The use, distribution or\\nreproduction in other forums is\\npermitted, provided the original author(s)\\nand the copyright owner(s) are credited\\nand that the original publication in this\\njournal is cited, in accordance with\\naccepted academic practice. No use,\\ndistribution or reproduction is permitted\\nwhich does not comply with these terms.\\n\\nBackground: Breast cancer has the highest incidence among malignant tumors in\\nwomen, and its prevalence ranks ﬁrst in global cancer morbidity.\\nAim: This study aimed to explore the feasibility of a prognostic model for patients\\nwith breast cancer based on the differential expression of genes related to fatty\\nacid metabolism.\\nMethods: The mRNA expression matrix of breast cancer and paracancer tissues\\nwas downloaded from The Cancer Genome Atlas database. The differentially\\nexpressed genes related to fatty acid metabolism were screened in R language.\\nThe TRRUST database was used to predict transcriptional regulators related to hub\\ngenes and construct an mRNA–transcription factor interaction network. A\\nconsensus clustering approach was used to identify different fatty acid\\nregulatory patterns. In combination with patient survival data, Lasso and\\nmultivariate Cox proportional risk regression models were used to establish\\npolygenic prognostic models based on fatty acid metabolism. The median risk\\nscore was used to categorize patients into high- and low-risk groups.\\nKaplan–Meier survival curves were used to analyze the survival differences\\nbetween both groups. The Cox regression analysis included risk score and\\nclinicopathological factors to determine whether risk score was an\\nindependent risk factor. Models based on genes associated with fatty acid\\nmetabolism were evaluated using receiver operating characteristic curves. A\\ncomparison was made between risk score levels and the fatty acid\\nmetabolism-associated genes in different subtypes of breast cancer. The\\ndifferential gene sets of the Kyoto Encyclopedia of Genes and Genomes for\\nscreening high- and low-risk populations were compared using a gene set\\nenrichment analysis. Furthermore, we utilized CIBERSORT to examine the\\nabundance of immune cells in breast cancer in different clustering models.\\n\\nAbbreviations: ACADL, long-chain acyl coenzyme A dehydrogenase; DEG, differentially expressed gene;\\nFA, fatty acid; GO, gene ontology; GSEA, gene set enrichment analysis; HSD17B7, 17β-hydroxysteroid\\ndehydrogenase type 7; KEGG, Kyoto Encyclopedia of Genes and Genomes; NK, natural killer; MAOA,\\nmonoamine oxidase A; ME1, malic enzyme 1; TCGA, The Cancer Genome Atlas; TDO2, tryptophan 2,3dioxygenase; TF, transcription factor; TME, tumor microenvironment; BRCA, breast cancer.\\n\\nFrontiers in Genetics\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'RESEARCH ARTICLE\\n\\nSelect HDAC Inhibitors Enhance Osteolysis and Bone\\nMetastasis Outgrowth but Can Be Mitigated\\nWith Bisphosphonate Therapy\\nMiranda E Clements,1,2 Lauren Holtslander,2,3 Joshua R Johnson,2,3 and Rachelle W Johnson1,2,3\\n1\\n\\nProgram in Cancer Biology, Vanderbilt University, Nashville, TN, USA\\nVanderbilt Center for Bone Biology, Department of Medicine, Division of Clinical Pharmacology, Vanderbilt University Medical Center, Nashville,\\nTN, USA\\n3\\nDepartment of Medicine, Division of Clinical Pharmacology, Vanderbilt University Medical Center, Nashville, TN, USA\\n2\\n\\nABSTRACT\\nBreast cancer has a high predilection for spreading to bone with approximately 70% of patients who succumb to disease harboring\\nbone disseminated tumor cells. Despite this high prevalence, treatments for bone metastatic breast cancer predominantly manage\\nmorbidities, including pain and hypercalcemia, rather than reducing bone metastasis incidence or growth. Histone deacetylase inhibitors (HDACi), including panobinostat, entinostat, and valproic acid, typically slow primary tumor progression and are currently in clinical trials for the treatment of many cancers, including primary and metastatic breast cancer, but their effects on bone metastatic\\ndisease have not been examined in preclinical models. We report that treatment with the HDACi panobinostat, but not entinostat\\nor valproic acid, signiﬁcantly reduced trabecular bone volume in tumor-naïve mice, consistent with previous reports of\\nHDACi-induced bone loss. Surprisingly, treatment with entinostat or panobinostat, but not valproic acid, increased tumor burden\\nand incidence in an experimental model of breast cancer bone metastasis. In vitro, multiple HDACi stimulated expression of proosteolytic genes in breast tumor cells, suggesting this may be a mechanism by which HDACi fuel tumor growth. In support of this,\\ncombination therapy of panobinostat or entinostat with the antiresorptive bisphosphonate zoledronic acid prevented bone metastatic progression; however, the addition of zoledronic acid to panobinostat therapy failed to fully correct panobinostat-induced\\nbone loss. Together these data demonstrate that select HDACi fuel bone metastatic growth and provide potential mechanistic\\nand therapeutic avenues to offset these effects. © 2022 The Authors. JBMR Plus published by Wiley Periodicals LLC on behalf of American Society for Bone and Mineral Research.\\nKEY WORDS: BREAST CANCER; HISTONE DEACETYLASE INHIBITORS; BONE METASTASIS\\n\\nIntroduction\\n\\nB\\n\\nreast cancer is the second-leading cause of cancer deaths in\\nwomen predominantly because of the metastatic spread of\\nbreast cancer cells to distant sites. Bone is the most common site\\nof metastasis in patients with breast cancer(1) and occurs in\\napproximately 60% of metastatic estrogen receptor-positive\\n(ER+) patients and 40% of metastatic triple negative breast cancer (TNBC) patients.(2,3) Despite the high prevalence of bone\\nmetastasis, relatively few therapeutic options to effectively target bone metastases exist, and thus metastatic breast cancer\\nremains incurable.\\nGiven the central role of epigenetic modiﬁcations in driving\\ntumor progression and metastasis, there has been signiﬁcant\\ninterest in the development and use of epigenetic therapies for\\n\\ncancer management. Histone deacetylation is a major regulator\\nof abnormal gene expression in tumor cells, and thus histone\\ndeacetylase inhibitors (HDACi) have been heavily pursued for\\ntheir ability to reprogram tumor cells. Several HDACi have been\\napproved by the FDA for hematological malignancies,(4) and\\nmany HDACi are currently being tested in different combination\\ntherapies for their use in solid tumors, including metastatic\\nbreast cancer.\\nOf critical importance to bone metastatic disease, many cancer therapies have been shown to negatively impact bone\\nhealth, resulting in increased risk of skeletal-related events,\\nmorbidities, and mortality.(5) The HDACi valproic acid is\\nFDA-approved for long-term treatment of epilepsy, bipolar disorder, and migraines,(6) but has been shown in several clinical studies to reduce bone mineral density, bone mass, and bone\\n\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,\\nprovided the original work is properly cited.\\nReceived in original form July 12, 2022; revised form September 30, 2022; accepted October 18, 2022.\\nAddress correspondence to: Rachelle W Johnson, PhD, Department of Medicine, Division of Clinical Pharmacology, Vanderbilt University Medical Center, 2215b\\nGarland Avenue, 1165C Medical Research Building IV, Nashville, TN 37232, USA. E-mail: rachelle.johnson@vumc.org\\nJBMR® Plus (WOA), Vol. 7, No. 3, March 2023, e10694.\\nDOI: 10.1002/jbm4.10694\\n© 2022 The Authors. JBMR Plus published by Wiley Periodicals LLC on behalf of American Society for Bone and Mineral Research.\\n\\n1 of 11\\n\\nn\\n\\n\\x0c',\n",
       " 'Original Research\\n23 December 2022\\nDOI 10.3389/fonc.2022.964508\\nTYPE\\n\\nPUBLISHED\\n\\nOPEN ACCESS\\nEDITED BY\\n\\nShaoquan Zheng,\\nThe First Afﬁliated Hospital of Sun\\nYat-sen University, China\\nREVIEWED BY\\n\\nKailin Xu,\\nGuangdong University of Technology,\\nChina\\nYu Cai,\\nJinan University, China\\n*CORRESPONDENCE\\n\\nZhiyu Wang\\nwangzhiyu@gzucm.edu.cn\\nCheng Peng\\npengchengchengdu@126.com\\nNeng Wang\\nellen0000@gzucm.edu.cn\\n†\\nThese authors have contributed\\nequally to this work\\n\\nSPECIALTY SECTION\\n\\nThis article was submitted to\\nMolecular and Cellular Oncology,\\na section of the journal\\nFrontiers in Oncology\\n\\nA pyroptosis-related gene\\nsignature for prognostic and\\nimmunological evaluation in\\nbreast cancer\\nYue Zhong 1†, Fu Peng 2,3†, Xiaoru Luo 1, Xuan Wang 4,5,6,\\nBowen Yang 4,5,6, Xinglinzi Tang 1, Zheng Xu 1, Linlin Ren 1,\\nZhiyu Wang 4,5,6*, Cheng Peng 2* and Neng Wang 1*\\n1\\nIntegrative Medicine Research Center, School of Basic Medical Sciences, Guangzhou\\nUniversity of Chinese Medicine, Guangzhou, Guangdong, China, 2 State Key Laboratory of\\nSouthwestern Chinese Medicine Resources, Chengdu University of Traditional Chinese\\nMedicine, Chengdu, Sichuan, China, 3 Key Laboratory of Drug-Targeting and Drug Delivery System\\nof the Education Ministry and Sichuan Province, West China School of Pharmacy, Sichuan\\nUniversity, Chengdu, China, 4 State Key Laboratory of Dampness Syndrome of Chinese Medicine,\\nThe Second Afﬁliated Hospital of Guangzhou University of Chinese Medicine, Guangzhou,\\nGuangdong, China, 5 Guangdong Provincial Key Laboratory of Clinical Research on Traditional\\nChinese Medicine Syndrome, Guangdong Provincial Academy of Chinese Medical Sciences,\\nGuangdong Provincial Hospital of Chinese Medicine, Guangzhou, Guangdong, China,\\n6\\nGuangdong-Hong Kong-Macau Joint Lab on Chinese Medicine and Immune Disease Research,\\nGuangzhou University of Chinese Medicine, Guangzhou, Guangdong, China\\n\\n08 June 2022\\n21 November 2022\\nPUBLISHED 23 December 2022\\nRECEIVED\\n\\nACCEPTED\\n\\nCITATION\\n\\nZhong Y, Peng F, Luo X, Wang X,\\nYang B, Tang X, Xu Z, Ren L, Wang Z,\\nPeng C and Wang N (2022) A\\npyroptosis-related gene signature for\\nprognostic and immunological\\nevaluation in breast cancer.\\nFront. Oncol. 12:964508.\\ndoi: 10.3389/fonc.2022.964508\\nCOPYRIGHT\\n\\n© 2022 Zhong, Peng, Luo, Wang, Yang,\\nTang, Xu, Ren, Wang, Peng and Wang.\\nThis is an open-access article\\ndistributed under the terms of the\\nCreative Commons Attribution License\\n(CC BY). The use, distribution or\\nreproduction in other forums is\\npermitted, provided the original\\nauthor(s) and the copyright owner(s)\\nare credited and that the original\\npublication in this journal is cited, in\\naccordance with accepted academic\\npractice. No use, distribution or\\nreproduction is permitted which does\\nnot comply with these terms.\\n\\nFrontiers in Oncology\\n\\nPurpose: Pyroptosis exerts an undesirable impact on the clinical outcome of\\nbreast cancer. Since any single gene is insufﬁcient to be an appropriate marker for\\npyroptosis, our aim is to develop a pyroptosis-related gene (PRG) signature to\\npredict the survival status and immunological landscape for breast cancer patients.\\nMethods: The information of breast cancer patients was retrieved from The\\nCancer Genome Atlas (TCGA) and the Gene Expression Omnibus (GEO)\\ndatabases. Quantitative real-time polymerase chain reaction (qRT-PCR) was\\nperformed to verify the gene expressions of this signature in breast cancer. Its\\nprognostic value was evaluated by univariate Cox analysis, least absolute\\nshrinkage and selection operator (LASSO) regression analysis, receiver\\noperating characteristics (ROCs), univariate/multivariate analysis, and\\nnomogram. Analyses of Gene Ontology (GO) and Kyoto Encyclopedia of\\nGenes and Genomes (KEGG) were performed to explore its potential\\nbiological function in breast cancer. The potential correlation between this\\nsignature and tumor immunity was revealed based on single sample gene set\\nenrichment analysis (ssGSEA), ESTIMATE and CIBERSORT algorithms.\\nResults: A PRG signature containing GSDMC, GZMB, IL18, and TP63 was\\ncreated in a TCGA training cohort and validated in two validation GEO\\ncohorts GSE58812 and GSE37751. Compared with a human mammary\\nepithelial cell line MCF-10A, the expression levels of GSDMC, GZMB and IL18\\nwere upregulated, while TP63 was found with lower expression level in breast\\ncancer cells SK-BR-3, BT-549, MCF-7, and MDA-MB-231 using RT-qPCR\\n\\n01\\n\\nfrontiersin.org\\n\\n\\x0c',\n",
       " 'https://www.jsir.or.jp/en/\\n\\nCASE REPORT\\n\\nA Case of Bilateral Internal Mammary Arterial Infusion Chemotherapy Using\\nImplantable Port-catheter Systems for Anterior Chest Wall Metastasis of Breast\\nCancer\\nAtsushi Saiga, Takeshi Aramaki, Rui Sato and Kenji Iwai\\nDivision of Interventional Radiology, Shizuoka Cancer Center, Japan\\n\\nAbstract:\\nThe patient was a 53-year-old woman who presented with a metastatic tumor of the anterior chest wall after chemotherapy for right accessory breast cancer and radiation therapy for right axillary lymphadenopathy. She had already\\nreceived standard treatment, but the metastatic tumor was uncontrollable and exacerbated. Palliative treatment by arterial infusion chemotherapy with an implantable catheter-port system was planned. We indwelled a port system\\nthrough the arteries because the bilateral internal mammary arteries presented a dominant blood supply. The metastatic tumor almost disappeared by arterial infusion chemotherapy after 3 months. This method may be effective in selected cases of arterial infusion chemotherapy for breast cancer.\\nKeywords:\\narterial infusion chemotherapy, breast cancer, implantable port system, internal mammary artery reservoir\\nInterventional Radiology 2023; 8(1): 14-17\\nhttps://doi.org/10.22575/interventionalradiology.2022-0018\\nhttp://interventionalradiology.jp.net/\\n\\nsternal or parasternal areas. We herein report the case of anterior chest wall metastasis from breast cancer treated by bilateral internal mammary AIC using implantable portcatheter systems.\\n\\nIntroduction\\nLocally advanced breast cancer (LABC) includes breast\\ntumors more than 5 cm in size with regional lymphadenopathy, those of any size that involve direct extension to the\\nchest wall or skin, or both, regardless of regional lymphadenopathy, or presence of regional lymphadenopathy (clinically fixed or matted axillary lymph nodes or any of infraclavicular, supraclavicular, or internal mammary lymphadenopathy), regardless of tumor stage [1]. The treatment of\\nLABC is recommended as follows: systemic chemotherapy\\n[2] and local therapy as surgical resection with postoperative\\nradiation [3]. Arterial infusion chemotherapy (AIC) was effective for downstaging or local control in patients with\\nLABC [4-6]. Takizawa et al. have reported AIC under an\\nimplanted catheter-port system after redistribution by internal mammary artery (IMA) embolization. They have called\\nthis method redistributed subclavian infusion chemotherapy (RESAIC) [7]. RESAIC showed good response for patients with LABC who have been previously treated [8].\\nThis procedure is reasonable for advanced primary breast\\ncancer lesion and axillary lymph node metastasis treatments;\\nhowever, it seems unreasonable for midline lesions, such as\\n\\nCase Report\\nA 53-year-old woman presented with a right axillary tumor 7 years earlier. After the diagnosis of right accessory\\nbreast cancer (estrogen receptor: 0%; progesterone receptor:\\n0%; human epidermal growth factor receptor 2: 1+; Ki-67:\\n60%), four cycles of epirubicin plus cyclophosphamide therapy were performed. Since she had a stable response according to RECIST version 1.1 [9], she was admitted to our\\nhospital for further treatment. Paclitaxel plus bevacizumab,\\neribulin, gemcitabine, and capecitabine therapies had been\\nperformed for 5 years. Radiation therapy of 50 Gy was performed because the right axillary lesion was progressive.\\nLocal control was temporally acquired, and she was\\ntreatment-free for 〜1 year after radiation therapy; however,\\na computed tomography (CT) demonstrated a 9-cm huge\\nmetastatic tumor of the anterior chest wall and bilateral\\npleural effusions without pulmonary metastases. RESAIC\\n\\nCorresponding author: Atsushi Saiga, agiasaiga@gmail.com\\nReceived: May 21, 2022, Accepted: October 11, 2022, Advance Publication by J-STAGE: February 9, 2023\\nCopyright Ⓒ The Japanese Society of Interventional Radiology\\n\\n14\\n\\n\\x0c']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "start_time = time.time()\n",
    "headers = list(map(lambda url: url2head(url[\"url\"])[\"htext\"], urls))\n",
    "print(\"--- %s seconds ---\" % (time.time() - start_time))\n",
    "headers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0f08c771-591e-4636-9a8c-226dbb6cd8af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'locations': ['Tanshinones',\n",
       "  'Salvia',\n",
       "  'Biochemistry',\n",
       "  'Pakistan',\n",
       "  'Pharmacy',\n",
       "  'Paris Descartes',\n",
       "  'Paris',\n",
       "  'France',\n",
       "  'Melbourne',\n",
       "  'Parkville',\n",
       "  'Australia',\n",
       "  'Pakistan Pakistan',\n",
       "  'Science',\n",
       "  'Korean Medicine',\n",
       "  'Korean Medicine Seoul',\n",
       "  'South',\n",
       "  'Biochemistry Pakistan',\n",
       "  'Science Korean Medicine',\n",
       "  'Korean Medicine Seoul South Korea',\n",
       "  'Singapore',\n",
       "  'Tanshinones Salvia']}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def text2locations(text):\n",
    "    try:\n",
    "        return post_json_request(\n",
    "            \"http://localhost:5002/text2locations\",\n",
    "            { \"text\": text }\n",
    "        )\n",
    "    except:\n",
    "        return {\"locations\": []}\n",
    "text2locations(\"Exploration of Targeted Anti-tumor Therapy\\nOpen Access Review\\n\\nAn overview of the anti-cancer actions of Tanshinones, derived\\nfrom Salvia miltiorrhiza (Danshen)\\nIrum Naz1, Myriam Merarchi2, Shanaya Ramchandani3, Muhammad Rashid Khan4*, Muhammad Nouman\\nMalik1, Sumaira Sarwar1, Acharan S Narula5, Kwang Seok Ahn6*\\nDepartment of Biochemistry, Faculty of Biological Sciences, Quaid-i-Azam University, Islamabad 45320, Pakistan\\n\\n1\\n\\nFaculty of Pharmacy, University of Paris Descartes, 75006 Paris, France\\n\\n2\\n\\nDepartment of Pharmacology-Biomedicine, The University of Melbourne, Parkville, VIC 3010, Australia\\n\\n3\\n\\nHigher Education Commission of Pakistan, Islamabad 44000, Pakistan\\n\\n4\\n\\nNarula Research, Chapel Hill, NC 27516, USA\\n\\n5\\n\\nDepartment of Science in Korean Medicine, College of Korean Medicine, Kyung Hee University, 24 Kyungheedae-ro,\\nDongdaemun-gu, Seoul 02447, South Korea\\n6\\n\\n*Correspondence: Muhammad Rashid Khan, Department of Biochemistry, Faculty of Biological Sciences, Quaid-i-Azam\\nUniversity, Islamabad 45320, Pakistan. mrkhanqau@yahoo.com; Kwang Seok Ahn, Department of Science in Korean Medicine,\\nCollege of Korean Medicine, Kyung Hee University, 24 Kyungheedae-ro, Dongdaemun-gu, Seoul 02447, South Korea. ksahn@\\nkhu.ac.kr\\nAcademic Editor: Gautam Sethi, National University of Singapore, Singapore\\nReceived: April 16, 2020 Accepted: May 17, 2020 Published: June 29, 2020\\n\\nCite this article: Naz I, Merarchi M, Ramchandani S, Khan MR, Malik MN, Sarwar S, et al. An overview of the anti-cancer\\nactions of Tanshinones, derived from Salvia miltiorrhiza (Danshen). Explor Target Antitumor Ther. 2020;1:153-70. https://doi.\\norg/10.37349/etat.2020.00010\\n\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "935b9eff-4d18-47e0-a76b-7480f79d3857",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 5.018128871917725 seconds ---\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[['Cellular',\n",
       "  'Author',\n",
       "  'Athens',\n",
       "  'Athens Greece',\n",
       "  'Biophysics',\n",
       "  'Biology',\n",
       "  'Athens Athens',\n",
       "  'Greece',\n",
       "  'Cancer',\n",
       "  'Genetics',\n",
       "  'Columbus',\n",
       "  'Columbus Pathology',\n",
       "  'Athens Athens Greece'],\n",
       " ['Bayesian',\n",
       "  'Mathematics',\n",
       "  'Hong Kong',\n",
       "  'Hong Kong China',\n",
       "  'Mathematics Shenzhen',\n",
       "  'China',\n",
       "  'Life',\n",
       "  'Northeastern University',\n",
       "  'China China',\n",
       "  'Chengdu',\n",
       "  'China Ching',\n",
       "  'Bayesian UNITED STATES',\n",
       "  'China Bayesian'],\n",
       " ['Author',\n",
       "  'New Zealand',\n",
       "  'Epidemiology',\n",
       "  'Biostatistics',\n",
       "  'Auckland',\n",
       "  'Auckland New',\n",
       "  'Oncology',\n",
       "  'Oncology Hamilton',\n",
       "  'New',\n",
       "  'NIDEA',\n",
       "  'Demographic',\n",
       "  'Waikato',\n",
       "  'Hamilton'],\n",
       " ['Embase',\n",
       "  'Nursing',\n",
       "  'English',\n",
       "  'Canada',\n",
       "  'Australia',\n",
       "  'Lebanon',\n",
       "  'England',\n",
       "  'Dublin',\n",
       "  'Ireland',\n",
       "  'Dublin Ireland',\n",
       "  'Ireland Irish'],\n",
       " ['England',\n",
       "  'Scotland',\n",
       "  'Mean',\n",
       "  'London',\n",
       "  'Nutrition',\n",
       "  'Health',\n",
       "  'Medicine',\n",
       "  'Brazil',\n",
       "  'Cancer',\n",
       "  'France'],\n",
       " ['Cancer',\n",
       "  'U.S.',\n",
       "  'US',\n",
       "  'Guam',\n",
       "  'Guam Hawai',\n",
       "  'Guam Filipino',\n",
       "  'Guam Filipino Guam',\n",
       "  'Mangilao',\n",
       "  'Honolulu',\n",
       "  'Medicine'],\n",
       " ['AR', 'Author'],\n",
       " ['Author',\n",
       "  'Medicine',\n",
       "  'Tanta',\n",
       "  'Egypt',\n",
       "  'Medicine Tanta',\n",
       "  'Egypt Medicine',\n",
       "  'Tanta Egypt',\n",
       "  'Medicine Tanta Egypt'],\n",
       " ['European',\n",
       "  'Author',\n",
       "  'Strasbourg',\n",
       "  'France',\n",
       "  'European Exercise',\n",
       "  'Strasbourg France',\n",
       "  'Cancerology',\n",
       "  'Europe'],\n",
       " ['China',\n",
       "  'Chinese',\n",
       "  'Sciences',\n",
       "  'China Biomaterials',\n",
       "  'Chinese China',\n",
       "  'Fig',\n",
       "  'Fig Fig',\n",
       "  'U.S.'],\n",
       " ['China', 'China Science', 'China China', 'Sciences', 'Prague', 'China U.S.'],\n",
       " ['Brazil',\n",
       "  'Health Sciences',\n",
       "  'Goiás',\n",
       "  'Goiânia',\n",
       "  'Brazil Goiás',\n",
       "  'Goiânia Brazil',\n",
       "  'Goiânia Brazil Brazil',\n",
       "  'Brazil Brazil'],\n",
       " ['Adult',\n",
       "  'Md',\n",
       "  'Amarillo',\n",
       "  'Rheumatology',\n",
       "  'Policy',\n",
       "  'Baltimore',\n",
       "  'Brooklyn',\n",
       "  'Bangladesh',\n",
       "  'Medicine',\n",
       "  'Bangladesh Adult'],\n",
       " ['Obstetrics',\n",
       "  'Luzhou',\n",
       "  'Luzhou Rawalpindi',\n",
       "  'Hematology',\n",
       "  'Luzhou Rawalpindi Pathology',\n",
       "  'Luzhou Radiology',\n",
       "  'Luzhou Oncology',\n",
       "  'China',\n",
       "  'Burkitt'],\n",
       " ['Diabetes', 'Health', 'Bethesda', 'U.S.A'],\n",
       " ['Malaysia',\n",
       "  'Malaysia Malaysia',\n",
       "  'Malaysia Chengdu',\n",
       "  'China',\n",
       "  'Pharmacology',\n",
       "  'Xiao',\n",
       "  'US',\n",
       "  'United States'],\n",
       " ['Author', 'Gainesville', 'United States', 'Gainesville United States'],\n",
       " ['Young',\n",
       "  'Medicine',\n",
       "  'Medicine Medan',\n",
       "  'Indonesia',\n",
       "  'Anatomy',\n",
       "  'Medicine Medan Indonesia',\n",
       "  'Neurosurgery',\n",
       "  'Medicine Medan Indonesia Surgery',\n",
       "  'Medicine Medan Indonesia Medicine',\n",
       "  'Medan',\n",
       "  'Indonesia Dermatology',\n",
       "  'Venereology',\n",
       "  'Medicine Medan Indonesia Ear',\n",
       "  'Throat',\n",
       "  'Medicine Medan Indonesia Anatomy',\n",
       "  'Medicine Indonesia',\n",
       "  'Delyuzar',\n",
       "  'Asrul'],\n",
       " ['Chemical',\n",
       "  'Chemistry',\n",
       "  'Toronto',\n",
       "  'Toronto Canada',\n",
       "  'Tel',\n",
       "  'Toronto Toronto',\n",
       "  'Toronto Toronto Canada',\n",
       "  'Toronto Canada Environment',\n",
       "  'Toronto Toronto Beijing',\n",
       "  'Technology',\n",
       "  'Beijing',\n",
       "  'China',\n",
       "  'Beijing Beijing',\n",
       "  'Technology Beijing',\n",
       "  'China Saskatchewan',\n",
       "  'Canada',\n",
       "  'Hunan Province',\n",
       "  'Hunan',\n",
       "  'China Author'],\n",
       " ['Synthesis',\n",
       "  'Science',\n",
       "  'Alexandria',\n",
       "  'Egypt',\n",
       "  'Zoology',\n",
       "  'Science Ismailia',\n",
       "  'Egypt Pharmacy',\n",
       "  'Cairo',\n",
       "  'Egypt Pharmacy Egypt',\n",
       "  'Ismailia',\n",
       "  'Egypt Biochemistry',\n",
       "  'Science Cairo',\n",
       "  'Egypt Science',\n",
       "  'Alexandria Egypt',\n",
       "  'Pharmacy',\n",
       "  'Cairo Science',\n",
       "  'New Borg',\n",
       "  'Alexandria Egypt Thalidomide',\n",
       "  'License'],\n",
       " ['West Lafayette', 'West Lafayette West Lafayette', 'West Lafayette Author'],\n",
       " ['Australia',\n",
       "  'Oncology',\n",
       "  'Zoology',\n",
       "  'Science',\n",
       "  'Egypt',\n",
       "  'Anatomy',\n",
       "  'Medicine',\n",
       "  'Singapore',\n",
       "  'Singapore Singapore',\n",
       "  'Obstetrics',\n",
       "  'Germany'],\n",
       " ['Nigeria',\n",
       "  'China',\n",
       "  'Surgery',\n",
       "  'Sweden',\n",
       "  'Malmö',\n",
       "  'Sweden Oncology',\n",
       "  'Sweden Sweden',\n",
       "  'Sweden Surgery',\n",
       "  'Malmö Skarping',\n",
       "  'Hjärtström'],\n",
       " ['Parma',\n",
       "  'Italy',\n",
       "  'Italy Medicine',\n",
       "  'Fujian',\n",
       "  'Fuzhou',\n",
       "  'Fujian China',\n",
       "  'Fuzhou Fujian',\n",
       "  'China',\n",
       "  'Fuzhou Fujian China',\n",
       "  'Fuzhou Fujian China Medicine',\n",
       "  'Xiamen',\n",
       "  'Fujian China Fu',\n",
       "  'Oncology',\n",
       "  'Epidemiology',\n",
       "  'Fujian ER',\n",
       "  'PR',\n",
       "  'T',\n",
       "  'N'],\n",
       " ['China',\n",
       "  'Romagna',\n",
       "  'Tumors',\n",
       "  'Italy',\n",
       "  'China China',\n",
       "  'Anesthesiology',\n",
       "  'China Du'],\n",
       " ['West Attica',\n",
       "  'Greece',\n",
       "  'Athens',\n",
       "  'Greece Cambridge',\n",
       "  'United',\n",
       "  'Chinese',\n",
       "  'Beijing',\n",
       "  'China',\n",
       "  'Chinese Beijing',\n",
       "  'China Zhai',\n",
       "  'Lv',\n",
       "  'Oncology',\n",
       "  'Sindex',\n",
       "  'Sindex Oncopredict',\n",
       "  'Sindex China'],\n",
       " ['Ospedale',\n",
       "  'Italy',\n",
       "  'Medicine',\n",
       "  'Mexico',\n",
       "  'Radiology',\n",
       "  'Shenzhen',\n",
       "  'China',\n",
       "  'China Radiology',\n",
       "  'Kunming',\n",
       "  'China Radiology Shenzhen',\n",
       "  'China Yu',\n",
       "  'Hu',\n",
       "  'ADCmean'],\n",
       " ['Hong Kong SAR',\n",
       "  'China',\n",
       "  'Crete',\n",
       "  'Greece',\n",
       "  'Molecular',\n",
       "  'Engineering',\n",
       "  'Chengdu',\n",
       "  'China West China Hospital',\n",
       "  'Biotherapy',\n",
       "  'Chengdu China',\n",
       "  'Education',\n",
       "  'China United States',\n",
       "  'Shenzhen',\n",
       "  'China Molecular'],\n",
       " ['Breast', 'Asian', 'Singapore', 'California', 'Irvine', 'Breast Asian'],\n",
       " ['Oncology',\n",
       "  'Serbia',\n",
       "  'Toruń',\n",
       "  'Poland',\n",
       "  'Italy',\n",
       "  'Genetics',\n",
       "  'Biostatistics',\n",
       "  'California',\n",
       "  'Los Angeles',\n",
       "  'Los Angeles United States',\n",
       "  'Honolulu',\n",
       "  'United States',\n",
       "  'Epidemiology',\n",
       "  'California Los Angeles',\n",
       "  'Los Angeles United States California',\n",
       "  'Los Angeles Los Angeles',\n",
       "  'United States Nursing',\n",
       "  'California Los Angeles Los Angeles',\n",
       "  'United States California',\n",
       "  'Los Angeles Los Angeles United States',\n",
       "  'Genetics Tumor'],\n",
       " ['Science',\n",
       "  'China',\n",
       "  'China Stomatology',\n",
       "  'China Oncogenomics',\n",
       "  'Stomatology',\n",
       "  'Hunan',\n",
       "  'China Endocrinology',\n",
       "  'Hunan China',\n",
       "  'Stomatology China',\n",
       "  'Chengdu',\n",
       "  'China Su',\n",
       "  'Chen',\n",
       "  'Hu',\n",
       "  'Zhong',\n",
       "  'Genetics',\n",
       "  'HNSC',\n",
       "  'HNSC Thyroid',\n",
       "  'Bladder',\n",
       "  'GSTK1'],\n",
       " ['United States',\n",
       "  'China',\n",
       "  'China Pathology',\n",
       "  'Nantong',\n",
       "  'China Oncology',\n",
       "  'Shanghai',\n",
       "  'China Oncogenomics',\n",
       "  'Qian',\n",
       "  'Lu',\n",
       "  'R',\n",
       "  'Genes',\n",
       "  'Genes Genomes'],\n",
       " ['Cancer Biology',\n",
       "  'Nashville',\n",
       "  'Medicine',\n",
       "  'Nashville Medicine',\n",
       "  'Nashville American',\n",
       "  'Medicine Nashville',\n",
       "  'No',\n",
       "  'American'],\n",
       " ['China',\n",
       "  'Technology',\n",
       "  'China China',\n",
       "  'Oncology',\n",
       "  'Chinese Medicine',\n",
       "  'Guangdong',\n",
       "  'China Chengdu',\n",
       "  'China West China',\n",
       "  'Pharmacy',\n",
       "  'Chengdu',\n",
       "  'China Chinese Medicine',\n",
       "  'Chinese Medicine Guangdong',\n",
       "  'China Chinese',\n",
       "  'Chinese Medicine Guangdong China',\n",
       "  'Chinese Medicine Chinese Medicine',\n",
       "  'Guangdong China',\n",
       "  'Zhong',\n",
       "  'Xu',\n",
       "  'Oncology Genes',\n",
       "  'Genomes'],\n",
       " ['Japan', 'Japanese']]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "start_time = time.time()\n",
    "locations = list(map(lambda text: text2locations(text)[\"locations\"], headers))\n",
    "print(\"--- %s seconds ---\" % (time.time() - start_time))\n",
    "locations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "1e7a9755-7440-4c11-89eb-258784bb83d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'places': [['Pakistan', 4],\n",
       "  ['Singapore', 2],\n",
       "  ['France', 1],\n",
       "  ['Australia', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['South Korea', 1],\n",
       "  ['Canada', 1],\n",
       "  ['India', 1],\n",
       "  ['Sweden', 1],\n",
       "  ['New Zealand', 1],\n",
       "  ['Denmark', 1],\n",
       "  ['Indonesia', 1]]}"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def text2places(text):\n",
    "    try:\n",
    "        return post_json_request(\n",
    "            \"http://localhost:5002/text2places\",\n",
    "            { \"text\": text }\n",
    "        )\n",
    "    except:\n",
    "        return {\"places\": []}\n",
    "text2places(\"Exploration of Targeted Anti-tumor Therapy\\nOpen Access Review\\n\\nAn overview of the anti-cancer actions of Tanshinones, derived\\nfrom Salvia miltiorrhiza (Danshen)\\nIrum Naz1, Myriam Merarchi2, Shanaya Ramchandani3, Muhammad Rashid Khan4*, Muhammad Nouman\\nMalik1, Sumaira Sarwar1, Acharan S Narula5, Kwang Seok Ahn6*\\nDepartment of Biochemistry, Faculty of Biological Sciences, Quaid-i-Azam University, Islamabad 45320, Pakistan\\n\\n1\\n\\nFaculty of Pharmacy, University of Paris Descartes, 75006 Paris, France\\n\\n2\\n\\nDepartment of Pharmacology-Biomedicine, The University of Melbourne, Parkville, VIC 3010, Australia\\n\\n3\\n\\nHigher Education Commission of Pakistan, Islamabad 44000, Pakistan\\n\\n4\\n\\nNarula Research, Chapel Hill, NC 27516, USA\\n\\n5\\n\\nDepartment of Science in Korean Medicine, College of Korean Medicine, Kyung Hee University, 24 Kyungheedae-ro,\\nDongdaemun-gu, Seoul 02447, South Korea\\n6\\n\\n*Correspondence: Muhammad Rashid Khan, Department of Biochemistry, Faculty of Biological Sciences, Quaid-i-Azam\\nUniversity, Islamabad 45320, Pakistan. mrkhanqau@yahoo.com; Kwang Seok Ahn, Department of Science in Korean Medicine,\\nCollege of Korean Medicine, Kyung Hee University, 24 Kyungheedae-ro, Dongdaemun-gu, Seoul 02447, South Korea. ksahn@\\nkhu.ac.kr\\nAcademic Editor: Gautam Sethi, National University of Singapore, Singapore\\nReceived: April 16, 2020 Accepted: May 17, 2020 Published: June 29, 2020\\n\\nCite this article: Naz I, Merarchi M, Ramchandani S, Khan MR, Malik MN, Sarwar S, et al. An overview of the anti-cancer\\nactions of Tanshinones, derived from Salvia miltiorrhiza (Danshen). Explor Target Antitumor Ther. 2020;1:153-70. https://doi.\\norg/10.37349/etat.2020.00010\\n\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "888559a8-2111-45d3-b326-f305983367e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 163.43850755691528 seconds ---\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[[['Greece', 4], ['United States of America', 2], ['Canada', 1], ['Spain', 1]],\n",
       " [[\"People's Republic of China\", 6],\n",
       "  ['United States of America', 1],\n",
       "  ['Japan', 1],\n",
       "  ['India', 1],\n",
       "  ['Mexico', 1]],\n",
       " [['New Zealand', 2],\n",
       "  ['Denmark', 1],\n",
       "  ['Canada', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['United Kingdom', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Kazakhstan', 1],\n",
       "  ['France', 1],\n",
       "  ['India', 1]],\n",
       " [['Ireland', 3],\n",
       "  ['United States of America', 1],\n",
       "  ['Canada', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Lebanon', 1],\n",
       "  ['Vietnam', 1],\n",
       "  ['Belarus', 1],\n",
       "  ['Netherlands', 1],\n",
       "  ['Ukraine', 1]],\n",
       " [['United Kingdom', 2],\n",
       "  ['Brazil', 1],\n",
       "  ['France', 1],\n",
       "  ['Cameroon', 1],\n",
       "  ['Canada', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Serbia', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Hungary', 1],\n",
       "  ['Portugal', 1],\n",
       "  ['Indonesia', 1],\n",
       "  ['Ukraine', 1]],\n",
       " [['United States of America', 4], ['India', 1], ['Spain', 1], ['Japan', 1]],\n",
       " [['Argentina', 10]],\n",
       " [['Egypt', 4],\n",
       "  ['United States of America', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Indonesia', 1]],\n",
       " [['France', 3],\n",
       "  ['United States of America', 1],\n",
       "  ['Brazil', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Canada', 1],\n",
       "  [\"People's Republic of China\", 1],\n",
       "  ['Belgium', 1]],\n",
       " [[\"People's Republic of China\", 3],\n",
       "  ['United States of America', 1],\n",
       "  ['Japan', 1],\n",
       "  ['India', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Albania', 1]],\n",
       " [[\"People's Republic of China\", 5],\n",
       "  ['Czech Republic', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Japan', 1],\n",
       "  ['India', 1],\n",
       "  ['Mexico', 1]],\n",
       " [['Brazil', 9],\n",
       "  ['United States of America', 1],\n",
       "  ['France', 1],\n",
       "  ['Portugal', 1],\n",
       "  ['Oman', 1]],\n",
       " [['United States of America', 3],\n",
       "  ['Bangladesh', 3],\n",
       "  ['Moldova', 1],\n",
       "  ['United Kingdom', 1],\n",
       "  ['India', 1],\n",
       "  ['New Zealand', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Uruguay', 1],\n",
       "  ['South Africa', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Canada', 1]],\n",
       " [[\"People's Republic of China\", 1],\n",
       "  ['Pakistan', 1],\n",
       "  ['Japan', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['India', 1],\n",
       "  ['Mexico', 1]],\n",
       " [['Thailand', 2],\n",
       "  ['Moldova', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Antigua and Barbuda', 1],\n",
       "  ['United Kingdom', 1]],\n",
       " [['Malaysia', 4],\n",
       "  ['United States of America', 2],\n",
       "  [\"People's Republic of China\", 1],\n",
       "  ['Liechtenstein', 1],\n",
       "  ['Japan', 1],\n",
       "  ['Germany', 1],\n",
       "  ['Austria', 1],\n",
       "  ['India', 1],\n",
       "  ['Norway', 1],\n",
       "  ['Mexico', 1]],\n",
       " [['Gabon', 2], ['United States of America', 2], ['Italy', 1]],\n",
       " [['Indonesia', 8],\n",
       "  ['Uruguay', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Japan', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Canada', 1]],\n",
       " [['Canada', 6],\n",
       "  [\"People's Republic of China\", 3],\n",
       "  ['Slovakia', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Japan', 1],\n",
       "  ['Australia', 1],\n",
       "  ['India', 1],\n",
       "  ['Philippines', 1],\n",
       "  ['Greece', 1],\n",
       "  ['Azerbaijan', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Netherlands', 1],\n",
       "  ['Ukraine', 1]],\n",
       " [['Egypt', 8],\n",
       "  ['Iran', 1],\n",
       "  ['Niger', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Romania', 1],\n",
       "  ['Brazil', 1],\n",
       "  ['Australia', 1],\n",
       "  ['United Kingdom', 1],\n",
       "  ['South Africa', 1],\n",
       "  ['Italy', 1],\n",
       "  ['Canada', 1]],\n",
       " [['United States of America', 3], ['Estonia', 1], ['Russia', 1]],\n",
       " [['Singapore', 3],\n",
       "  ['Australia', 1],\n",
       "  ['Egypt', 1],\n",
       "  ['Germany', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Switzerland', 1]],\n",
       " [['Sweden', 6],\n",
       "  ['Nigeria', 1],\n",
       "  [\"People's Republic of China\", 1],\n",
       "  ['South Korea', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Japan', 1],\n",
       "  ['Norway', 1],\n",
       "  ['Romania', 1],\n",
       "  ['India', 1],\n",
       "  ['Iran', 1],\n",
       "  ['Germany', 1],\n",
       "  ['Serbia', 1],\n",
       "  ['Denmark', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Italy', 1]],\n",
       " [[\"People's Republic of China\", 5],\n",
       "  ['Italy', 2],\n",
       "  ['Vietnam', 1],\n",
       "  ['Eritrea', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Japan', 1],\n",
       "  ['Turkey', 1],\n",
       "  ['India', 1],\n",
       "  ['Russia', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Estonia', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Indonesia', 1],\n",
       "  ['Albania', 1],\n",
       "  ['Belgium', 1]],\n",
       " [[\"People's Republic of China\", 4],\n",
       "  ['Italy', 1],\n",
       "  ['Liechtenstein', 1],\n",
       "  ['Japan', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['India', 1],\n",
       "  ['Norway', 1],\n",
       "  ['Hungary', 1],\n",
       "  ['Denmark', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Netherlands', 1],\n",
       "  ['Indonesia', 1],\n",
       "  ['Ukraine', 1]],\n",
       " [[\"People's Republic of China\", 3],\n",
       "  ['Greece', 2],\n",
       "  ['Liechtenstein', 1],\n",
       "  ['Latvia', 1],\n",
       "  ['Morocco', 1],\n",
       "  ['Canada', 1],\n",
       "  ['United Kingdom', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['New Zealand', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Japan', 1],\n",
       "  ['Turkey', 1],\n",
       "  ['Kazakhstan', 1],\n",
       "  ['Australia', 1],\n",
       "  ['India', 1],\n",
       "  ['Norway', 1],\n",
       "  ['France', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Belgium', 1]],\n",
       " [[\"People's Republic of China\", 4],\n",
       "  ['Italy', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Liechtenstein', 1],\n",
       "  ['Hungary', 1],\n",
       "  ['Philippines', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Cyprus', 1],\n",
       "  ['Japan', 1],\n",
       "  ['India', 1],\n",
       "  ['Norway', 1],\n",
       "  ['Estonia', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Iran', 1],\n",
       "  ['Indonesia', 1],\n",
       "  ['Albania', 1]],\n",
       " [[\"People's Republic of China\", 5],\n",
       "  ['Greece', 1],\n",
       "  ['Panama', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Thailand', 1],\n",
       "  ['Japan', 1],\n",
       "  ['Germany', 1],\n",
       "  ['Austria', 1],\n",
       "  ['India', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Albania', 1],\n",
       "  ['Italy', 1],\n",
       "  ['Iran', 1],\n",
       "  ['Oman', 1],\n",
       "  ['Egypt', 1],\n",
       "  ['South Africa', 1],\n",
       "  ['Turkey', 1]],\n",
       " [['Singapore', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['United Kingdom', 1],\n",
       "  ['Philippines', 1],\n",
       "  ['Colombia', 1],\n",
       "  ['Italy', 1],\n",
       "  ['El Salvador', 1],\n",
       "  ['Mexico', 1]],\n",
       " [['United States of America', 6],\n",
       "  ['Canada', 5],\n",
       "  ['Serbia', 1],\n",
       "  ['Poland', 1],\n",
       "  ['Italy', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Kazakhstan', 1],\n",
       "  ['Philippines', 1],\n",
       "  ['United Kingdom', 1],\n",
       "  ['Sweden', 1],\n",
       "  ['France', 1],\n",
       "  ['Australia', 1],\n",
       "  ['Colombia', 1],\n",
       "  [\"People's Republic of China\", 1],\n",
       "  ['Kiribati', 1],\n",
       "  ['Russia', 1],\n",
       "  ['El Salvador', 1],\n",
       "  ['Mexico', 1]],\n",
       " [[\"People's Republic of China\", 7],\n",
       "  ['Yemen', 1],\n",
       "  ['Hungary', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Japan', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Turkey', 1],\n",
       "  ['Kazakhstan', 1],\n",
       "  ['India', 1],\n",
       "  ['Philippines', 1],\n",
       "  ['France', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Indonesia', 1],\n",
       "  ['Belgium', 1]],\n",
       " [[\"People's Republic of China\", 4],\n",
       "  ['United States of America', 1],\n",
       "  ['Luxembourg', 1],\n",
       "  ['Spain', 1],\n",
       "  ['Japan', 1],\n",
       "  ['Kazakhstan', 1],\n",
       "  ['Italy', 1],\n",
       "  ['India', 1],\n",
       "  ['France', 1],\n",
       "  ['Estonia', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Indonesia', 1]],\n",
       " [['Tunisia', 3],\n",
       "  ['United States of America', 3],\n",
       "  ['Norway', 1],\n",
       "  ['Japan', 1],\n",
       "  ['Denmark', 1],\n",
       "  ['Indonesia', 1]],\n",
       " [[\"People's Republic of China\", 9],\n",
       "  ['Japan', 1],\n",
       "  ['United States of America', 1],\n",
       "  ['Germany', 1],\n",
       "  ['Austria', 1],\n",
       "  ['India', 1],\n",
       "  ['Netherlands', 1],\n",
       "  ['Mexico', 1],\n",
       "  ['Iran', 1],\n",
       "  ['Indonesia', 1]],\n",
       " [['Japan', 1], ['Indonesia', 1], ['United States of America', 1]]]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "start_time = time.time()\n",
    "places = list(map(lambda text: text2places(text)[\"places\"], headers))\n",
    "print(\"--- %s seconds ---\" % (time.time() - start_time))\n",
    "places"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "2c947bc1-53d8-4a48-b7fb-ba7bebe3a4f2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['South Africa', 'Canada', 'United States of America', 'Turkey', 'United Kingdom', 'Czech Republic', 'Japan', 'Australia', 'Spain', 'Tanzania', 'Norway', 'France', 'Italy', 'East Timor', 'Philippines', 'New Zealand', 'Sweden', 'Indonesia', 'Serbia', 'Mexico', 'Guyana', 'Brazil', 'Libya', 'Argentina', 'Netherlands', 'Mali', 'Pakistan', 'Ukraine', 'Bosnia and Herzegovina']\n",
      "[('Canada', 3), ('South Africa', 2), ('United States of America', 1), ('Turkey', 1), ('United Kingdom', 1), ('Czech Republic', 1), ('Japan', 1), ('Australia', 1), ('Spain', 1), ('Tanzania', 1), ('Norway', 1), ('France', 1), ('Italy', 1), ('East Timor', 1), ('Philippines', 1), ('New Zealand', 1), ('Sweden', 1), ('Indonesia', 1), ('Serbia', 1), ('Mexico', 1), ('Guyana', 1), ('Brazil', 1), ('Libya', 1), ('Argentina', 1), ('Netherlands', 1), ('Mali', 1), ('Pakistan', 1), ('Ukraine', 1), ('Bosnia and Herzegovina', 1)]\n",
      "[('Baker', 2), ('Liverpool', 1), ('Eastern', 1), ('Briton', 1), ('British', 1), ('Yuan', 1), ('Adelbert', 1), ('Queen', 1), ('Johnson', 1), ('Woman', 1), ('Victorian', 1), ('Montmartre', 1), ('Continental', 1), ('Miss Violet', 1), ('Mane', 1), ('South', 1), ('Miss de Merville', 1), ('Southampton', 1), ('America', 1), ('South Africa', 1), ('Miss Winter', 1), ('Alas', 1), ('Mazarin Stone', 1), ('Baron Gruner', 1), ('Baron', 1)]\n",
      "[('Watson', 37), ('Holmes', 29), ('Godfrey', 23), ('Adventure', 11), ('Baron', 11), ('Ralph', 10), ('London', 6), ('Merville', 6), ('British', 4), ('General', 4), ('Johnson', 4), ('Dodd', 4), ('Canada', 3), ('England', 3), ('Canadian', 2), ('Sherlock', 2), ('Kingston', 2), ('Strand', 2), ('Eastern', 2), ('English', 2), ('Pretoria', 2), ('Boer', 2), ('Dutch', 2), ('Doyle', 1), ('Date', 1), ('Richardson', 1), ('Scott', 1), ('Dickens', 1), ('Valhalla', 1), ('Scarlet', 1), ('Four', 1), ('Bohemia', 1), ('Edward', 1), ('Lion', 1), ('Mane', 1), ('Damery', 1), ('Queen', 1), ('Sharp', 1), ('Continental', 1), ('Prague', 1), ('Hurlingham', 1), ('Wainwright', 1), ('Parkhurst', 1), ('Simpson', 1), ('Violet', 1), ('Montmartre', 1), ('Hell', 1), ('Same', 1), ('Berkeley Square', 1), ('Baker', 1), ('Lucky', 1), ('Cunard', 1), ('Liverpool', 1), ('James', 1), ('Lomax', 1), ('Christie', 1), ('Imperial', 1), ('South', 1), ('Pray', 1), ('Oriental', 1), ('Nara', 1), ('Yard', 1), ('America', 1), ('Middlesex', 1), ('Southampton', 1), ('Bedford', 1), ('Africa', 1), ('Duke', 1), ('Sultan', 1), ('Turkey', 1), ('Euston', 1), ('Alas', 1), ('Tell', 1), ('Emsworth', 1), ('Anderson', 1)]\n"
     ]
    }
   ],
   "source": [
    "import geograpy\n",
    "url = 'http://www.gutenberg.ca/ebooks/doyleac-casebookofsherlockholmes/doyleac-casebookofsherlockholmes-00-h.html'\n",
    "places = geograpy.get_place_context(url=url)\n",
    "print(places.countries)\n",
    "print(places.country_mentions)\n",
    "print(places.region_mentions)\n",
    "print(places.city_mentions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f3ab40c6-2e88-47a9-a1a2-7bbb787f538e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Brazil', 'United Kingdom', 'France', 'Cameroon', 'Canada', 'United States of America', 'Serbia', 'Spain', 'Hungary', 'Portugal', 'Indonesia', 'Ukraine']\n",
      "[('United Kingdom', 2), ('Brazil', 1), ('France', 1), ('Cameroon', 1), ('Canada', 1), ('United States of America', 1), ('Serbia', 1), ('Spain', 1), ('Hungary', 1), ('Portugal', 1), ('Indonesia', 1), ('Ukraine', 1)]\n",
      "United Kingdom\n",
      "[('Center', 3), ('São Paulo', 1), ('Wales', 1), ('England', 1), ('Scotland', 1)]\n",
      "[('São Paulo', 2), ('England', 1), ('Scotland', 1), ('Wales', 1), ('London', 1), ('Center', 1), ('Health', 1), ('University', 1), ('Brazil', 1), ('Cancer', 1), ('Lyon', 1)]\n"
     ]
    }
   ],
   "source": [
    "import geograpy\n",
    "text = headers[4]\n",
    "places = geograpy.get_place_context(text=text)\n",
    "print(places.countries)\n",
    "print(places.country_mentions)\n",
    "print(places.country_mentions[0][0])\n",
    "print(places.region_mentions)\n",
    "print(places.city_mentions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7b3e873-cbfe-48bd-b9c5-45335aec3d4c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "9eec4d3c-c5ab-450b-91c0-4a3fe54e2f0a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'entities': [['Exploration', 'NN'],\n",
       "  ['of', 'IN'],\n",
       "  [['Targeted', 'NNP']],\n",
       "  ['Anti-tumor', 'NNP'],\n",
       "  ['Therapy', 'NNP'],\n",
       "  ['Open', 'NNP'],\n",
       "  ['Access', 'NNP'],\n",
       "  ['Review', 'NNP'],\n",
       "  ['An', 'DT'],\n",
       "  ['overview', 'NN'],\n",
       "  ['of', 'IN'],\n",
       "  ['the', 'DT'],\n",
       "  ['anti-cancer', 'JJ'],\n",
       "  ['actions', 'NNS'],\n",
       "  ['of', 'IN'],\n",
       "  [['Tanshinones', 'NNP']],\n",
       "  [',', ','],\n",
       "  ['derived', 'VBD'],\n",
       "  ['from', 'IN'],\n",
       "  [['Salvia', 'NNP']],\n",
       "  ['miltiorrhiza', 'NN'],\n",
       "  ['(', '('],\n",
       "  [['Danshen', 'NNP']],\n",
       "  [')', ')'],\n",
       "  [['Irum', 'NNP'], ['Naz1', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Myriam', 'NNP'], ['Merarchi2', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Shanaya', 'NNP'], ['Ramchandani3', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Muhammad', 'NNP'], ['Rashid', 'NNP'], ['Khan4', 'NNP']],\n",
       "  ['*', 'NNP'],\n",
       "  [',', ','],\n",
       "  [['Muhammad', 'NNP'], ['Nouman', 'NNP'], ['Malik1', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Sumaira', 'NNP'], ['Sarwar1', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Acharan', 'NNP'], ['S', 'NNP'], ['Narula5', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Kwang', 'NNP'], ['Seok', 'NNP'], ['Ahn6', 'NNP']],\n",
       "  ['*', 'NNP'],\n",
       "  ['Department', 'NNP'],\n",
       "  ['of', 'IN'],\n",
       "  [['Biochemistry', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Faculty', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Biological', 'NNP'], ['Sciences', 'NNPS']],\n",
       "  [',', ','],\n",
       "  ['Quaid-i-Azam', 'NNP'],\n",
       "  ['University', 'NNP'],\n",
       "  [',', ','],\n",
       "  [['Islamabad', 'NNP']],\n",
       "  ['45320', 'CD'],\n",
       "  [',', ','],\n",
       "  [['Pakistan', 'NNP']],\n",
       "  ['1', 'CD'],\n",
       "  ['Faculty', 'NNP'],\n",
       "  ['of', 'IN'],\n",
       "  [['Pharmacy', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['University', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Paris', 'NNP'], ['Descartes', 'NNP']],\n",
       "  [',', ','],\n",
       "  ['75006', 'CD'],\n",
       "  [['Paris', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['France', 'NNP']],\n",
       "  ['2', 'CD'],\n",
       "  ['Department', 'NNP'],\n",
       "  ['of', 'IN'],\n",
       "  ['Pharmacology-Biomedicine', 'NNP'],\n",
       "  [',', ','],\n",
       "  [['The', 'DT'], ['University', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Melbourne', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Parkville', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['VIC', 'NNP']],\n",
       "  ['3010', 'CD'],\n",
       "  [',', ','],\n",
       "  [['Australia', 'NNP']],\n",
       "  ['3', 'CD'],\n",
       "  ['Higher', 'NNP'],\n",
       "  ['Education', 'NNP'],\n",
       "  ['Commission', 'NNP'],\n",
       "  ['of', 'IN'],\n",
       "  [['Pakistan', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Islamabad', 'NNP']],\n",
       "  ['44000', 'CD'],\n",
       "  [',', ','],\n",
       "  [['Pakistan', 'NNP']],\n",
       "  ['4', 'CD'],\n",
       "  [['Narula', 'NNP'], ['Research', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Chapel', 'NNP'], ['Hill', 'NNP']],\n",
       "  [',', ','],\n",
       "  ['NC', 'NNP'],\n",
       "  ['27516', 'CD'],\n",
       "  [',', ','],\n",
       "  [['USA', 'NNP']],\n",
       "  ['5', 'CD'],\n",
       "  ['Department', 'NNP'],\n",
       "  ['of', 'IN'],\n",
       "  [['Science', 'NNP']],\n",
       "  ['in', 'IN'],\n",
       "  [['Korean', 'NNP'], ['Medicine', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['College', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Korean', 'NNP'], ['Medicine', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Kyung', 'NNP'], ['Hee', 'NNP'], ['University', 'NNP']],\n",
       "  [',', ','],\n",
       "  ['24', 'CD'],\n",
       "  ['Kyungheedae-ro', 'NNP'],\n",
       "  [',', ','],\n",
       "  ['Dongdaemun-gu', 'NNP'],\n",
       "  [',', ','],\n",
       "  [['Seoul', 'NNP']],\n",
       "  ['02447', 'CD'],\n",
       "  [',', ','],\n",
       "  [['South', 'NNP']],\n",
       "  ['Korea', 'NNP'],\n",
       "  ['6', 'CD'],\n",
       "  ['*', 'NN'],\n",
       "  ['Correspondence', 'NN'],\n",
       "  [':', ':'],\n",
       "  [['Muhammad', 'NNP'], ['Rashid', 'NNP'], ['Khan', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Department', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Biochemistry', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Faculty', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Biological', 'NNP'], ['Sciences', 'NNPS']],\n",
       "  [',', ','],\n",
       "  ['Quaid-i-Azam', 'NNP'],\n",
       "  ['University', 'NNP'],\n",
       "  [',', ','],\n",
       "  [['Islamabad', 'NNP']],\n",
       "  ['45320', 'CD'],\n",
       "  [',', ','],\n",
       "  [['Pakistan', 'NNP']],\n",
       "  ['.', '.'],\n",
       "  ['mrkhanqau', 'NN'],\n",
       "  ['@', 'NNP'],\n",
       "  ['yahoo.com', 'NN'],\n",
       "  [';', ':'],\n",
       "  [['Kwang', 'NNP'], ['Seok', 'NNP'], ['Ahn', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Department', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Science', 'NNP']],\n",
       "  ['in', 'IN'],\n",
       "  [['Korean', 'NNP'], ['Medicine', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['College', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Korean', 'NNP'], ['Medicine', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Kyung', 'NNP'], ['Hee', 'NNP'], ['University', 'NNP']],\n",
       "  [',', ','],\n",
       "  ['24', 'CD'],\n",
       "  ['Kyungheedae-ro', 'NNP'],\n",
       "  [',', ','],\n",
       "  ['Dongdaemun-gu', 'NNP'],\n",
       "  [',', ','],\n",
       "  [['Seoul', 'NNP']],\n",
       "  ['02447', 'CD'],\n",
       "  [',', ','],\n",
       "  [['South', 'NNP'], ['Korea', 'NNP']],\n",
       "  ['.', '.'],\n",
       "  ['ksahn', 'NN'],\n",
       "  ['@', 'JJ'],\n",
       "  ['khu.ac.kr', 'NN'],\n",
       "  ['Academic', 'NNP'],\n",
       "  ['Editor', 'NN'],\n",
       "  [':', ':'],\n",
       "  [['Gautam', 'NNP'], ['Sethi', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['National', 'NNP'], ['University', 'NNP']],\n",
       "  ['of', 'IN'],\n",
       "  [['Singapore', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Singapore', 'NNP']],\n",
       "  ['Received', 'VBD'],\n",
       "  [':', ':'],\n",
       "  ['April', 'NNP'],\n",
       "  ['16', 'CD'],\n",
       "  [',', ','],\n",
       "  ['2020', 'CD'],\n",
       "  ['Accepted', 'VBD'],\n",
       "  [':', ':'],\n",
       "  ['May', 'NNP'],\n",
       "  ['17', 'CD'],\n",
       "  [',', ','],\n",
       "  ['2020', 'CD'],\n",
       "  ['Published', 'VBN'],\n",
       "  [':', ':'],\n",
       "  ['June', 'NNP'],\n",
       "  ['29', 'CD'],\n",
       "  [',', ','],\n",
       "  ['2020', 'CD'],\n",
       "  ['Cite', 'NNP'],\n",
       "  ['this', 'DT'],\n",
       "  ['article', 'NN'],\n",
       "  [':', ':'],\n",
       "  ['Naz', 'NN'],\n",
       "  ['I', 'PRP'],\n",
       "  [',', ','],\n",
       "  [['Merarchi', 'NNP'], ['M', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Ramchandani', 'NNP'], ['S', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Khan', 'NNP'], ['MR', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Malik', 'NNP'], ['MN', 'NNP']],\n",
       "  [',', ','],\n",
       "  [['Sarwar', 'NNP'], ['S', 'NNP']],\n",
       "  [',', ','],\n",
       "  ['et', 'RB'],\n",
       "  ['al', 'NN'],\n",
       "  ['.', '.'],\n",
       "  ['An', 'DT'],\n",
       "  ['overview', 'NN'],\n",
       "  ['of', 'IN'],\n",
       "  ['the', 'DT'],\n",
       "  ['anti-cancer', 'JJ'],\n",
       "  ['actions', 'NNS'],\n",
       "  ['of', 'IN'],\n",
       "  [['Tanshinones', 'NNP']],\n",
       "  [',', ','],\n",
       "  ['derived', 'VBD'],\n",
       "  ['from', 'IN'],\n",
       "  [['Salvia', 'NNP']],\n",
       "  ['miltiorrhiza', 'NN'],\n",
       "  ['(', '('],\n",
       "  [['Danshen', 'NNP']],\n",
       "  [')', ')'],\n",
       "  ['.', '.'],\n",
       "  [['Explor', 'NNP'], ['Target', 'NNP'], ['Antitumor', 'NNP']],\n",
       "  ['Ther', 'NNP'],\n",
       "  ['.', '.'],\n",
       "  ['2020', 'CD'],\n",
       "  [';', ':'],\n",
       "  ['1:153-70.', 'JJ'],\n",
       "  ['https', 'NN'],\n",
       "  [':', ':'],\n",
       "  ['//doi', 'NN'],\n",
       "  ['.', '.'],\n",
       "  ['org/10.37349/etat.2020.00010', 'NN']]}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def text2ner(text):\n",
    "    try:\n",
    "        return post_json_request(\n",
    "            \"http://localhost:5002/text2ner\",\n",
    "            { \"text\": text }\n",
    "        )\n",
    "    except:\n",
    "        return {\"ner\": []}\n",
    "text2ner(\"Exploration of Targeted Anti-tumor Therapy\\nOpen Access Review\\n\\nAn overview of the anti-cancer actions of Tanshinones, derived\\nfrom Salvia miltiorrhiza (Danshen)\\nIrum Naz1, Myriam Merarchi2, Shanaya Ramchandani3, Muhammad Rashid Khan4*, Muhammad Nouman\\nMalik1, Sumaira Sarwar1, Acharan S Narula5, Kwang Seok Ahn6*\\nDepartment of Biochemistry, Faculty of Biological Sciences, Quaid-i-Azam University, Islamabad 45320, Pakistan\\n\\n1\\n\\nFaculty of Pharmacy, University of Paris Descartes, 75006 Paris, France\\n\\n2\\n\\nDepartment of Pharmacology-Biomedicine, The University of Melbourne, Parkville, VIC 3010, Australia\\n\\n3\\n\\nHigher Education Commission of Pakistan, Islamabad 44000, Pakistan\\n\\n4\\n\\nNarula Research, Chapel Hill, NC 27516, USA\\n\\n5\\n\\nDepartment of Science in Korean Medicine, College of Korean Medicine, Kyung Hee University, 24 Kyungheedae-ro,\\nDongdaemun-gu, Seoul 02447, South Korea\\n6\\n\\n*Correspondence: Muhammad Rashid Khan, Department of Biochemistry, Faculty of Biological Sciences, Quaid-i-Azam\\nUniversity, Islamabad 45320, Pakistan. mrkhanqau@yahoo.com; Kwang Seok Ahn, Department of Science in Korean Medicine,\\nCollege of Korean Medicine, Kyung Hee University, 24 Kyungheedae-ro, Dongdaemun-gu, Seoul 02447, South Korea. ksahn@\\nkhu.ac.kr\\nAcademic Editor: Gautam Sethi, National University of Singapore, Singapore\\nReceived: April 16, 2020 Accepted: May 17, 2020 Published: June 29, 2020\\n\\nCite this article: Naz I, Merarchi M, Ramchandani S, Khan MR, Malik MN, Sarwar S, et al. An overview of the anti-cancer\\nactions of Tanshinones, derived from Salvia miltiorrhiza (Danshen). Explor Target Antitumor Ther. 2020;1:153-70. https://doi.\\norg/10.37349/etat.2020.00010\\n\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f7798730-ae6a-4f07-b40e-0395f5aec0d3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'emails': ['mrkhanqau@yahoo.com']}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import time\n",
    "def text2emails(text):\n",
    "    try:\n",
    "        return post_json_request(\n",
    "            \"http://localhost:5002/text2emails\",\n",
    "            { \"text\": text }\n",
    "        )\n",
    "    except:\n",
    "        return {\"emails\": []}\n",
    "text2emails(\"Exploration of Targeted Anti-tumor Therapy\\nOpen Access Review\\n\\nAn overview of the anti-cancer actions of Tanshinones, derived\\nfrom Salvia miltiorrhiza (Danshen)\\nIrum Naz1, Myriam Merarchi2, Shanaya Ramchandani3, Muhammad Rashid Khan4*, Muhammad Nouman\\nMalik1, Sumaira Sarwar1, Acharan S Narula5, Kwang Seok Ahn6*\\nDepartment of Biochemistry, Faculty of Biological Sciences, Quaid-i-Azam University, Islamabad 45320, Pakistan\\n\\n1\\n\\nFaculty of Pharmacy, University of Paris Descartes, 75006 Paris, France\\n\\n2\\n\\nDepartment of Pharmacology-Biomedicine, The University of Melbourne, Parkville, VIC 3010, Australia\\n\\n3\\n\\nHigher Education Commission of Pakistan, Islamabad 44000, Pakistan\\n\\n4\\n\\nNarula Research, Chapel Hill, NC 27516, USA\\n\\n5\\n\\nDepartment of Science in Korean Medicine, College of Korean Medicine, Kyung Hee University, 24 Kyungheedae-ro,\\nDongdaemun-gu, Seoul 02447, South Korea\\n6\\n\\n*Correspondence: Muhammad Rashid Khan, Department of Biochemistry, Faculty of Biological Sciences, Quaid-i-Azam\\nUniversity, Islamabad 45320, Pakistan. mrkhanqau@yahoo.com; Kwang Seok Ahn, Department of Science in Korean Medicine,\\nCollege of Korean Medicine, Kyung Hee University, 24 Kyungheedae-ro, Dongdaemun-gu, Seoul 02447, South Korea. ksahn@\\nkhu.ac.kr\\nAcademic Editor: Gautam Sethi, National University of Singapore, Singapore\\nReceived: April 16, 2020 Accepted: May 17, 2020 Published: June 29, 2020\\n\\nCite this article: Naz I, Merarchi M, Ramchandani S, Khan MR, Malik MN, Sarwar S, et al. An overview of the anti-cancer\\nactions of Tanshinones, derived from Salvia miltiorrhiza (Danshen). Explor Target Antitumor Ther. 2020;1:153-70. https://doi.\\norg/10.37349/etat.2020.00010\\n\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "12578c93-ca5d-4a09-a497-2530c27a673d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 0.1497042179107666 seconds ---\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[['nsimigdala@bioacademy.gr',\n",
       "  'konstantopoulos@fleming.gr',\n",
       "  'aklinakis@bioacademy.gr',\n",
       "  'itrougakos@biol.uoa.gr',\n",
       "  'achalari@bioacademy.gr',\n",
       "  'asklirou@biol.uoa.gr',\n",
       "  'echavdoula@gmail.com',\n",
       "  'geo.papafotiou@gmail.com',\n",
       "  'pmelissa@bioacademy.gr',\n",
       "  'akafalidou@bioacademy.gr',\n",
       "  'npaschal@bioacademy.gr',\n",
       "  'ipateras@med.uoa.gr',\n",
       "  'mathan@bioacademy.gr'],\n",
       " ['yushanqiu2526374@163.com'],\n",
       " ['o.scott@auckland.ac.nz'],\n",
       " ['kathleen.frazer@ucd.ie'],\n",
       " ['chu-mei.chang@imperial.ac.uk'],\n",
       " ['monicake@hawaii.edu',\n",
       "  'kristimh@hawaii.edu',\n",
       "  'taflague@triton.uog.edu',\n",
       "  'bernice7@hawaii.edu',\n",
       "  'kaholoku@hawaii.edu'],\n",
       " ['zhanglei6@hrbmu.edu.cn'],\n",
       " ['dina.elgendy3@med.tanta.edu.eg'],\n",
       " ['t.hureau@unistra.fr'],\n",
       " ['hr.zheng@siat.ac.cn', 'meilin7@mail.sysu.edu.cn'],\n",
       " ['xinzhang@hmfl.ac.cn'],\n",
       " ['alinebandeirageriatra@gmail.com'],\n",
       " ['drmdrockybhasan@gmail.com'],\n",
       " ['sirinihao@163.com'],\n",
       " [],\n",
       " ['2017202040046@whu.edu.cn', 'xiaochong@stu.cdutcm.edu.cn'],\n",
       " [],\n",
       " [],\n",
       " ['aaron.wheeler@utoronto.ca'],\n",
       " ['noran.tawfik@ejust.edu.eg', 'ahmed.osman@ejust.edu.eg'],\n",
       " ['raji@purdue.edu'],\n",
       " ['isherif@cu.edu.eg', 'isherif@sci.cu.edu.eg', 'mgotte@uni-muenster.de'],\n",
       " ['looket.dihge@med.lu.se', 'lisa.ryden@med.lu.se'],\n",
       " ['zhhan@fjmu.edu.cn', 'xuchunsen@fjmu.edu.cn'],\n",
       " ['nyorthop@163.com', 'lihaomiao1977@hotmail.com'],\n",
       " ['xubinghebm@163.com', 'drmafei@126.com'],\n",
       " ['lhhospital@sina.com'],\n",
       " ['zhanglanx_9@126.com', 'zhangjin1989@szu.edu.cn'],\n",
       " ['a.chan@uci.edu'],\n",
       " ['ambinder@hawaii.edu'],\n",
       " ['guoyue@csu.edu.cn', 'fengyunzhi001@csu.edu.cn'],\n",
       " ['ntdxliuyifei@sina.com', 'jgz_edu@163.com'],\n",
       " ['rachelle.johnson@vumc.org'],\n",
       " ['wangzhiyu@gzucm.edu.cn',\n",
       "  'pengchengchengdu@126.com',\n",
       "  'ellen0000@gzucm.edu.cn'],\n",
       " ['agiasaiga@gmail.com']]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "start_time = time.time()\n",
    "emails = list(map(lambda text: text2emails(text)[\"emails\"], headers))\n",
    "print(\"--- %s seconds ---\" % (time.time() - start_time))\n",
    "emails"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "16a53569-991c-4856-9bb6-14b42c0c1cdc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'126.com',\n",
       " '163.com',\n",
       " 'auckland.ac.nz',\n",
       " 'bioacademy.gr',\n",
       " 'biol.uoa.gr',\n",
       " 'csu.edu.cn',\n",
       " 'cu.edu.eg',\n",
       " 'ejust.edu.eg',\n",
       " 'fjmu.edu.cn',\n",
       " 'fleming.gr',\n",
       " 'gmail.com',\n",
       " 'gzucm.edu.cn',\n",
       " 'hawaii.edu',\n",
       " 'hmfl.ac.cn',\n",
       " 'hotmail.com',\n",
       " 'hrbmu.edu.cn',\n",
       " 'imperial.ac.uk',\n",
       " 'mail.sysu.edu.cn',\n",
       " 'med.lu.se',\n",
       " 'med.tanta.edu.eg',\n",
       " 'med.uoa.gr',\n",
       " 'purdue.edu',\n",
       " 'sci.cu.edu.eg',\n",
       " 'siat.ac.cn',\n",
       " 'sina.com',\n",
       " 'stu.cdutcm.edu.cn',\n",
       " 'szu.edu.cn',\n",
       " 'triton.uog.edu',\n",
       " 'ucd.ie',\n",
       " 'uci.edu',\n",
       " 'uni-muenster.de',\n",
       " 'unistra.fr',\n",
       " 'utoronto.ca',\n",
       " 'vumc.org',\n",
       " 'whu.edu.cn'}"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "domains = set([email.split(\"@\")[1] for group_emails in emails for email in group_emails])\n",
    "domains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "dd793f2a-f010-4c90-bd4c-9d3af6a9f4f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n"
     ]
    }
   ],
   "source": [
    "# pip3 install python-whois\n",
    "import whois\n",
    "\n",
    "def is_registered(domain_name):\n",
    "    try:\n",
    "        w = whois.whois(domain_name)\n",
    "    except Exception:\n",
    "        return False\n",
    "    else:\n",
    "        return bool(w.domain_name)\n",
    "\n",
    "domain_name = \"med.toho‑u.ac.jp\"\n",
    "if is_registered(domain_name):\n",
    "    whois_info = whois.whois(domain_name)\n",
    "    print(whois_info)\n",
    "    \n",
    "print(is_registered(domain_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "9afdf5f7-4ca8-48cd-9c74-4625ab12fe64",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[94m{\u001b[32m'>>> LAST UPDATE OF WHOIS DATABASE'\u001b[91m: \u001b[32m'2023-03-21T11:35:55Z <<<'\u001b[91m,\n",
      "\u001b[32m 'DOMAIN NAME'\u001b[91m: \u001b[32m'NASA.GOV'\u001b[91m,\n",
      "\u001b[32m 'SECURITY CONTACT EMAIL'\u001b[91m: \u001b[32m'soc@nasa.gov'\u001b[91m,\n",
      "\u001b[32m 'STATUS'\u001b[91m: \u001b[32m'ACTIVE'\u001b[94m}\u001b[32m\u001b[0m\n",
      "\u001b[32m─┬ \u001b[95mroot\n",
      "\u001b[32m \u001b[32m├───┬ \u001b[95mDOMAIN NAME\n",
      "\u001b[32m \u001b[32m│   \u001b[32m└──── \u001b[95mNASA.GOV\n",
      "\u001b[32m \u001b[32m├───┬ \u001b[95mSTATUS\n",
      "\u001b[32m \u001b[32m│   \u001b[32m└──── \u001b[95mACTIVE\n",
      "\u001b[32m \u001b[32m├───┬ \u001b[95mSECURITY CONTACT EMAIL\n",
      "\u001b[32m \u001b[32m│   \u001b[32m└──── \u001b[95msoc@nasa.gov\n",
      "\u001b[32m \u001b[32m└───┬ \u001b[95m>>> LAST UPDATE OF WHOIS DATABASE\n",
      "\u001b[32m \u001b[32m    \u001b[32m└──── \u001b[95m2023-03-21T11:35:55Z <<<\n",
      "\u001b[0m\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "% DOTGOV WHOIS Server ready\n",
      "   Domain Name: NASA.GOV\n",
      "   Status: ACTIVE\n",
      "   Security Contact Email: soc@nasa.gov\n",
      "\n",
      ">>> Last update of whois database: 2023-03-21T11:35:55Z <<<\n",
      "\n",
      "Please be advised that this whois server only contains information pertaining\n",
      "to the .GOV domain. For information for other domains please use the whois\n",
      "server at RS.INTERNIC.NET. \n",
      "\n",
      "Creation date   : \n",
      "Expiration date : \n",
      "Updated date    : \n",
      "Updated date    : \n"
     ]
    }
   ],
   "source": [
    "# First step is to import the package\n",
    "import whois21\n",
    "\n",
    "def domain2whois(domain: str) -> str:\n",
    "    return whois21.WHOIS(domain).raw.decode('utf-8')\n",
    "\n",
    "query = 'nasa.gov'\n",
    "\n",
    "# Second step is to create an instance of the WHOIS class\n",
    "whois = whois21.WHOIS(query)\n",
    "\n",
    "# Third step is to check if the operation was successful\n",
    "if not whois.success:\n",
    "    print(whois.error)\n",
    "    exit()\n",
    "\n",
    "# And basically you are done!\n",
    "# Now you can print the results\n",
    "import log21  # I use log21 to print the results in a cool way 8D\n",
    "\n",
    "# Print the results in a nice way\n",
    "# PPrint the dictionary\n",
    "log21.pprint(whois.whois_data)\n",
    "# Tree-Print the dictionary\n",
    "log21.tree_print(whois.whois_data)\n",
    "\n",
    "# Or you can print the results in as raw text\n",
    "print(whois.raw.decode('utf-8'))\n",
    "\n",
    "# Or you can access each part of the results individually\n",
    "print(f'Creation date   : {whois.creation_date}')\n",
    "print(f'Expiration date : {whois.expires_date}')\n",
    "print(f'Updated date    : {whois.updated_date}')\n",
    "print(f'Updated date    : {whois.registrar_name}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "55e39221-df7f-4d60-bba1-4e2cb9f989bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "whois_list = [domain2whois(domain) for domain in domains]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "dd0738be-815a-4831-95f9-24536222c129",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Invalid parameter:mail.sysu.edu.cn\\n',\n",
       " '   Domain Name: HOTMAIL.COM\\r\\n   Registry Domain ID: 5244890_DOMAIN_COM-VRSN\\r\\n   Registrar WHOIS Server: whois.markmonitor.com\\r\\n   Registrar URL: http://www.markmonitor.com\\r\\n   Updated Date: 2021-02-02T17:04:58Z\\r\\n   Creation Date: 1996-03-27T05:00:00Z\\r\\n   Registry Expiry Date: 2024-03-28T04:00:00Z\\r\\n   Registrar: MarkMonitor Inc.\\r\\n   Registrar IANA ID: 292\\r\\n   Registrar Abuse Contact Email: abusecomplaints@markmonitor.com\\r\\n   Registrar Abuse Contact Phone: +1.2086851750\\r\\n   Domain Status: clientDeleteProhibited https://icann.org/epp#clientDeleteProhibited\\r\\n   Domain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited\\r\\n   Domain Status: clientUpdateProhibited https://icann.org/epp#clientUpdateProhibited\\r\\n   Domain Status: serverDeleteProhibited https://icann.org/epp#serverDeleteProhibited\\r\\n   Domain Status: serverTransferProhibited https://icann.org/epp#serverTransferProhibited\\r\\n   Domain Status: serverUpdateProhibited https://icann.org/epp#serverUpdateProhibited\\r\\n   Name Server: NS1-205.AZURE-DNS.COM\\r\\n   Name Server: NS2-205.AZURE-DNS.NET\\r\\n   Name Server: NS3-205.AZURE-DNS.ORG\\r\\n   Name Server: NS4-205.AZURE-DNS.INFO\\r\\n   DNSSEC: unsigned\\r\\n   URL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/\\r\\n>>> Last update of whois database: 2023-03-21T11:36:12Z <<<\\r\\n\\r\\nFor more information on Whois status codes, please visit https://icann.org/epp\\r\\n\\r\\nNOTICE: The expiration date displayed in this record is the date the\\r\\nregistrar\\'s sponsorship of the domain name registration in the registry is\\r\\ncurrently set to expire. This date does not necessarily reflect the expiration\\r\\ndate of the domain name registrant\\'s agreement with the sponsoring\\r\\nregistrar.  Users may consult the sponsoring registrar\\'s Whois database to\\r\\nview the registrar\\'s reported date of expiration for this registration.\\r\\n\\r\\nTERMS OF USE: You are not authorized to access or query our Whois\\r\\ndatabase through the use of electronic processes that are high-volume and\\r\\nautomated except as reasonably necessary to register domain names or\\r\\nmodify existing registrations; the Data in VeriSign Global Registry\\r\\nServices\\' (\"VeriSign\") Whois database is provided by VeriSign for\\r\\ninformation purposes only, and to assist persons in obtaining information\\r\\nabout or related to a domain name registration record. VeriSign does not\\r\\nguarantee its accuracy. By submitting a Whois query, you agree to abide\\r\\nby the following terms of use: You agree that you may use this Data only\\r\\nfor lawful purposes and that under no circumstances will you use this Data\\r\\nto: (1) allow, enable, or otherwise support the transmission of mass\\r\\nunsolicited, commercial advertising or solicitations via e-mail, telephone,\\r\\nor facsimile; or (2) enable high volume, automated, electronic processes\\r\\nthat apply to VeriSign (or its computer systems). The compilation,\\r\\nrepackaging, dissemination or other use of this Data is expressly\\r\\nprohibited without the prior written consent of VeriSign. You agree not to\\r\\nuse electronic processes that are automated and high-volume to access or\\r\\nquery the Whois database except as reasonably necessary to register\\r\\ndomain names or modify existing registrations. VeriSign reserves the right\\r\\nto restrict your access to the Whois database in its sole discretion to ensure\\r\\noperational stability.  VeriSign may restrict or terminate your access to the\\r\\nWhois database for failure to abide by these terms of use. VeriSign\\r\\nreserves the right to modify these terms at any time.\\r\\n\\r\\nThe Registry database contains ONLY .COM, .NET, .EDU domains and\\r\\nRegistrars.\\r\\n',\n",
       " '% This is the RIPE Database query service.\\n% The objects are in RPSL format.\\n%\\n% The RIPE Database is subject to Terms and Conditions.\\n% See http://www.ripe.net/db/support/db-terms-conditions.pdf\\n\\n% Note: this output has been filtered.\\n%       To receive output for a database update, use the \"-B\" flag.\\n\\n%ERROR:101: no entries found\\n%\\n% No entries found in source RIPE.\\n\\n% This query was served by the RIPE Database Query Service version 1.106 (SHETLAND)\\n\\n\\n',\n",
       " \"Domain Name: vumc.org\\r\\nRegistry Domain ID: 2dc7c266f0f149c2b370f9842472bdd5-LROR\\r\\nRegistrar WHOIS Server: whois.networksolutions.com\\r\\nRegistrar URL: http://www.networksolutions.com\\r\\nUpdated Date: 2022-04-21T14:16:06Z\\r\\nCreation Date: 1999-12-31T20:35:09Z\\r\\nRegistry Expiry Date: 2025-12-31T20:35:09Z\\r\\nRegistrar: Network Solutions, LLC\\r\\nRegistrar IANA ID: 2\\r\\nRegistrar Abuse Contact Email: domain.operations@web.com\\r\\nRegistrar Abuse Contact Phone: +1.8777228662\\r\\nDomain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited\\r\\nRegistry Registrant ID: REDACTED FOR PRIVACY\\r\\nRegistrant Name: REDACTED FOR PRIVACY\\r\\nRegistrant Organization: \\r\\nRegistrant Street: REDACTED FOR PRIVACY\\r\\nRegistrant City: REDACTED FOR PRIVACY\\r\\nRegistrant State/Province: FL\\r\\nRegistrant Postal Code: REDACTED FOR PRIVACY\\r\\nRegistrant Country: US\\r\\nRegistrant Phone: REDACTED FOR PRIVACY\\r\\nRegistrant Phone Ext: REDACTED FOR PRIVACY\\r\\nRegistrant Fax: REDACTED FOR PRIVACY\\r\\nRegistrant Fax Ext: REDACTED FOR PRIVACY\\r\\nRegistrant Email: Please query the RDDS service of the Registrar of Record identified in this output for information on how to contact the Registrant, Admin, or Tech contact of the queried domain name.\\r\\nRegistry Admin ID: REDACTED FOR PRIVACY\\r\\nAdmin Name: REDACTED FOR PRIVACY\\r\\nAdmin Organization: REDACTED FOR PRIVACY\\r\\nAdmin Street: REDACTED FOR PRIVACY\\r\\nAdmin City: REDACTED FOR PRIVACY\\r\\nAdmin State/Province: REDACTED FOR PRIVACY\\r\\nAdmin Postal Code: REDACTED FOR PRIVACY\\r\\nAdmin Country: REDACTED FOR PRIVACY\\r\\nAdmin Phone: REDACTED FOR PRIVACY\\r\\nAdmin Phone Ext: REDACTED FOR PRIVACY\\r\\nAdmin Fax: REDACTED FOR PRIVACY\\r\\nAdmin Fax Ext: REDACTED FOR PRIVACY\\r\\nAdmin Email: Please query the RDDS service of the Registrar of Record identified in this output for information on how to contact the Registrant, Admin, or Tech contact of the queried domain name.\\r\\nRegistry Tech ID: REDACTED FOR PRIVACY\\r\\nTech Name: REDACTED FOR PRIVACY\\r\\nTech Organization: REDACTED FOR PRIVACY\\r\\nTech Street: REDACTED FOR PRIVACY\\r\\nTech City: REDACTED FOR PRIVACY\\r\\nTech State/Province: REDACTED FOR PRIVACY\\r\\nTech Postal Code: REDACTED FOR PRIVACY\\r\\nTech Country: REDACTED FOR PRIVACY\\r\\nTech Phone: REDACTED FOR PRIVACY\\r\\nTech Phone Ext: REDACTED FOR PRIVACY\\r\\nTech Fax: REDACTED FOR PRIVACY\\r\\nTech Fax Ext: REDACTED FOR PRIVACY\\r\\nTech Email: Please query the RDDS service of the Registrar of Record identified in this output for information on how to contact the Registrant, Admin, or Tech contact of the queried domain name.\\r\\nName Server: ns1.vumc.org\\r\\nName Server: ns2.vumc.org\\r\\nDNSSEC: unsigned\\r\\nURL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/\\r\\n>>> Last update of WHOIS database: 2023-03-21T11:38:42Z <<<\\r\\n\\r\\nFor more information on Whois status codes, please visit https://icann.org/epp\\r\\n\\r\\nTerms of Use: Access to Public Interest Registry WHOIS information is provided to assist persons in determining the contents of a domain name registration record in the Public Interest Registry registry database. The data in this record is provided by Public Interest Registry for informational purposes only, and Public Interest Registry does not guarantee its accuracy. This service is intended only for query-based access. You agree that you will use this data only for lawful purposes and that, under no circumstances will you use this data to (a) allow, enable, or otherwise support the transmission by e-mail, telephone, or facsimile of mass unsolicited, commercial advertising or solicitations to entities other than the data recipient's own existing customers; or (b) enable high volume, automated, electronic processes that send queries or data to the systems of Registry Operator, a Registrar, or Identity Digital except as reasonably necessary to register domain names or modify existing registrations. All rights reserved. Public Interest Registry reserves the right to modify these terms at any time. By submitting this query, you agree to abide by this policy.  The Registrar of Record identified in this output may have an RDDS service that can be queried for additional information on how to contact the Registrant, Admin, or Tech contact of the queried domain name.\\r\\n\",\n",
       " '% This is the RIPE Database query service.\\n% The objects are in RPSL format.\\n%\\n% The RIPE Database is subject to Terms and Conditions.\\n% See http://www.ripe.net/db/support/db-terms-conditions.pdf\\n\\n% Note: this output has been filtered.\\n%       To receive output for a database update, use the \"-B\" flag.\\n\\n%ERROR:101: no entries found\\n%\\n% No entries found in source RIPE.\\n\\n% This query was served by the RIPE Database Query Service version 1.106 (ABERDEEN)\\n\\n\\n',\n",
       " '# Copyright (c) 1997- The Swedish Internet Foundation.\\n# All rights reserved.\\n# The information obtained through searches, or otherwise, is protected\\n# by the Swedish Copyright Act (1960:729) and international conventions.\\n# It is also subject to database protection according to the Swedish\\n# Copyright Act.\\n# Any use of this material to target advertising or\\n# similar activities is forbidden and will be prosecuted.\\n# If any of the information below is transferred to a third\\n# party, it must be done in its entirety. This server must\\n# not be used as a backend for a search engine.\\n# Result of search for registered domain names under\\n# the .se top level domain.\\n# This whois printout is printed with UTF-8 encoding.\\n#\\ndomain \"med.lu.se\" not found.\\n',\n",
       " '% This is the RIPE Database query service.\\n% The objects are in RPSL format.\\n%\\n% The RIPE Database is subject to Terms and Conditions.\\n% See http://www.ripe.net/db/support/db-terms-conditions.pdf\\n\\n% Note: this output has been filtered.\\n%       To receive output for a database update, use the \"-B\" flag.\\n\\n%ERROR:101: no entries found\\n%\\n% No entries found in source RIPE.\\n\\n% This query was served by the RIPE Database Query Service version 1.106 (SHETLAND)\\n\\n\\n',\n",
       " '%%\\n%% This is the AFNIC Whois server.\\n%%\\n%% complete date format: YYYY-MM-DDThh:mm:ssZ\\n%%\\n%% Rights restricted by copyright.\\n%% See https://www.afnic.fr/en/domain-names-and-support/everything-there-is-to-know-about-domain-names/find-a-domain-name-or-a-holder-using-whois/\\n%%\\n%%\\n\\ndomain:                        unistra.fr\\r\\nstatus:                        ACTIVE\\r\\neppstatus:                     active\\r\\nhold:                          NO\\r\\nholder-c:                      UDS65-FRNIC\\r\\nadmin-c:                       MD21851-FRNIC\\r\\ntech-c:                        GRST1-FRNIC\\r\\ntech-c:                        JB2348-FRNIC\\r\\nregistrar:                     GIP RENATER\\r\\nExpiry Date:                   2023-07-02T09:39:52Z\\r\\ncreated:                       2007-07-24T12:13:04Z\\r\\nlast-update:                   2022-07-02T10:35:28Z\\r\\nsource:                        FRNIC\\r\\n\\r\\nnserver:                       ns1.u-strasbg.fr\\r\\nnserver:                       ns2.u-strasbg.fr\\r\\nnserver:                       shiva.jussieu.fr\\r\\nsource:                        FRNIC\\r\\n\\r\\nregistrar:                     GIP RENATER\\r\\naddress:                       Array\\r\\naddress:                       75013 PARIS\\r\\ncountry:                       FR\\r\\nphone:                         +33.153942030\\r\\nfax-no:                        +33.153942031\\r\\ne-mail:                        domaine@renater.fr\\r\\nwebsite:                       http://www.renater.fr\\r\\nanonymous:                     No\\r\\nregistered:                    1997-12-29T00:00:00Z\\r\\nsource:                        FRNIC\\r\\n\\r\\nnic-hdl:                       GRST1-FRNIC\\r\\ntype:                          PERSON\\r\\ncontact:                       Gip Renater Support Technique Dns\\r\\naddress:                       GIP RENATER\\r\\naddress:                       23-25, rue Daviel\\r\\naddress:                       75013 Paris\\r\\ncountry:                       FR\\r\\nphone:                         +33.153942040\\r\\ne-mail:                        support-dns@renater.fr\\r\\nregistrar:                     GIP RENATER\\r\\nchanged:                       2023-03-04T15:46:24.018549Z\\r\\nanonymous:                     NO\\r\\nobsoleted:                     NO\\r\\neppstatus:                     associated\\r\\neppstatus:                     active\\r\\neligstatus:                    not identified\\r\\nreachstatus:                   not identified\\r\\nsource:                        FRNIC\\r\\n\\r\\nnic-hdl:                       UDS65-FRNIC\\r\\ntype:                          ORGANIZATION\\r\\ncontact:                       Université de Strasbourg\\r\\naddress:                       Université de Strasbourg - Direction Informatique\\r\\naddress:                       14 rue René Descartes\\r\\naddress:                       67000 Strasbourg\\r\\ncountry:                       FR\\r\\nphone:                         +33.368850000\\r\\ne-mail:                        di-dns@unistra.fr\\r\\nregistrar:                     GIP RENATER\\r\\nchanged:                       2019-04-25T09:19:43Z\\r\\nanonymous:                     NO\\r\\nobsoleted:                     NO\\r\\neppstatus:                     associated\\r\\neppstatus:                     active\\r\\neligstatus:                    ok\\r\\neligsource:                    REGISTRAR\\r\\neligdate:                      2019-04-25T00:00:00Z\\r\\nreachstatus:                   ok\\r\\nreachmedia:                    email\\r\\nreachsource:                   REGISTRAR\\r\\nreachdate:                     2019-04-25T00:00:00Z\\r\\nsource:                        FRNIC\\r\\n\\r\\nnic-hdl:                       JB2348-FRNIC\\r\\ntype:                          PERSON\\r\\ncontact:                       Jean Benoit\\r\\naddress:                       Universite de Strasbourg\\r\\naddress:                       7, rue Rene Descartes\\r\\naddress:                       67000 Strasbourg\\r\\ncountry:                       FR\\r\\nphone:                         +33.390240318\\r\\nfax-no:                        +33.390240312\\r\\ne-mail:                        jean@unistra.fr\\r\\nregistrar:                     GIP RENATER\\r\\nchanged:                       2022-12-20T08:43:01.863425Z\\r\\nanonymous:                     NO\\r\\nobsoleted:                     NO\\r\\neppstatus:                     associated\\r\\neppstatus:                     active\\r\\neligstatus:                    not identified\\r\\nreachstatus:                   not identified\\r\\nsource:                        FRNIC\\r\\n\\r\\nnic-hdl:                       MD21851-FRNIC\\r\\ntype:                          PERSON\\r\\ncontact:                       Michel Deneken\\r\\naddress:                       Université de Strasbourg\\r\\naddress:                       4, rue Blaise Pascal\\r\\naddress:                       67000 Strasbourg\\r\\ncountry:                       FR\\r\\nphone:                         +33.368850000\\r\\ne-mail:                        president@unistra.fr\\r\\nregistrar:                     GIP RENATER\\r\\nchanged:                       2022-12-20T08:43:01.85874Z\\r\\nanonymous:                     NO\\r\\nobsoleted:                     NO\\r\\neppstatus:                     associated\\r\\neppstatus:                     active\\r\\neligstatus:                    ok\\r\\neligdate:                      2016-10-31T00:00:00Z\\r\\nreachstatus:                   not identified\\r\\nsource:                        FRNIC\\r\\n\\n>>> WHOIS request date: 2023-03-21T11:41:02.143632Z <<<\\n\\r\\n',\n",
       " '   Domain Name: 163.COM\\r\\n   Registry Domain ID: 473619_DOMAIN_COM-VRSN\\r\\n   Registrar WHOIS Server: whois.markmonitor.com\\r\\n   Registrar URL: http://www.markmonitor.com\\r\\n   Updated Date: 2019-01-25T07:56:57Z\\r\\n   Creation Date: 1997-09-15T04:00:00Z\\r\\n   Registry Expiry Date: 2026-09-14T04:00:00Z\\r\\n   Registrar: MarkMonitor Inc.\\r\\n   Registrar IANA ID: 292\\r\\n   Registrar Abuse Contact Email: abusecomplaints@markmonitor.com\\r\\n   Registrar Abuse Contact Phone: +1.2086851750\\r\\n   Domain Status: clientDeleteProhibited https://icann.org/epp#clientDeleteProhibited\\r\\n   Domain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited\\r\\n   Domain Status: clientUpdateProhibited https://icann.org/epp#clientUpdateProhibited\\r\\n   Domain Status: serverDeleteProhibited https://icann.org/epp#serverDeleteProhibited\\r\\n   Domain Status: serverTransferProhibited https://icann.org/epp#serverTransferProhibited\\r\\n   Domain Status: serverUpdateProhibited https://icann.org/epp#serverUpdateProhibited\\r\\n   Name Server: NS1.NEASE.NET\\r\\n   Name Server: NS2.166.COM\\r\\n   Name Server: NS3.NEASE.NET\\r\\n   Name Server: NS4.NEASE.NET\\r\\n   Name Server: NS5.NEASE.NET\\r\\n   Name Server: NS6.NEASE.NET\\r\\n   Name Server: NS8.166.COM\\r\\n   DNSSEC: unsigned\\r\\n   URL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/\\r\\n>>> Last update of whois database: 2023-03-21T11:40:43Z <<<\\r\\n\\r\\nFor more information on Whois status codes, please visit https://icann.org/epp\\r\\n\\r\\nNOTICE: The expiration date displayed in this record is the date the\\r\\nregistrar\\'s sponsorship of the domain name registration in the registry is\\r\\ncurrently set to expire. This date does not necessarily reflect the expiration\\r\\ndate of the domain name registrant\\'s agreement with the sponsoring\\r\\nregistrar.  Users may consult the sponsoring registrar\\'s Whois database to\\r\\nview the registrar\\'s reported date of expiration for this registration.\\r\\n\\r\\nTERMS OF USE: You are not authorized to access or query our Whois\\r\\ndatabase through the use of electronic processes that are high-volume and\\r\\nautomated except as reasonably necessary to register domain names or\\r\\nmodify existing registrations; the Data in VeriSign Global Registry\\r\\nServices\\' (\"VeriSign\") Whois database is provided by VeriSign for\\r\\ninformation purposes only, and to assist persons in obtaining information\\r\\nabout or related to a domain name registration record. VeriSign does not\\r\\nguarantee its accuracy. By submitting a Whois query, you agree to abide\\r\\nby the following terms of use: You agree that you may use this Data only\\r\\nfor lawful purposes and that under no circumstances will you use this Data\\r\\nto: (1) allow, enable, or otherwise support the transmission of mass\\r\\nunsolicited, commercial advertising or solicitations via e-mail, telephone,\\r\\nor facsimile; or (2) enable high volume, automated, electronic processes\\r\\nthat apply to VeriSign (or its computer systems). The compilation,\\r\\nrepackaging, dissemination or other use of this Data is expressly\\r\\nprohibited without the prior written consent of VeriSign. You agree not to\\r\\nuse electronic processes that are automated and high-volume to access or\\r\\nquery the Whois database except as reasonably necessary to register\\r\\ndomain names or modify existing registrations. VeriSign reserves the right\\r\\nto restrict your access to the Whois database in its sole discretion to ensure\\r\\noperational stability.  VeriSign may restrict or terminate your access to the\\r\\nWhois database for failure to abide by these terms of use. VeriSign\\r\\nreserves the right to modify these terms at any time.\\r\\n\\r\\nThe Registry database contains ONLY .COM, .NET, .EDU domains and\\r\\nRegistrars.\\r\\n',\n",
       " '   Domain Name: 126.COM\\r\\n   Registry Domain ID: 1373158_DOMAIN_COM-VRSN\\r\\n   Registrar WHOIS Server: whois.markmonitor.com\\r\\n   Registrar URL: http://www.markmonitor.com\\r\\n   Updated Date: 2019-01-25T07:56:57Z\\r\\n   Creation Date: 1998-02-28T05:00:00Z\\r\\n   Registry Expiry Date: 2027-02-28T05:00:00Z\\r\\n   Registrar: MarkMonitor Inc.\\r\\n   Registrar IANA ID: 292\\r\\n   Registrar Abuse Contact Email: abusecomplaints@markmonitor.com\\r\\n   Registrar Abuse Contact Phone: +1.2086851750\\r\\n   Domain Status: clientDeleteProhibited https://icann.org/epp#clientDeleteProhibited\\r\\n   Domain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited\\r\\n   Domain Status: clientUpdateProhibited https://icann.org/epp#clientUpdateProhibited\\r\\n   Domain Status: serverDeleteProhibited https://icann.org/epp#serverDeleteProhibited\\r\\n   Domain Status: serverTransferProhibited https://icann.org/epp#serverTransferProhibited\\r\\n   Domain Status: serverUpdateProhibited https://icann.org/epp#serverUpdateProhibited\\r\\n   Name Server: NS1.NEASE.NET\\r\\n   Name Server: NS2.166.COM\\r\\n   Name Server: NS3.NEASE.NET\\r\\n   Name Server: NS4.NEASE.NET\\r\\n   Name Server: NS5.NEASE.NET\\r\\n   Name Server: NS6.NEASE.NET\\r\\n   Name Server: NS8.166.COM\\r\\n   DNSSEC: unsigned\\r\\n   URL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/\\r\\n>>> Last update of whois database: 2023-03-21T11:40:43Z <<<\\r\\n\\r\\nFor more information on Whois status codes, please visit https://icann.org/epp\\r\\n\\r\\nNOTICE: The expiration date displayed in this record is the date the\\r\\nregistrar\\'s sponsorship of the domain name registration in the registry is\\r\\ncurrently set to expire. This date does not necessarily reflect the expiration\\r\\ndate of the domain name registrant\\'s agreement with the sponsoring\\r\\nregistrar.  Users may consult the sponsoring registrar\\'s Whois database to\\r\\nview the registrar\\'s reported date of expiration for this registration.\\r\\n\\r\\nTERMS OF USE: You are not authorized to access or query our Whois\\r\\ndatabase through the use of electronic processes that are high-volume and\\r\\nautomated except as reasonably necessary to register domain names or\\r\\nmodify existing registrations; the Data in VeriSign Global Registry\\r\\nServices\\' (\"VeriSign\") Whois database is provided by VeriSign for\\r\\ninformation purposes only, and to assist persons in obtaining information\\r\\nabout or related to a domain name registration record. VeriSign does not\\r\\nguarantee its accuracy. By submitting a Whois query, you agree to abide\\r\\nby the following terms of use: You agree that you may use this Data only\\r\\nfor lawful purposes and that under no circumstances will you use this Data\\r\\nto: (1) allow, enable, or otherwise support the transmission of mass\\r\\nunsolicited, commercial advertising or solicitations via e-mail, telephone,\\r\\nor facsimile; or (2) enable high volume, automated, electronic processes\\r\\nthat apply to VeriSign (or its computer systems). The compilation,\\r\\nrepackaging, dissemination or other use of this Data is expressly\\r\\nprohibited without the prior written consent of VeriSign. You agree not to\\r\\nuse electronic processes that are automated and high-volume to access or\\r\\nquery the Whois database except as reasonably necessary to register\\r\\ndomain names or modify existing registrations. VeriSign reserves the right\\r\\nto restrict your access to the Whois database in its sole discretion to ensure\\r\\noperational stability.  VeriSign may restrict or terminate your access to the\\r\\nWhois database for failure to abide by these terms of use. VeriSign\\r\\nreserves the right to modify these terms at any time.\\r\\n\\r\\nThe Registry database contains ONLY .COM, .NET, .EDU domains and\\r\\nRegistrars.\\r\\n',\n",
       " 'This Registry database contains ONLY .EDU domains.\\nThe data in the EDUCAUSE Whois database is provided\\nby EDUCAUSE for information purposes in order to\\nassist in the process of obtaining information about\\nor related to .edu domain registration records.\\n\\nThe EDUCAUSE Whois database is authoritative for the\\n.EDU domain.\\n\\nA Web interface for the .EDU EDUCAUSE Whois Server is\\navailable at: http://whois.educause.edu\\n\\nBy submitting a Whois query, you agree that this information\\nwill not be used to allow, enable, or otherwise support\\nthe transmission of unsolicited commercial advertising or\\nsolicitations via e-mail.  The use of electronic processes to\\nharvest information from this server is generally prohibited\\nexcept as reasonably necessary to register or modify .edu\\ndomain names.\\n\\n-------------------------------------------------------------\\n\\nDomain Name: HAWAII.EDU\\n\\nRegistrant:\\n\\tUniversity of Hawaii\\n\\t2520 Correa Road\\n\\tHonolulu, HI 96822\\n\\tUSA\\n\\nAdministrative Contact:\\n\\tDomain Admin\\n\\tUniversity of Hawaii\\n\\t2520 Correa Road\\n\\tHonolulu, HI 96822\\n\\tUSA\\n\\t+1.8089562393\\n\\tnetcontact@hawaii.edu\\n\\nTechnical Contact:\\n\\tDomain Admin\\n\\tUniversity of Hawaii\\n\\t2520 Correa Road\\n\\tHonolulu, HI 96822\\n\\tUSA\\n\\t+1.8089562393\\n\\tnetcontact@hawaii.edu\\n\\nName Servers:\\n\\tDNS2.HAWAII.EDU\\n\\tDNS4.HAWAII.EDU\\n\\tDNS1.HAWAII.EDU\\n\\nDomain record activated:    27-Oct-1986\\nDomain record last updated: 20-Dec-2022\\nDomain expires:             31-Jul-2024\\n',\n",
       " '% This is the RIPE Database query service.\\n% The objects are in RPSL format.\\n%\\n% The RIPE Database is subject to Terms and Conditions.\\n% See http://www.ripe.net/db/support/db-terms-conditions.pdf\\n\\n% Note: this output has been filtered.\\n%       To receive output for a database update, use the \"-B\" flag.\\n\\n%ERROR:101: no entries found\\n%\\n% No entries found in source RIPE.\\n\\n% This query was served by the RIPE Database Query Service version 1.106 (BUSA)\\n\\n\\n',\n",
       " '% This is the RIPE Database query service.\\n% The objects are in RPSL format.\\n%\\n% The RIPE Database is subject to Terms and Conditions.\\n% See http://www.ripe.net/db/support/db-terms-conditions.pdf\\n\\n% Note: this output has been filtered.\\n%       To receive output for a database update, use the \"-B\" flag.\\n\\n%ERROR:101: no entries found\\n%\\n% No entries found in source RIPE.\\n\\n% This query was served by the RIPE Database Query Service version 1.106 (ABERDEEN)\\n\\n\\n',\n",
       " 'NO MATCH: triton.uog.edu\\n',\n",
       " 'Domain Name: siat.ac.cn\\nROID: 20060420s10071s00036340-cn\\nDomain Status: ok\\nRegistrant: 中国科学院深圳先进技术研究院\\nRegistrant Contact Email: jl.yan@siat.ac.cn\\nSponsoring Registrar: 北京中科三方网络技术有限公司\\nName Server: ns1.sfn.cn\\nName Server: ns2.sfn.cn\\nRegistration Time: 2006-04-20 10:14:40\\nExpiration Time: 2031-04-20 10:14:40\\nDNSSEC: unsigned\\n',\n",
       " 'Out of this registry.\\n',\n",
       " 'Domain Name: ucd.ie\\r\\nRegistry Domain ID: 666127-IEDR\\r\\nRegistrar WHOIS Server: whois.weare.ie\\r\\nRegistrar URL: https://www.heanet.ie/services/hosting/domain-registration\\r\\nUpdated Date: 2022-05-19T15:30:05Z\\r\\nCreation Date: 1997-04-03T23:00:00Z\\r\\nRegistry Expiry Date: 2023-04-04T15:22:38Z\\r\\nRegistrar: HEAnet\\r\\nRegistrar IANA ID: not applicable\\r\\nRegistrar Abuse Contact Email: noc@heanet.ie\\r\\nRegistrar Abuse Contact Phone: +353.16609040\\r\\nDomain Status: ok https://icann.org/epp#ok\\r\\nRegistry Registrant ID: 349514-IEDR\\r\\nRegistrant Name: REDACTED FOR PRIVACY\\r\\nRegistry Admin ID: 134172-IEDR\\r\\nRegistry Tech ID: 134172-IEDR\\r\\nRegistry Billing ID: REDACTED FOR PRIVACY\\r\\nName Server: auth-ns1.ucd.ie\\r\\nName Server: auth-ns2.ucd.ie\\r\\nName Server: scsnms.switch.ch\\r\\nName Server: sec-ns.tcd.ie\\r\\nName Server: uucp-gw-1.pa.dec.com\\r\\nName Server: uucp-gw-2.pa.dec.com\\r\\nDNSSEC: unsigned\\r\\n>>> Last update of WHOIS database: 2023-03-21T11:41:15Z <<<\\r\\n\\r\\n% Important Notice\\r\\n% If you believe that information published on this page is incorrect, or should not be published,please contact your Registrar, or the .IE Registration Services Department who will advise you further.\\r\\n% You can also contact your Registrar or .IE if your Registrant name is not showing and you would like it to be published.\\r\\n% Our contact information is available at https://www.weare.ie/contact-us/\\r\\n% Rights restricted by copyright: https://www.weare.ie/registrant-terms-conditions-policy/\\r\\n% Do not remove this notice\\r\\n',\n",
       " \"Domain Name: utoronto.ca\\r\\nRegistry Domain ID: D1782-CIRA\\r\\nRegistrar WHOIS Server: whois.ca.fury.ca\\r\\nRegistrar URL: www.rebel.ca\\r\\nUpdated Date: 2022-01-06T18:00:37Z\\r\\nCreation Date: 2000-09-28T14:01:07Z\\r\\nRegistry Expiry Date: 2031-03-02T05:00:00Z\\r\\nRegistrar: Rebel.ca Corp.\\r\\nRegistrar IANA ID: not applicable\\r\\nRegistrar Abuse Contact Email: abuse@rebel.com\\r\\nRegistrar Abuse Contact Phone: +1.6132252000\\r\\nDomain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited\\r\\nDomain Status: clientUpdateProhibited https://icann.org/epp#clientUpdateProhibited\\r\\nRegistry Registrant ID: 18785276-CIRA\\r\\nRegistrant Name: University Toronto\\r\\nRegistrant Organization:\\r\\nRegistrant Street: 4 Bancroft Avenue,  Room 102\\r\\nRegistrant City: Toronto\\r\\nRegistrant State/Province: ON\\r\\nRegistrant Postal Code: M5S1C1\\r\\nRegistrant Country: CA\\r\\nRegistrant Phone: +1.4169780470\\r\\nRegistrant Phone Ext:\\r\\nRegistrant Fax:\\r\\nRegistrant Fax Ext:\\r\\nRegistrant Email: russell.sutherland@utoronto.ca\\r\\nRegistry Admin ID: 2026370-CIRA\\r\\nAdmin Name: Abuse Contact\\r\\nAdmin Organization: University of Toronto\\r\\nAdmin Street: 4 Bancroft Ave.\\r\\nAdmin City: Toronto\\r\\nAdmin State/Province: ON\\r\\nAdmin Postal Code: M5S1C1\\r\\nAdmin Country: CA\\r\\nAdmin Phone: +1.4169780470\\r\\nAdmin Phone Ext:\\r\\nAdmin Fax: +1.4169786620\\r\\nAdmin Fax Ext:\\r\\nAdmin Email: abuse@utoronto.ca\\r\\nRegistry Tech ID: C851155-CIRA\\r\\nTech Name: Russell Sutherland\\r\\nTech Organization: University of Toronto\\r\\nTech Street: 4 Bancroft Ave.\\r\\nTech City: Toronto\\r\\nTech State/Province: ON\\r\\nTech Postal Code: M5S1C1\\r\\nTech Country: CA\\r\\nTech Phone: +1.4169780470\\r\\nTech Phone Ext:\\r\\nTech Fax: +1.4169786620\\r\\nTech Fax Ext:\\r\\nTech Email: russell.sutherland@utoronto.ca\\r\\nRegistry Billing ID:\\r\\nBilling Name:\\r\\nBilling Organization:\\r\\nBilling Street:\\r\\nBilling City:\\r\\nBilling State/Province:\\r\\nBilling Postal Code:\\r\\nBilling Country:\\r\\nBilling Phone:\\r\\nBilling Phone Ext:\\r\\nBilling Fax:\\r\\nBilling Fax Ext:\\r\\nBilling Email:\\r\\nName Server: ns1.d-zone.ca\\r\\nName Server: ns2.d-zone.ca\\r\\nDNSSEC: unsigned\\r\\nURL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/\\r\\n>>> Last update of WHOIS database: 2023-03-21T11:41:16Z <<<\\r\\n\\r\\nFor more information on Whois status codes, please visit https://icann.org/epp\\r\\n\\r\\n%\\r\\n% Use of CIRA's WHOIS service is governed by the Terms of Use in its Legal\\r\\n% Notice, available at http://www.cira.ca/legal-notice/?lang=en\\r\\n%\\r\\n% (c) 2023 Canadian Internet Registration Authority, (http://www.cira.ca/)\\r\\n\",\n",
       " '% This is the RIPE Database query service.\\n% The objects are in RPSL format.\\n%\\n% The RIPE Database is subject to Terms and Conditions.\\n% See http://www.ripe.net/db/support/db-terms-conditions.pdf\\n\\n% Note: this output has been filtered.\\n%       To receive output for a database update, use the \"-B\" flag.\\n\\n%ERROR:101: no entries found\\n%\\n% No entries found in source RIPE.\\n\\n% This query was served by the RIPE Database Query Service version 1.106 (BUSA)\\n\\n\\n',\n",
       " 'Domain: uni-muenster.de\\nStatus: connect\\n',\n",
       " 'Domain Name: hmfl.ac.cn\\nROID: 20080722s10071s02645069-cn\\nDomain Status: ok\\nRegistrant: 中国科学院合肥物质科学研究院\\nRegistrant Contact Email: mei@hfcas.ac.cn\\nSponsoring Registrar: 北京中科三方网络技术有限公司\\nName Server: ns.hfcas.ac.cn\\nName Server: ns1.hfcas.ac.cn\\nRegistration Time: 2008-07-22 08:30:06\\nExpiration Time: 2024-07-22 08:30:06\\nDNSSEC: unsigned\\n',\n",
       " '   Domain Name: SINA.COM\\r\\n   Registry Domain ID: 2243615_DOMAIN_COM-VRSN\\r\\n   Registrar WHOIS Server: whois.paycenter.com.cn\\r\\n   Registrar URL: http://www.xinnet.com\\r\\n   Updated Date: 2022-11-21T06:31:14Z\\r\\n   Creation Date: 1998-09-16T04:00:00Z\\r\\n   Registry Expiry Date: 2025-09-15T04:00:00Z\\r\\n   Registrar: Xin Net Technology Corporation\\r\\n   Registrar IANA ID: 120\\r\\n   Registrar Abuse Contact Email: supervision@xinnet.com\\r\\n   Registrar Abuse Contact Phone: +86.4008182233\\r\\n   Domain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited\\r\\n   Domain Status: clientUpdateProhibited https://icann.org/epp#clientUpdateProhibited\\r\\n   Domain Status: serverDeleteProhibited https://icann.org/epp#serverDeleteProhibited\\r\\n   Domain Status: serverTransferProhibited https://icann.org/epp#serverTransferProhibited\\r\\n   Domain Status: serverUpdateProhibited https://icann.org/epp#serverUpdateProhibited\\r\\n   Name Server: NS1.SINA.COM\\r\\n   Name Server: NS1.SINA.COM.CN\\r\\n   Name Server: NS2.SINA.COM\\r\\n   Name Server: NS2.SINA.COM.CN\\r\\n   Name Server: NS3.SINA.COM\\r\\n   Name Server: NS3.SINA.COM.CN\\r\\n   Name Server: NS4.SINA.COM\\r\\n   DNSSEC: unsigned\\r\\n   URL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/\\r\\n>>> Last update of whois database: 2023-03-21T11:41:13Z <<<\\r\\n\\r\\nFor more information on Whois status codes, please visit https://icann.org/epp\\r\\n\\r\\nNOTICE: The expiration date displayed in this record is the date the\\r\\nregistrar\\'s sponsorship of the domain name registration in the registry is\\r\\ncurrently set to expire. This date does not necessarily reflect the expiration\\r\\ndate of the domain name registrant\\'s agreement with the sponsoring\\r\\nregistrar.  Users may consult the sponsoring registrar\\'s Whois database to\\r\\nview the registrar\\'s reported date of expiration for this registration.\\r\\n\\r\\nTERMS OF USE: You are not authorized to access or query our Whois\\r\\ndatabase through the use of electronic processes that are high-volume and\\r\\nautomated except as reasonably necessary to register domain names or\\r\\nmodify existing registrations; the Data in VeriSign Global Registry\\r\\nServices\\' (\"VeriSign\") Whois database is provided by VeriSign for\\r\\ninformation purposes only, and to assist persons in obtaining information\\r\\nabout or related to a domain name registration record. VeriSign does not\\r\\nguarantee its accuracy. By submitting a Whois query, you agree to abide\\r\\nby the following terms of use: You agree that you may use this Data only\\r\\nfor lawful purposes and that under no circumstances will you use this Data\\r\\nto: (1) allow, enable, or otherwise support the transmission of mass\\r\\nunsolicited, commercial advertising or solicitations via e-mail, telephone,\\r\\nor facsimile; or (2) enable high volume, automated, electronic processes\\r\\nthat apply to VeriSign (or its computer systems). The compilation,\\r\\nrepackaging, dissemination or other use of this Data is expressly\\r\\nprohibited without the prior written consent of VeriSign. You agree not to\\r\\nuse electronic processes that are automated and high-volume to access or\\r\\nquery the Whois database except as reasonably necessary to register\\r\\ndomain names or modify existing registrations. VeriSign reserves the right\\r\\nto restrict your access to the Whois database in its sole discretion to ensure\\r\\noperational stability.  VeriSign may restrict or terminate your access to the\\r\\nWhois database for failure to abide by these terms of use. VeriSign\\r\\nreserves the right to modify these terms at any time.\\r\\n\\r\\nThe Registry database contains ONLY .COM, .NET, .EDU domains and\\r\\nRegistrars.\\r\\n',\n",
       " 'Out of this registry.\\n',\n",
       " 'Out of this registry.\\n',\n",
       " 'Invalid parameter:stu.cdutcm.edu.cn\\n',\n",
       " 'Out of this registry.\\n',\n",
       " \"    Domain name:\\r\\n        ac.uk\\r\\n\\r\\n    Registrant:\\r\\n        The JNT Association\\r\\n\\r\\n    Registrant type:\\r\\n        Other\\r\\n\\r\\n    Registrant's address:\\r\\n        Lumen House\\r\\n        Library Avenue\\r\\n        Harwell\\r\\n        Didcot\\r\\n        OX11 OSG\\r\\n        GB\\r\\n\\r\\n    Registrar:\\r\\n        No registrar listed.  This domain is directly registered with Nominet.\\r\\n\\r\\n    Relevant dates:\\r\\n        Registered on: before Aug-1996\\r\\n    Registration status:\\r\\n        No registration status listed.\\r\\n\\r\\n    Name servers:\\r\\n        ns0.ja.net.               \\r\\n        ns2.ja.net.               \\r\\n        ns3.ja.net.               \\r\\n        ns4.ja.net.               \\r\\n        auth03.ns.uu.net.               \\r\\n        ns1.surfnet.nl.               \\r\\n        dns-3.dfn.de.               \\r\\n\\r\\n\\r\\n    WHOIS lookup made at 11:41:34 21-Mar-2023\\r\\n\\r\\n-- \\r\\nThis WHOIS information is provided for free by Nominet UK the central registry\\r\\nfor .uk domain names. This information and the .uk WHOIS are:\\r\\n\\r\\n    Copyright Nominet UK 1996 - 2023.\\r\\n\\r\\nYou may not access the .uk WHOIS or use any data from it except as permitted\\r\\nby the terms of use available in full at https://www.nominet.uk/whoisterms,\\r\\nwhich includes restrictions on: (A) use of the data for advertising, or its\\r\\nrepackaging, recompilation, redistribution or reuse (B) obscuring, removing\\r\\nor hiding any or all of this notice and (C) exceeding query rate or volume\\r\\nlimits. The data is provided on an 'as-is' basis and may lag behind the\\r\\nregister. Access may be withdrawn or restricted at any time. \\r\\n\",\n",
       " 'Out of this registry.\\n',\n",
       " '% This is the RIPE Database query service.\\n% The objects are in RPSL format.\\n%\\n% The RIPE Database is subject to Terms and Conditions.\\n% See http://www.ripe.net/db/support/db-terms-conditions.pdf\\n\\n% Note: this output has been filtered.\\n%       To receive output for a database update, use the \"-B\" flag.\\n\\n%ERROR:101: no entries found\\n%\\n% No entries found in source RIPE.\\n\\n% This query was served by the RIPE Database Query Service version 1.106 (SHETLAND)\\n\\n\\n',\n",
       " '   Domain Name: GMAIL.COM\\r\\n   Registry Domain ID: 4667231_DOMAIN_COM-VRSN\\r\\n   Registrar WHOIS Server: whois.markmonitor.com\\r\\n   Registrar URL: http://www.markmonitor.com\\r\\n   Updated Date: 2022-07-11T09:25:59Z\\r\\n   Creation Date: 1995-08-13T04:00:00Z\\r\\n   Registry Expiry Date: 2023-08-12T04:00:00Z\\r\\n   Registrar: MarkMonitor Inc.\\r\\n   Registrar IANA ID: 292\\r\\n   Registrar Abuse Contact Email: abusecomplaints@markmonitor.com\\r\\n   Registrar Abuse Contact Phone: +1.2086851750\\r\\n   Domain Status: clientDeleteProhibited https://icann.org/epp#clientDeleteProhibited\\r\\n   Domain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited\\r\\n   Domain Status: clientUpdateProhibited https://icann.org/epp#clientUpdateProhibited\\r\\n   Domain Status: serverDeleteProhibited https://icann.org/epp#serverDeleteProhibited\\r\\n   Domain Status: serverTransferProhibited https://icann.org/epp#serverTransferProhibited\\r\\n   Domain Status: serverUpdateProhibited https://icann.org/epp#serverUpdateProhibited\\r\\n   Name Server: NS1.GOOGLE.COM\\r\\n   Name Server: NS2.GOOGLE.COM\\r\\n   Name Server: NS3.GOOGLE.COM\\r\\n   Name Server: NS4.GOOGLE.COM\\r\\n   DNSSEC: unsigned\\r\\n   URL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/\\r\\n>>> Last update of whois database: 2023-03-21T11:41:28Z <<<\\r\\n\\r\\nFor more information on Whois status codes, please visit https://icann.org/epp\\r\\n\\r\\nNOTICE: The expiration date displayed in this record is the date the\\r\\nregistrar\\'s sponsorship of the domain name registration in the registry is\\r\\ncurrently set to expire. This date does not necessarily reflect the expiration\\r\\ndate of the domain name registrant\\'s agreement with the sponsoring\\r\\nregistrar.  Users may consult the sponsoring registrar\\'s Whois database to\\r\\nview the registrar\\'s reported date of expiration for this registration.\\r\\n\\r\\nTERMS OF USE: You are not authorized to access or query our Whois\\r\\ndatabase through the use of electronic processes that are high-volume and\\r\\nautomated except as reasonably necessary to register domain names or\\r\\nmodify existing registrations; the Data in VeriSign Global Registry\\r\\nServices\\' (\"VeriSign\") Whois database is provided by VeriSign for\\r\\ninformation purposes only, and to assist persons in obtaining information\\r\\nabout or related to a domain name registration record. VeriSign does not\\r\\nguarantee its accuracy. By submitting a Whois query, you agree to abide\\r\\nby the following terms of use: You agree that you may use this Data only\\r\\nfor lawful purposes and that under no circumstances will you use this Data\\r\\nto: (1) allow, enable, or otherwise support the transmission of mass\\r\\nunsolicited, commercial advertising or solicitations via e-mail, telephone,\\r\\nor facsimile; or (2) enable high volume, automated, electronic processes\\r\\nthat apply to VeriSign (or its computer systems). The compilation,\\r\\nrepackaging, dissemination or other use of this Data is expressly\\r\\nprohibited without the prior written consent of VeriSign. You agree not to\\r\\nuse electronic processes that are automated and high-volume to access or\\r\\nquery the Whois database except as reasonably necessary to register\\r\\ndomain names or modify existing registrations. VeriSign reserves the right\\r\\nto restrict your access to the Whois database in its sole discretion to ensure\\r\\noperational stability.  VeriSign may restrict or terminate your access to the\\r\\nWhois database for failure to abide by these terms of use. VeriSign\\r\\nreserves the right to modify these terms at any time.\\r\\n\\r\\nThe Registry database contains ONLY .COM, .NET, .EDU domains and\\r\\nRegistrars.\\r\\n',\n",
       " 'This Registry database contains ONLY .EDU domains.\\nThe data in the EDUCAUSE Whois database is provided\\nby EDUCAUSE for information purposes in order to\\nassist in the process of obtaining information about\\nor related to .edu domain registration records.\\n\\nThe EDUCAUSE Whois database is authoritative for the\\n.EDU domain.\\n\\nA Web interface for the .EDU EDUCAUSE Whois Server is\\navailable at: http://whois.educause.edu\\n\\nBy submitting a Whois query, you agree that this information\\nwill not be used to allow, enable, or otherwise support\\nthe transmission of unsolicited commercial advertising or\\nsolicitations via e-mail.  The use of electronic processes to\\nharvest information from this server is generally prohibited\\nexcept as reasonably necessary to register or modify .edu\\ndomain names.\\n\\n-------------------------------------------------------------\\n\\nDomain Name: PURDUE.EDU\\n\\nRegistrant:\\n\\tPurdue University\\n\\t155 S. Grant Street\\n\\tYoung Hall\\n\\tWest Lafayette, IN 47907-2114\\n\\tUSA\\n\\nAdministrative Contact:\\n\\tPurdue Hostmaster\\n\\tPurdue University\\n\\t191 North University Street\\n\\tTelecommunications Building\\n\\tWest Lafayette, IN 47907-2068\\n\\tUSA\\n\\t+1.7654944000\\n\\thostmaster@purdue.edu\\n\\nTechnical Contact:\\n\\tPurdue Hostmaster\\n\\tPurdue University\\n\\t191 North University Street\\n\\tTelecommunications Building\\n\\tWest Lafayette, IN 47907-2068\\n\\tUSA\\n\\t+1.7654944000\\n\\thostmaster@purdue.edu\\n\\nName Servers:\\n\\tPENDRAGON.CS.PURDUE.EDU\\n\\tNS4.PURDUE.EDU\\n\\tHARBOR.ECN.PURDUE.EDU\\n\\tNS1.RICE.EDU\\n\\tNS3.PURDUE.EDU\\n\\nDomain record activated:    24-Apr-1985\\nDomain record last updated: 26-Dec-2022\\nDomain expires:             31-Jul-2023\\n',\n",
       " '% This is the RIPE Database query service.\\n% The objects are in RPSL format.\\n%\\n% The RIPE Database is subject to Terms and Conditions.\\n% See http://www.ripe.net/db/support/db-terms-conditions.pdf\\n\\n% Note: this output has been filtered.\\n%       To receive output for a database update, use the \"-B\" flag.\\n\\n%ERROR:101: no entries found\\n%\\n% No entries found in source RIPE.\\n\\n% This query was served by the RIPE Database Query Service version 1.106 (SHETLAND)\\n\\n\\n',\n",
       " 'This Registry database contains ONLY .EDU domains.\\nThe data in the EDUCAUSE Whois database is provided\\nby EDUCAUSE for information purposes in order to\\nassist in the process of obtaining information about\\nor related to .edu domain registration records.\\n\\nThe EDUCAUSE Whois database is authoritative for the\\n.EDU domain.\\n\\nA Web interface for the .EDU EDUCAUSE Whois Server is\\navailable at: http://whois.educause.edu\\n\\nBy submitting a Whois query, you agree that this information\\nwill not be used to allow, enable, or otherwise support\\nthe transmission of unsolicited commercial advertising or\\nsolicitations via e-mail.  The use of electronic processes to\\nharvest information from this server is generally prohibited\\nexcept as reasonably necessary to register or modify .edu\\ndomain names.\\n\\n-------------------------------------------------------------\\n\\nDomain Name: UCI.EDU\\n\\nRegistrant:\\n\\tUniversity of California, Irvine\\n\\t6366 Ayala Science Library\\n\\tIrvine, CA 92697-1175\\n\\tUSA\\n\\nAdministrative Contact:\\n\\tDomain Admin\\n\\tUniversity of California, Irvine\\n\\t6366 Ayala Science Library\\n\\tIrvine, CA 92697-1175\\n\\tUSA\\n\\t+1.9498242222\\n\\toit-nsp@uci.edu\\n\\nTechnical Contact:\\n\\tDomain Admin\\n\\tUniversity of California, Irvine\\n\\t6366 Ayala Science Library\\n\\tIrvine, CA 92697-1175\\n\\tUSA\\n\\t+1.9498242222\\n\\toit-nsp@uci.edu\\n\\nName Servers:\\n\\tNS6.SERVICE.UCI.EDU\\n\\tNS5.SERVICE.UCI.EDU\\n\\nDomain record activated:    30-Sep-1985\\nDomain record last updated: 26-Dec-2022\\nDomain expires:             31-Jul-2023\\n',\n",
       " 'Domain Name: auckland.ac.nz\\r\\nRegistrar URL: https://www.webcentral.com.au/\\r\\nUpdated Date: 2022-12-13T14:09:31Z\\r\\nCreation Date: 1996-12-08T11:00:00Z\\r\\nOriginal Created: 1996-12-08T11:00:00Z\\r\\nRegistrar: Webcentral Group Ltd T/A Melbourne IT\\r\\nDomain Status: ok https://icann.org/epp#ok\\r\\nName Server: dns1.auckland.ac.nz\\r\\nName Server: dns2.auckland.ac.nz\\r\\nDNSSEC: unsigned\\r\\nURL of the Domain Name Commission Limited Inaccuracy Complaint Form: https://dnc.org.nz/enquiry-form/\\r\\n>>> Last update of WHOIS database: 2023-03-21T11:41:47Z <<<\\r\\n\\r\\n% Terms of Use\\r\\n%\\r\\n% By submitting a WHOIS query you are entering into an agreement with Domain\\r\\n% Name Commission Ltd on the following terms and conditions, and subject to\\r\\n% all relevant .nz Policies and procedures as found at https://dnc.org.nz/.\\r\\n%\\r\\n% It is prohibited to:\\r\\n% - Send high volume WHOIS queries with the effect of downloading part of or\\r\\n%   all of the .nz Register or collecting register data or records;\\r\\n% - Access the .nz Register in bulk through the WHOIS service (ie. where a\\r\\n%   user is able to access WHOIS data other than by sending individual queries\\r\\n%   to the database);\\r\\n% - Use WHOIS data to allow, enable, or otherwise support mass unsolicited\\r\\n%   commercial advertising, or mass solicitations to registrants or to\\r\\n%   undertake market research via direct mail, electronic mail, SMS, telephone\\r\\n%   or any other medium;\\r\\n% - Use WHOIS data in contravention of any applicable data and privacy laws,\\r\\n%   including the Unsolicited Electronic Messages Act 2007;\\r\\n% - Store or compile WHOIS data to build up a secondary register of\\r\\n%   information;\\r\\n% - Publish historical or non-current versions of WHOIS data; and\\r\\n% - Publish any WHOIS data in bulk.\\r\\n%\\r\\n% Copyright Domain Name Commission Limited (a company wholly-owned by Internet\\r\\n% New Zealand Incorporated) which may enforce its rights against any person or\\r\\n% entity that undertakes any prohibited activity without its written\\r\\n% permission.\\r\\n%\\r\\n% The WHOIS service is provided by Internet New Zealand Incorporated.\\r\\n%\\r\\n% Additional information may be available at https://www.dnc.org.nz\\r\\n',\n",
       " 'Out of this registry.\\n']"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "whois_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "2318fd68-d5fc-4db9-b4f8-8abb4e272277",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "% DOTGOV WHOIS Server ready\n",
      "   Domain Name: NASA.GOV\n",
      "   Status: ACTIVE\n",
      "   Security Contact Email: soc@nasa.gov\n",
      "\n",
      ">>> Last update of whois database: 2023-03-21T11:19:02Z <<<\n",
      "\n",
      "Please be advised that this whois server only contains information pertaining\n",
      "to the .GOV domain. For information for other domains please use the whois\n",
      "server at RS.INTERNIC.NET. \n",
      "\n"
     ]
    }
   ],
   "source": [
    "import whois11\n",
    "\n",
    "print(whois11.whois('nasa.gov'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "132321a3-c5ce-49f1-9fa2-6506abc69aa7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pdftotext\n",
    "import io\n",
    "from urllib.request import (\n",
    "    Request,\n",
    "    urlopen,\n",
    ")\n",
    "def file_download(url):\n",
    "    req = Request(url, headers={\"User-Agent\": \"Mozilla/5.0\"})\n",
    "    pdf_pages = pdftotext.PDF(io.BytesIO(urlopen(req).read()))\n",
    "    return pdf_pages\n",
    "    \n",
    "def text_from_pdf_bytes(pdf_bytes):\n",
    "    pdf_text = pdftotext.PDF(io.BytesIO(urlopen(req).read()))\n",
    "    return \"\\n\\n\".join(pdf_text)\n",
    "\n",
    "def head_extract(pdf_bytes):\n",
    "    text = text_from_pdf_file(path)\n",
    "    abstract_pos = re.search(r\"[aA]bstract\", text)\n",
    "    return text[: abstract_pos.start(0) if abstract_pos else 10000]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "a768bdcc-07b2-4c6c-8ff2-d1159bcbcd97",
   "metadata": {},
   "outputs": [],
   "source": [
    "pdf_pages = file_download(\"http://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC9983066&blobtype=pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "6f9f3664-7e98-4180-b979-dcadf1c9f2c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pdf_pages)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "ce247c33-fbd0-4f84-9694-148a74c7fe68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'                                                 MEDICINE INTERNATIONAL 3: 11, 2023\\n           The combination of positive anti‑WDR1 antibodies with\\n       negative anti‑CFL1 antibodies in serum is a poor prognostic\\n                      factor for patients with esophageal carcinoma\\n          MASAAKI ITO1, SATOSHI YAJIMA2, TAKASHI SUZUKI2, YOKO OSHIMA2, TATSUKI NANAMI2,\\n      MAKOTO SUMAZAKI2, FUMIAKI SHIRATORI2, HAO WANG3,4, LIUBING HU4, HIROTAKA TAKIZAWA5,\\n                  SHU‑YANG LI6, YASUO IWADATE6, TAKAKI HIWASA1,6 and HIDEAKI SHIMADA1,2\\n1\\n  Department of Clinical Oncology, Toho University Graduate School of Medicine; 2Department of Gastroenterological Surgery,\\n  Toho University School of Medicine, Tokyo 143‑8541, Japan; 3Stroke Center, The First Affiliated Hospital, Jinan University;\\n    4\\n      Department of Anesthesiology, Stroke Center, The First Affiliated Hospital and Health Science Center, Jinan University,\\n        Guangzhou, Guangdong 510630, P.R. China; 5Port Square Kashiwado Clinic, Kashiwado Memorial Foundation,\\n  Chiba 260‑0025; 6Department of Neurological Surgery, Graduate School of Medicine, Chiba University, Chiba 260‑8670, Japan\\n                                       Received October 8, 2022; Accepted January 23, 2023\\n                                                        DOI: 10.3892/mi.2023.71\\nAbstract. WD repeat‑containing protein 1 (WDR1) regulates            s‑WDR1‑Abs or s‑CFL1‑Abs alone in the Kaplan‑Meier test,\\nthe cofilin 1 (CFL1) activity, promotes cytoskeleton remod‑          the patients in the s‑WDR1‑Ab‑positive and s‑CFL1‑Ab‑nega‑\\neling, and thus, facilitates cell migration and invasion. A          tive groups exhibited a significantly poorer prognosis in the\\nprevious study reported that autoantibodies against CFL1 and         overall survival analysis. On the whole, the present study\\nβ‑actin were useful biomarkers for diagnosing and predicting         demonstrates that the combination of positive anti‑WDR1 anti‑\\nthe prognosis of patients with esophageal carcinoma.                 bodies with negative anti‑CFL1 antibodies in serum may be a\\nTherefore, the present study aimed to evaluate the serum             poor prognostic factor for patients with esophageal carcinoma.\\nlevels of anti‑WDR1 antibodies (s‑WDR1‑Abs) combined\\nwith serum levels of anti‑CFL1 antibodies (s‑CFL1‑Abs) in            Introduction\\npatients with esophageal carcinoma. Serum samples obtained\\nfrom 192 patients with esophageal carcinoma and other solid          Esophageal carcinoma is a rapidly progressive disease and\\ncancers. And s‑WDR1‑Ab and s‑CFL1‑Ab titers were analyzed            is life‑threatening even at the early stages. The therapeutic\\nusing the amplified luminescent proximity homogeneous                efficacy of esophageal carcinoma is less satisfactory than that\\nassay‑linked immunosorbent assay. Compared with those                of several other types of cancer; therefore, the detection and\\nof healthy donors, the s‑WDR1‑Ab levels were significantly           diagnosis at the early stage of carcinoma are indispensable for\\nhigher in the 192 patients with esophageal, whereas these            improving the therapeutic outcomes of patients (1).\\nwere not significantly higher in the samples from patients with          The serum levels of several antigens and antibodies have\\ngastric, colorectal, lung, or breast cancer. In 91 patients treated  long been measured using the serological analysis of recom‑\\nwith surgery, sex, tumor depth, lymph node metastasis, stage         binant tumor cDNA expression libraries (SEREX) method, a\\nand C‑reactive protein levels were significantly associated          useful screening method for tumor markers (2). SEREX uses\\nwith overall survival, as determined using the log‑rank test,        patient serum to immunoscreen cDNA libraries prepared from\\nwhereas the squamous cell carcinoma antigen, p53 antibody            cancer specimens. By sequencing the isolated cDNA clones,\\nand s‑WDR1‑Ab levels tended to be associated with a worse            the SEREX method is suitable as for the large‑scale screening\\nprognosis. Although no significant difference was observed           for tumor antigens. Large‑scale SEREX screening has already\\nin the survival between the positive and negative groups of          identified numerous antibodies or antigens, such as trophoblast\\n                                                                     cell surface antigen 2 (3), solute carrier family 2 member 1 (4),\\n                                                                     striatin 4 (5), LDL receptor related protein associated protein\\n                                                                     1 (6), proprotein convertase subtilisin/kexin type 9 (7), cofilin\\nCorrespondence to: Dr Hideaki Shimada, Department of                 1 (CFL1) and β ‑actin (ACTB) (8). CFL1 can depolymerize\\nGastroenterological Surgery, Toho University School of Medicine,     F‑actin in a pH‑dependent manner and is also involved in lung,\\n6‑11‑1 Omori‑Nishi, Ota‑ku, Tokyo 143‑8541, Japan                    pancreatic, and gastric cancer invasion and metastasis (9‑11).\\nE‑mail: hideaki.shimada@med.toho‑u.ac.jp                                 WD repeat‑containing protein 1 (WDR1; also known as\\n                                                                     actin‑interacting protein 1) regulates cofilin activity, promotes\\nKey words: WD repeat‑containing protein 1, cofilin 1, esophageal     cytoskeletal remodeling, and thus facilitates cell migration and\\ncarcinoma, antibody biomarker, overall survival                      invasion (12). The presence of autoantibodies against WDR1\\n                                                                     has been found in the sera of patients who have suffered a\\n                                                                     transient ischemic attack using SEREX screening (13). The\\n'"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdf_pages[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2439d128-fead-4ddd-9f8c-7baefbb158d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['https://en.wikipedia.org/wiki/Toho_University', 'https://www.wikidata.org/wiki/Q107016761', 'https://www.toho-u.ac.jp/english/', 'https://m.facebook.com/profile.php?id=112389765445207', 'https://www.timeshighereducation.com/world-university-rankings/toho-university', 'https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0280475', 'http://www.tohoku.ac.jp/en/', 'https://en-academic.com/dic.nsf/enwiki/4346092', 'https://www.timeshighereducation.com/world-university-rankings/toho-university', 'https://www.tripadvisor.com.sg/ShowTopic-g298184-i861-k7892653-Plz_help_with_suggestions_regards_hospitals-Tokyo_Tokyo_Prefecture_Kanto.html', 'http://www.tohoku.ac.jp/']\n"
     ]
    }
   ],
   "source": [
    "from googlesearch import search \n",
    "\n",
    "list_of_queries = [\"wikipedia med.toho‑u.ac.jp\"]\n",
    "results = []\n",
    "\n",
    "for query in list_of_queries:\n",
    "    results.extend(list(search(query, num_results=10)))\n",
    "\n",
    "print(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28ed1818-cf94-467c-93a5-9e4cba826802",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87d5c9d7-e06f-4a11-850e-6a5b76064091",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "vscode": {
   "interpreter": {
    "hash": "f9f85f796d01129d0dd105a088854619f454435301f6ffec2fea96ecbd9be4ac"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
